<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Genome Mining in Prokaryotes: Homologous BGC Clusterization</title><meta name="viewport" content="width=device-width, initial-scale=1"><script src="assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="assets/styles.css"><script src="assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="favicons/incubator/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="favicons/incubator/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="favicons/incubator/favicon-16x16.png"><link rel="manifest" href="favicons/incubator/site.webmanifest"><link rel="mask-icon" href="favicons/incubator/safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" media="(prefers-color-scheme: light)" content="white"><meta name="theme-color" media="(prefers-color-scheme: dark)" content="black"></head><body>
    <header id="top" class="navbar navbar-expand-md top-nav incubator"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="incubator-logo" alt="Lesson Description" src="assets/images/incubator-logo.svg"><span class="badge text-bg-danger">
          <abbr title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.">
            <a href="https://docs.carpentries.org/resources/curriculum/lesson-life-cycle.html" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
              Pre-Alpha
            </a>
            <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text"><li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul></li>
      </div>


      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Learner View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='instructor/05-bigslice-bigfam.html';">Instructor View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Lesson Description" src="assets/images/incubator-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Genome Mining in Prokaryotes
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Genome Mining in Prokaryotes
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="reference.html#glossary">Glossary</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="profiles.html">Learner Profiles</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><li><a class="dropdown-item" href="discuss.html">Discussion</a></li><li><a class="dropdown-item" href="reference.html">Glossary</a></li>
          </ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Genome Mining in Prokaryotes
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 34%" class="percentage">
    34%
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: 34%" aria-valuenow="34" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text"><li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul></li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Learner View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="instructor/05-bigslice-bigfam.html">Instructor View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->

            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Setup</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="01-introduction.html">1. Introduction to Genome Mining</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="02-antismash.html">2. Secondary metabolite biosynthetic gene cluster identification</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="03-antiSMASHdb-MiBiG.html">3. Genome Mining Databases</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="04-bigscape.html">4. BGC Similarity Networks</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
      <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapsecurrent" aria-expanded="true" aria-controls="flush-collapsecurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        5. Homologous BGC Clusterization
        </span>
      </button>
    </div><!--/div.accordion-header-->

    <div id="flush-collapsecurrent" class="accordion-collapse collapse show" aria-labelledby="flush-headingcurrent" data-bs-parent="#accordionFlushcurrent">
      <div class="accordion-body">
        <ul><li><a href="#big-slice-and-big-fam-a-set-of-tools-to-compare-the-microbial-metabolic-diversity">BiG-SLiCE and BiG-FAM a set of tools to compare the microbial
metabolic diversity</a></li>
<li><a href="#big-fam">BiG-FAM</a></li>
        </ul></div><!--/div.accordion-body-->
    </div><!--/div.accordion-collapse-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="06-corason.html">6. Finding Variation on Genomic Vicinities</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="07-evomining.html">7. Evolutionary Genome Mining</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="08-GatorGC.html">8. GATOR-GC: Genomic Assessment Tool for Orthologous Regions and Gene Clusters</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush10">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading10">
        <a href="09-metabolomics.html">9. Metabolomics workshop</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush11">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading11">
        <a href="10-OtherResources.html">10. Other Resources</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="reference.html#glossary">Glossary</a>
                      </li>
                      <li>
                        <a href="profiles.html">Learner Profiles</a>
                      </li>
                      <li><a href="discuss.html">Discussion</a></li><li><a href="reference.html">Glossary</a></li>
                    </ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources"><a href="aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">

            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="04-bigscape.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="06-corason.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="04-bigscape.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: BGC Similarity
        </a>
        <a class="chapter-link float-end" href="06-corason.html" rel="next">
          Next: Finding Variation on...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>Homologous BGC Clusterization</h1>
        <p>Last updated on 2026-02-20 |

        <a href="https://github.com/carpentries-incubator/genome-mining/edit/main/episodes/05-bigslice-bigfam.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>



        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>How can I identify Gene Cluster Families?</li>
<li>How can I predict the production of similar metabolites</li>
<li>How can I clusterize BGCs into groups that produce similar
metabolites?</li>
<li>How can I compare the metabolic capability of different bacterial
lineages?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>Understand the purpose of using <code>BiG-SLiCE</code> and
<code>BiG-FAM</code>
</li>
<li>Learn the inputs required to run <code>BiG-SLiCE</code> and
<code>BiG-FAM</code>
</li>
<li>Run an example with our <em>Streptococcus</em> data on both
softwares</li>
<li>Understand the obtained results</li>
</ul></div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="big-slice-and-big-fam-a-set-of-tools-to-compare-the-microbial-metabolic-diversity">BiG-SLiCE and BiG-FAM a set of tools to compare the microbial
metabolic diversity<a class="anchor" aria-label="anchor" href="#big-slice-and-big-fam-a-set-of-tools-to-compare-the-microbial-metabolic-diversity"></a></h2>
<hr class="half-width"><p>Around the curriculum, we have been learning about the metabolic
capability of bacteria lineages encoded in BGCs. Before we start this
lesson, let’s remember that the products of these clusters are
biomolecules that are (mostly) not essential for life. They take the
role of “tools” that offer some ecological and physiological advantage
to the bacterial lineage producing it.</p>
<p><a href="fig/02-04-01.png"> &lt;img src=“fig/02-04-01.png”
alt=“Biosynthetic Gene Cluster encoding three different biomolecules,
which might not be essential for survival. In the figure, the first gene
in the cluster synthesizes a shield, the second gene is able to produce
a sword. All together with the third gene, the BGC provides
advantageous”tools” to the organism, represented by a warrior.” /&gt;
</a></p>
<p>The presence/absence of a set of BGCs in a genome, can be associated
to the ecological niches and (micro)environments which the lineage
faces.</p>
<p>The counterpart of this can be taken as how biosynthetically diverse
a bacterial lineage can be due to its environment.</p>
<p><a href="fig/02-04-02.png"></a></p>
<figure><img src="fig/02-04-02.png" alt="Three biomolecules are depicted in three different colors; blue, red and purple. These molecules are present in BGCs of diverse bacterial lineages and in turn grouped into Gene Cluster Families (GCFs). GF1 contains three domains related to the blue molecule and three from the red molecule. GF2 contains four domains of each of the molecules, blue, red and purple. GCF3 and GCF4 contain each three domains associated with the purple biomolecule but different from each other. Every GCF produces a different metabolite, here represented as weapons or tools." class="figure mx-auto d-block"></figure><p></p>
<p>To answer some of this questions, we need a way to compare the
metabolic capabilities of a set of diverse bacterial lineages. One way
to do this is to compare the state (identity) of homologous BGCs. The
homologous clusters that share a similar domain architecture and in
consequence will produce a similar metabolite, are grouped into a bigger
group that we call <strong>Gene Cluster Families (GCFs)</strong>.</p>
<p>There are some programs capable of clustering BGCs in GCFs,
<em>e.g.</em><a href="https://github.com/petercim/ClusterFinder" class="external-link">ClusterFinder</a> and
<a href="https://github.com/Merck/deepbgc" class="external-link">DeepBGC</a>.
<code>BiG-SLICE</code> is a software that has been <a href="https://academic.oup.com/gigascience/article/10/1/giaa154/6092777" class="external-link">designed</a>
to do this work in less time, even when dealing with large datasets (1.2
million BGCs). This characteristic makes it a good tool to compare the
metabolic diversity of several bacterial lineages.</p>
<div class="section level3">
<h3 id="the-input-for-big-slice">The input for BiG-SLiCE<a class="anchor" aria-label="anchor" href="#the-input-for-big-slice"></a></h3>
<p>Firstly, activate the conda environment, where <code>BiG-SLiCE</code>
has been installed as well as all the softwares required for its
usage:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">$</span> conda activate GenomeMining_Global</span></code></pre>
</div>
<p>You will have now a <code>(GenomeMining)</code> label at the
beginning of the prompt line.</p>
<p>If we seek for help using the <code>--help</code> flag of
<code>bigslice</code>, we can get a first glance at the input that we
need:</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="ex">$</span> bigslice <span class="at">--help</span> <span class="kw">|</span> <span class="fu">head</span> <span class="at">-n20</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>usage: bigslice [-i &lt;folder_path&gt;] [--resume] [--complete] [--threshold T]
            	[--threshold_pct &lt;N&gt;] [--query &lt;folder_path&gt;]
            	[--query_name &lt;name&gt;] [--run_id &lt;id&gt;] [--n_ranks N_RANKS]
            	[-t &lt;N&gt;] [--hmmscan_chunk_size &lt;N&gt;] [--subpfam_chunk_size &lt;N&gt;]
            	[--extraction_chunk_size EXTRACTION_CHUNK_SIZE] [--scratch]
            	[-h] [--program_db_folder PROGRAM_DB_FOLDER] [--version]
            	&lt;output_folder_path&gt;

                        	_________
 ___	_____|\____|\____|\ \____	\
|   \  |   	} 	} 	}  ___)_ _/__
| &gt;  | _--/--|/----|/----|/__/  __||  __|
|   &lt; | ||  __/ (  (| | \___/  /__/|  _|
| &gt;  || || |_ | _)  ) |_ | |\  \__ | |__
|____/|_| \___/|___/|___||_| \____||____| [ Version 1.1.0 ]

Biosynthetic Gene clusters - Super Linear Clustering Engine
(https://github.com/medema-group/bigslice)

positional arguments:</code></pre>
</div>
<p><code>BiG-SLiCE</code> is asking for a folder as the input. It is
important to highlight that the bones of this folder are the BGCs from a
group of genomes that has been obtained by <code>antiSMASH</code>. In
its <a href="https://github.com/medema-group/bigslice" class="external-link">GitHub</a> page,
we can search for an input folder <a href="https://github.com/medema-group/bigslice/tree/master/misc/input_folder_template" class="external-link">template</a>.
The folder needs three main components:</p>
<ul><li>
<strong>datasets.tsv:</strong> This is a file that <strong>needs to
have this exact name</strong>. This must be a tabular separated file
where the information of each dataset (BGCs from a group of genomes)
must be specified.</li>
</ul><p><a href="fig/02-04-03.png">
<img src="fig/02-04-03.png" alt="Example of tsv table composed by five columns and two rows.
                                  	The first row contains the title for each column; # Dataset name, Path to folder, Path to taxonomy, Description." class="figure"></a></p>
<ul><li><p><strong>dataset_n:</strong> Each dataset will be composed by the
BGCs of different genomes. We can give <code>BiG-SLiCE</code>
<strong>n</strong> groups of BGCs. The separation of the BGCs in
different datasets is due to the different characteristics that the
bacterial lineages can have. For example, if we have a set of bacterial
populations that came from maize roots and other obtained from tomato
roots, we can put them into dataset_1 and dataset_2
respectively.</p></li>
<li>
<p>**taxonomy_n.tsv:**This is also a tabular separated file that
will carry the taxonomic information of each of the genomes where the
input BGCs were found and the path to reach these BGCs inside the
input-folder:</p>
<ol style="list-style-type: decimal"><li>Genome folder name (ends with ‘/’)</li>
<li>Kingdom / Domain name</li>
<li>Class name</li>
<li>Order name</li>
<li>Family name</li>
<li>Genus name</li>
<li>Species name</li>
<li>Organism / Strain name</li>
</ol></li>
</ul><p>Here, we have an example of the structure of the
<code>input-folder</code>:</p>
<p><a href="fig/02-04-04.png">
<img src="fig/02-04-04.png" alt="Example of the structure of the input-folder.
                                  	The content of the directory input-folder is displayed in a tree-like format, listing the files and directories inside it." class="figure"></a></p>
</div>
<div class="section level3">
<h3 id="creating-the-input-folder">Creating the input-folder<a class="anchor" aria-label="anchor" href="#creating-the-input-folder"></a></h3>
<p>We will create a <code>bigslice</code> folder inside the
<code>results</code> directory.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="ex">$</span> cd ~/gm_workshop/results/genome_mining</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="ex">$</span> mkdir bigslice</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="ex">$</span> cd bigslice</span></code></pre>
</div>
<p>Now that we know what input <code>BiG-SLiCE</code> requires, we will
do our input-folder step by step. First, let’s remember how many genomes
we have:</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="ex">$</span> ls <span class="at">-F</span> ../antismash</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>agalactiae_18RS21/  agalactiae_A909/	agalactiae_COH1/
agalactiae_515/ 	agalactiae_CJB111/  agalactiae_H36B/
athermophilus_LMD-9/  athermophilus_LMG_18311/</code></pre>
</div>
<p>We have six <em>Streptococcus</em> genomes from the original paper,
and two<br>
public genomes. We will allocate their respective BGC <code>.gbk</code>s
in two <code>dataset</code>s, one for the genomes from Tettelin <em>et
al.</em> paper, and the other for the public ones.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="ex">$</span> mkdir input-folder</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="ex">$</span> mkdir input-folder/dataset_1</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="ex">$</span> mkdir input-folder/dataset_2</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="ex">$</span> mkdir input-folder/taxonomy</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a><span class="ex">$</span> tree</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>.
└── input-folder
	├── dataset_1
	├── dataset_2
	└── taxonomy

4 directories, 0 files</code></pre>
</div>
<p>Now we will create the folders for the BGCs of the dataset_1 genomes.
We will take advantage of the strain name of our <em>Streptococcus
agalactiae</em> lineages to create their folders:</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="ex">$</span> for i in 18RS21 COH1 515 H36B A909 CJB111<span class="kw">;</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>  <span class="cf">do</span> <span class="fu">mkdir</span> input-folder/dataset_1/Streptococcus_agalactiae_<span class="va">$i</span><span class="kw">;</span></span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>  <span class="cf">done</span></span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a><span class="ex">$</span> tree <span class="at">-F</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>.
└── input-folder/
	├── dataset_1/
	│   ├── Streptococcus_agalactiae_18RS21/
	│   ├── Streptococcus_agalactiae_515/
	│   ├── Streptococcus_agalactiae_A909/
	│   ├── Streptococcus_agalactiae_CJB111/
	│   ├── Streptococcus_agalactiae_COH1/
	│   └── Streptococcus_agalactiae_H36B/
	├── dataset_2/
	└── taxonomy/

10 directories, 0 files</code></pre>
</div>
<p>We will do the same for the public genomes:</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="ex">$</span> for i in LMD-9 LMG_18311<span class="kw">;</span></span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>  <span class="cf">do</span> <span class="fu">mkdir</span> input-folder/dataset_2/Streptococcus_thermophilus_<span class="va">$i</span><span class="kw">;</span></span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a>  <span class="cf">done</span></span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="ex">$</span> tree <span class="at">-F</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="bu">.</span></span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a><span class="ex">└──</span> input-folder/</span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a>	<span class="ex">├──</span> dataset_1/</span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a>	<span class="ex">│</span>   ├── Streptococcus_agalactiae_18RS21/</span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a>	<span class="ex">│</span>   ├── Streptococcus_agalactiae_515/</span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a>	<span class="ex">│</span>   ├── Streptococcus_agalactiae_A909/</span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a>	<span class="ex">│</span>   ├── Streptococcus_agalactiae_CJB111/</span>
<span id="cb12-8"><a href="#cb12-8" tabindex="-1"></a>	<span class="ex">│</span>   ├── Streptococcus_agalactiae_COH1/</span>
<span id="cb12-9"><a href="#cb12-9" tabindex="-1"></a>	<span class="ex">│</span>   └── Streptococcus_agalactiae_H36B/</span>
<span id="cb12-10"><a href="#cb12-10" tabindex="-1"></a>	<span class="ex">├──</span> dataset_2/</span>
<span id="cb12-11"><a href="#cb12-11" tabindex="-1"></a>	<span class="ex">│</span>   ├── Streptococcus_thermophilus_LMD-9/</span>
<span id="cb12-12"><a href="#cb12-12" tabindex="-1"></a>	<span class="ex">│</span>   └── Streptococcus_thermophilus_LMG_18311/</span>
<span id="cb12-13"><a href="#cb12-13" tabindex="-1"></a>	<span class="ex">└──</span> taxonomy/</span>
<span id="cb12-14"><a href="#cb12-14" tabindex="-1"></a></span>
<span id="cb12-15"><a href="#cb12-15" tabindex="-1"></a><span class="ex">12</span> directories, 0 files</span></code></pre>
</div>
<p>Next, we will copy the BGCs <code>.gbk</code> files from the
dataset_1 genomes from each <code>AntiSMASH</code> output directory to
its respective folder .</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="ex">$</span> ls input-folder/dataset_1/ <span class="kw">|</span> <span class="cf">while</span> <span class="bu">read</span> <span class="va">line</span><span class="kw">;</span></span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>  <span class="cf">do</span> <span class="fu">cp</span> ../antismash/output/<span class="va">$line</span>/<span class="pp">*</span>region<span class="pp">*</span>.gbk input-folder/dataset_1/<span class="va">$line</span><span class="kw">;</span></span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a>  <span class="cf">done</span></span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a><span class="ex">$</span> tree <span class="at">-F</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>.
└── input-folder/
	├── dataset_1/
	│   ├── Streptococcus_agalactiae_18RS21/
	│   │   ├── AAJO01000016.1.region001.gbk
	│   │   ├── AAJO01000043.1.region001.gbk
	│   │   └── AAJO01000226.1.region001.gbk
	│   ├── Streptococcus_agalactiae_515/
	│   │   ├── AAJP01000027.1.region001.gbk
	│   │   └── AAJP01000037.1.region001.gbk
	│   ├── Streptococcus_agalactiae_A909/
	│   │   ├── CP000114.1.region001.gbk
	│   │   └── CP000114.1.region002.gbk
	│   ├── Streptococcus_agalactiae_CJB111/
	│   │   ├── AAJQ01000010.1.region001.gbk
	│   │   └── AAJQ01000025.1.region001.gbk
	│   ├── Streptococcus_agalactiae_COH1/
	│   │   ├── AAJR01000002.1.region001.gbk
	│   │   └── AAJR01000044.1.region001.gbk
	│   └── Streptococcus_agalactiae_H36B/
	│   	├── AAJS01000020.1.region001.gbk
	│   	└── AAJS01000117.1.region001.gbk
	├── dataset_2/
	│   ├── Streptococcus_thermophilus_LMD-9/
	│   └── Streptococcus_thermophilus_LMG_18311/
	└── taxonomy/

12 directories, 13 files</code></pre>
</div>
<p>Again, we will do the same for the BGCs <code>.gbk</code>s of the
dataset_2:</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="ex">$</span> ls input-folder/dataset_2/ <span class="kw">|</span> <span class="cf">while</span> <span class="bu">read</span> <span class="va">line</span><span class="kw">;</span></span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a>  <span class="cf">do</span> <span class="fu">cp</span> ../antismash/output/<span class="va">$line</span>/<span class="pp">*</span>region<span class="pp">*</span>.gbk input-folder/dataset_2/<span class="va">$line</span><span class="kw">;</span></span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a>  <span class="cf">done</span></span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a><span class="ex">$</span> tree <span class="at">-F</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="bu">.</span></span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a><span class="ex">└──</span> input-folder/</span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a>	<span class="ex">├──</span> dataset_1/</span>
<span id="cb16-4"><a href="#cb16-4" tabindex="-1"></a>	<span class="ex">│</span>   ├── Streptococcus_agalactiae_18RS21/</span>
<span id="cb16-5"><a href="#cb16-5" tabindex="-1"></a>	<span class="ex">│</span>   │   ├── AAJO01000016.1.region001.gbk</span>
<span id="cb16-6"><a href="#cb16-6" tabindex="-1"></a>	<span class="ex">│</span>   │   ├── AAJO01000043.1.region001.gbk</span>
<span id="cb16-7"><a href="#cb16-7" tabindex="-1"></a>	<span class="ex">│</span>   │   └── AAJO01000226.1.region001.gbk</span>
<span id="cb16-8"><a href="#cb16-8" tabindex="-1"></a>	<span class="ex">│</span>   ├── Streptococcus_agalactiae_515/</span>
<span id="cb16-9"><a href="#cb16-9" tabindex="-1"></a>	<span class="ex">│</span>   │   ├── AAJP01000027.1.region001.gbk</span>
<span id="cb16-10"><a href="#cb16-10" tabindex="-1"></a>	<span class="ex">│</span>   │   └── AAJP01000037.1.region001.gbk</span>
<span id="cb16-11"><a href="#cb16-11" tabindex="-1"></a>	<span class="ex">│</span>   ├── Streptococcus_agalactiae_A909/</span>
<span id="cb16-12"><a href="#cb16-12" tabindex="-1"></a>	<span class="ex">│</span>   │   ├── CP000114.1.region001.gbk</span>
<span id="cb16-13"><a href="#cb16-13" tabindex="-1"></a>	<span class="ex">│</span>   │   └── CP000114.1.region002.gbk</span>
<span id="cb16-14"><a href="#cb16-14" tabindex="-1"></a>	<span class="ex">│</span>   ├── Streptococcus_agalactiae_CJB111/</span>
<span id="cb16-15"><a href="#cb16-15" tabindex="-1"></a>	<span class="ex">│</span>   │   ├── AAJQ01000010.1.region001.gbk</span>
<span id="cb16-16"><a href="#cb16-16" tabindex="-1"></a>	<span class="ex">│</span>   │   └── AAJQ01000025.1.region001.gbk</span>
<span id="cb16-17"><a href="#cb16-17" tabindex="-1"></a>	<span class="ex">│</span>   ├── Streptococcus_agalactiae_COH1/</span>
<span id="cb16-18"><a href="#cb16-18" tabindex="-1"></a>	<span class="ex">│</span>   │   ├── AAJR01000002.1.region001.gbk</span>
<span id="cb16-19"><a href="#cb16-19" tabindex="-1"></a>	<span class="ex">│</span>   │   └── AAJR01000044.1.region001.gbk</span>
<span id="cb16-20"><a href="#cb16-20" tabindex="-1"></a>	<span class="ex">│</span>   └── Streptococcus_agalactiae_H36B/</span>
<span id="cb16-21"><a href="#cb16-21" tabindex="-1"></a>	<span class="ex">│</span>   	├── AAJS01000020.1.region001.gbk</span>
<span id="cb16-22"><a href="#cb16-22" tabindex="-1"></a>	<span class="ex">│</span>   	└── AAJS01000117.1.region001.gbk</span>
<span id="cb16-23"><a href="#cb16-23" tabindex="-1"></a>	<span class="ex">├──</span> dataset_2/</span>
<span id="cb16-24"><a href="#cb16-24" tabindex="-1"></a>	<span class="ex">│</span>   ├── Streptococcus_thermophilus_LMD-9/</span>
<span id="cb16-25"><a href="#cb16-25" tabindex="-1"></a>	<span class="ex">│</span>   │   ├── CP000419.1.region001.gbk</span>
<span id="cb16-26"><a href="#cb16-26" tabindex="-1"></a>	<span class="ex">│</span>   │   ├── CP000419.1.region002.gbk</span>
<span id="cb16-27"><a href="#cb16-27" tabindex="-1"></a>	<span class="ex">│</span>   │   ├── CP000419.1.region003.gbk</span>
<span id="cb16-28"><a href="#cb16-28" tabindex="-1"></a>	<span class="ex">│</span>   │   ├── CP000419.1.region004.gbk</span>
<span id="cb16-29"><a href="#cb16-29" tabindex="-1"></a>	<span class="ex">│</span>   │   └── CP000419.1.region005.gbk</span>
<span id="cb16-30"><a href="#cb16-30" tabindex="-1"></a>	<span class="ex">│</span>   └── Streptococcus_thermophilus_LMG_18311/</span>
<span id="cb16-31"><a href="#cb16-31" tabindex="-1"></a>	<span class="ex">│</span>   	├── NC_006448.1.region001.gbk</span>
<span id="cb16-32"><a href="#cb16-32" tabindex="-1"></a>	<span class="ex">│</span>   	├── NC_006448.1.region002.gbk</span>
<span id="cb16-33"><a href="#cb16-33" tabindex="-1"></a>	<span class="ex">│</span>   	├── NC_006448.1.region003.gbk</span>
<span id="cb16-34"><a href="#cb16-34" tabindex="-1"></a>	<span class="ex">│</span>   	├── NC_006448.1.region004.gbk</span>
<span id="cb16-35"><a href="#cb16-35" tabindex="-1"></a>	<span class="ex">│</span>   	├── NC_006448.1.region005.gbk</span>
<span id="cb16-36"><a href="#cb16-36" tabindex="-1"></a>	<span class="ex">│</span>   	└── NC_006448.1.region006.gbk</span>
<span id="cb16-37"><a href="#cb16-37" tabindex="-1"></a>	<span class="ex">└──</span> taxonomy/</span>
<span id="cb16-38"><a href="#cb16-38" tabindex="-1"></a></span>
<span id="cb16-39"><a href="#cb16-39" tabindex="-1"></a><span class="ex">12</span> directories, 24 files</span></code></pre>
</div>
<div id="exercise-1.-input-folder-structure" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="exercise-1.-input-folder-structure" class="callout-inner">
<h3 class="callout-title">Exercise 1. Input-folder structure</h3>
<div class="callout-content">
<p>As we have seen, the structure of the input folder for BiG-SLiCE is
quite difficult to get. Imagine “Sekiro” wants to copy the directory
structure to use it for future BiG-SLiCE inputs. Consider the following
directory structure:</p>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  └── input_folder/               	
   	├── datasets.tsv      	
   	├── dataset_1/
   	|   ├── genome_1A/
   	|   |   ├── genome_1A.region001.gbk
   	|   |   └── genome_1A.region002.gbk
   	|   ├── genome_1B/
   	|   |   ├── genome_1B.region001.gbk
   	|   |   └── genome_1B.region002.gbk
   	├── dataset_2/
   	|   ├── genome_2A/
   	|   |   ├── genome_2A.region001.gbk
   	|   |   └── genome_2A.region002.gbk
   	|   ├── genome_2B/
   	|   |   ├── genome_2B.region001.gbk
   	|   |   └── genome_2B.region002.gbk
   	└── taxonomy/
       	├── taxonomy_dataset_1.tsv
       	└── taxonomy_dataset_2.tsv                	 </code></pre>
</div>
<p>Which would be the commands to copy the directory structure without
the files (just the folders)? <br> a)</p>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="ex">$</span> cp <span class="at">-r</span> input_folder input_folder_copy</span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a><span class="ex">$</span> cd input_folder_copy</span>
<span id="cb18-3"><a href="#cb18-3" tabindex="-1"></a><span class="ex">$</span> rm <span class="at">-r</span></span></code></pre>
</div>
<ol start="2" style="list-style-type: lower-alpha"><li>
</li></ol><div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a><span class="ex">$</span> mv <span class="at">-r</span> input_folder input_folder_copy</span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a><span class="ex">$</span> rm input_folder_copy/dataset_1/genome_1A/<span class="pp">*</span> input_folder_copy/dataset_1/genome_1B/<span class="pp">*</span></span>
<span id="cb19-3"><a href="#cb19-3" tabindex="-1"></a><span class="ex">$</span> rm input_folder_copy/dataset_2/genome_2A/<span class="pp">*</span> input_folder_copy/dataset_2/genome_2B/<span class="pp">*</span></span>
<span id="cb19-4"><a href="#cb19-4" tabindex="-1"></a><span class="ex">$</span> rm input_folder_copy/taxonomy/<span class="pp">*</span></span></code></pre>
</div>
<ol start="3" style="list-style-type: lower-alpha"><li>
</li></ol><div class="codewrapper sourceCode" id="cb20">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a><span class="ex">$</span> cp <span class="at">-r</span> input_folder input_folder_copy</span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a><span class="ex">$</span> rm input_folder_copy/dataset_1/genome_1A/<span class="pp">*</span> input_folder_copy/dataset_1/genome_1B/<span class="pp">*</span></span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a><span class="ex">$</span> rm input_folder_copy/dataset_2/genome_2A/<span class="pp">*</span> input_folder_copy/dataset_2/genome_2B/<span class="pp">*</span></span>
<span id="cb20-4"><a href="#cb20-4" tabindex="-1"></a><span class="ex">$</span> rm input_folder_copy/taxonomy/<span class="pp">*</span></span></code></pre>
</div>
<ol start="4" style="list-style-type: lower-alpha"><li>
</li></ol><div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a><span class="ex">$</span> cp <span class="at">-r</span> input_folder input_folder_copy</span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a><span class="ex">$</span> rm input_folder_copy/dataset_1/genome_1A/<span class="pp">*</span> /genome_1B/<span class="pp">*</span></span>
<span id="cb21-3"><a href="#cb21-3" tabindex="-1"></a><span class="ex">$</span> rm input_folder_copy/dataset_2/genome_2A/<span class="pp">*</span> /genome_2B/<span class="pp">*</span></span>
<span id="cb21-4"><a href="#cb21-4" tabindex="-1"></a><span class="ex">$</span> rm input_folder_copy/taxonomy/<span class="pp">*</span></span></code></pre>
</div>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p>First we need to copy the whole directory to make an
input_folder_copy. After that we want to erase the files inside the
directories dataset_1/genome_1A, dataset_1/genome_1B,
dataset_2/genome_2A, dataset2/genome_2B and taxonomy/ <br> c)</p>
<div class="codewrapper sourceCode" id="cb22">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a><span class="ex">$</span> cp <span class="at">-r</span> input_folder input_folder_copy</span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a><span class="ex">$</span> rm input_folder_copy/dataset_1/genome_1A/<span class="pp">*</span> input_folder_copy/dataset_1/genome_1B/<span class="pp">*</span></span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a><span class="ex">$</span> rm input_folder_copy/dataset_2/genome_2A/<span class="pp">*</span> input_folder_copy/dataset_2/genome_2B/<span class="pp">*</span></span>
<span id="cb22-4"><a href="#cb22-4" tabindex="-1"></a><span class="ex">$</span> rm input_folder_copy/taxonomy/<span class="pp">*</span></span></code></pre>
</div>
</div>
</div>
</div>
</div>
<p><strong>datasets.tsv file - The <a href="https://lotr.fandom.com/wiki/Silmarils" class="external-link">silmaril</a> of
BiG-SLiCE</strong></p>
<p>The next step is to create the main file
<strong>datasets.tsv</strong>. First we put the first lines to this
file. In each of these <code>.tsv</code> files, we can put lines that
begin with <code>#</code>. This will not be read by the code, and it can
be of help to know which information is located in those files. We will
use the command <code>echo</code> with the <code>-e</code> flag that
will enable us to put tab separations between the text:</p>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a><span class="ex">$</span> cd input-folder</span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a><span class="ex">$</span> echo <span class="at">-e</span> <span class="st">"#Dataset name""\t""Path to folder""\t""Path to taxonomy""\t""Description"</span> <span class="op">&gt;</span> datasets.tsv</span>
<span id="cb23-3"><a href="#cb23-3" tabindex="-1"></a><span class="ex">$</span> more datasets.tsv</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code><span><span class="co">#Dataset name   Path to folder  Path to taxonomy      Description</span></span></code></pre>
</div>
<p>We will use <code>echo</code> <em>One more time</em> to put the
information of our two datasets. But this time, we will indicate bash
that we want to insert this new line in the existing
<code>datasets.tsv</code> with the <code>&gt;&gt;</code> option.</p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a><span class="ex">$</span> echo <span class="at">-e</span> <span class="st">"dataset_1""\t""dataset_1/""\t""taxonomy/dataset_1_taxonomy.tsv""\t""Tetteling genomes"</span> <span class="op">&gt;&gt;</span> datasets.tsv</span>
<span id="cb25-2"><a href="#cb25-2" tabindex="-1"></a><span class="ex">$</span> echo <span class="at">-e</span> <span class="st">"dataset_2""\t""dataset_2/""\t""taxonomy/dataset_2_taxonomy.tsv""\t""Public genomes"</span> <span class="op">&gt;&gt;</span> datasets.tsv</span>
<span id="cb25-3"><a href="#cb25-3" tabindex="-1"></a><span class="ex">$</span> more datasets.tsv</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>#Dataset name   Path to folder  Path to taxonomy    	Description
dataset_1   	dataset_1/  	taxonomy/dataset_1_taxonomy.tsv Tetteling genomes
dataset_2   	dataset_2/  	taxonomy/dataset_2_taxonomy.tsv Public genomes</code></pre>
</div>
<p>The main file of the input-folder is finished!</p>
<div class="section level4">
<h4 id="the-taxonomy-of-the-input-folder">The taxonomy of the input-folder<a class="anchor" aria-label="anchor" href="#the-taxonomy-of-the-input-folder"></a></h4>
<p>We also need to specify the taxonomic assignation of each of the
input genomes BGC’s. We will build one <code>dataset_taxonomy.tsv</code>
file for each one of our datasets, <em>i.e.</em> two in total. Firstly,
create the header of each one of these files using
<code>echo</code>:</p>
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a><span class="ex">$</span> echo <span class="at">-e</span> <span class="st">"#Genome folder""\t"</span>Kingdom<span class="st">"\t"</span>Phylum<span class="st">"\t"</span>Class<span class="st">"\t"</span>Order<span class="st">"\t"</span>Family<span class="st">"\t"</span>Genus<span class="st">"\t""Species Organism"</span> <span class="op">&gt;</span> taxonomy/dataset_1_taxonomy.tsv</span>
<span id="cb27-2"><a href="#cb27-2" tabindex="-1"></a><span class="ex">$</span> echo <span class="at">-e</span> <span class="st">"#Genome folder""\t"</span>Kingdom<span class="st">"\t"</span>Phylum<span class="st">"\t"</span>Class<span class="st">"\t"</span>Order<span class="st">"\t"</span>Family<span class="st">"\t"</span>Genus<span class="st">"\t""Species Organism"</span> <span class="op">&gt;</span> taxonomy/dataset_2_taxonomy.tsv</span>
<span id="cb27-3"><a href="#cb27-3" tabindex="-1"></a></span>
<span id="cb27-4"><a href="#cb27-4" tabindex="-1"></a><span class="ex">$</span> more taxonomy/dataset_1_taxonomy.tsv</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code><span><span class="co">#Genome folder  Kingdom Phylum  Class   Order   Family  Genus   Species Organism</span></span></code></pre>
</div>
<p>There are several ways to obtain the taxonomic data. One of them is
searching for each lineage taxonomic identification on <a href="https://www.ncbi.nlm.nih.gov/guide/taxonomy/" class="external-link">NCBI</a>. There is
also a way to get this information from our <code>.gbk</code> files:</p>
<div class="codewrapper sourceCode" id="cb29">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb29-1"><a href="#cb29-1" tabindex="-1"></a><span class="ex">$</span> head <span class="at">-n20</span> ../../annotated/Streptococcus_agalactiae_18RS21.gbk</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>LOCUS   	AAJO01000169.1      	2501 bp	DNA 	linear   UNK
DEFINITION  Streptococcus agalactiae 18RS21
ACCESSION   AAJO01000169.1
KEYWORDS	.
SOURCE  	Streptococcus agalactiae 18RS21.
  ORGANISM  Streptococcus agalactiae 18RS21
        	Bacteria; Terrabacteria group; Firmicutes; Bacilli;
        	Lactobacillales; Streptococcaceae; Streptococcus; Streptococcus
        	agalactiae.
FEATURES         	Location/Qualifiers
 	source      	1..2501
                 	/mol_type="genomic DNA"
                 	/db_xref="taxon:342613"
                 	/genome_md5=""
                 	/project="nselem35_342613"
                 	/genome_id="342613.37"
                 	/organism="Streptococcus agalactiae 18RS21"
 	CDS         	1..213
                 	/db_xref="SEED:fig|342613.37.peg.1973"
                 	/db_xref="GO:0003735"</code></pre>
</div>
<p>We will use again <code>echo</code> to write this information into
our <code>dataset_taxonomy.tsv</code>. This is going to be a little long
because we have 8 genomes and we will need to write 8 <code>echo</code>
lines, but bear with us. We are almost there!</p>
<div class="codewrapper sourceCode" id="cb31">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a><span class="ex">$</span> echo <span class="at">-e</span> <span class="st">"Streptococcus_agalactiae_18RS21/""\t"</span>Bacteria<span class="st">"\t"</span>Firmicutes<span class="st">"\t"</span>Bacilli<span class="st">"\t"</span>Lactobacillales<span class="st">"\t"</span>Streptococcaceae<span class="st">"\t"</span>Streptococcus<span class="st">"\t""Streptococcus agalactiae 18RS21"</span> <span class="op">&gt;&gt;</span> taxonomy/dataset_1_taxonomy.tsv</span>
<span id="cb31-2"><a href="#cb31-2" tabindex="-1"></a></span>
<span id="cb31-3"><a href="#cb31-3" tabindex="-1"></a><span class="ex">$</span> echo <span class="at">-e</span> <span class="st">"Streptococcus_agalactiae_515/""\t"</span>Bacteria<span class="st">"\t"</span>Firmicutes<span class="st">"\t"</span>Bacilli<span class="st">"\t"</span>Lactobacillales<span class="st">"\t"</span>Streptococcaceae<span class="st">"\t"</span>Streptococcus<span class="st">"\t""Streptococcus agalactiae 515"</span> <span class="op">&gt;&gt;</span> taxonomy/dataset_1_taxonomy.tsv</span>
<span id="cb31-4"><a href="#cb31-4" tabindex="-1"></a></span>
<span id="cb31-5"><a href="#cb31-5" tabindex="-1"></a><span class="ex">$</span> echo <span class="at">-e</span> <span class="st">"Streptococcus_agalactiae_A909/""\t"</span>Bacteria<span class="st">"\t"</span>Firmicutes<span class="st">"\t"</span>Bacilli<span class="st">"\t"</span>Lactobacillales<span class="st">"\t"</span>Streptococcaceae<span class="st">"\t"</span>Streptococcus<span class="st">"\t""Streptococcus agalactiae A909"</span> <span class="op">&gt;&gt;</span> taxonomy/dataset_1_taxonomy.tsv</span>
<span id="cb31-6"><a href="#cb31-6" tabindex="-1"></a></span>
<span id="cb31-7"><a href="#cb31-7" tabindex="-1"></a><span class="ex">$</span> echo <span class="at">-e</span> <span class="st">"Streptococcus_agalactiae_CJB111/""\t"</span>Bacteria<span class="st">"\t"</span>Firmicutes<span class="st">"\t"</span>Bacilli<span class="st">"\t"</span>Lactobacillales<span class="st">"\t"</span>Streptococcaceae<span class="st">"\t"</span>Streptococcus<span class="st">"\t""Streptococcus agalactiae CJB111"</span> <span class="op">&gt;&gt;</span> taxonomy/dataset_1_taxonomy.tsv</span>
<span id="cb31-8"><a href="#cb31-8" tabindex="-1"></a></span>
<span id="cb31-9"><a href="#cb31-9" tabindex="-1"></a><span class="ex">$</span> echo <span class="at">-e</span> <span class="st">"Streptococcus_agalactiae_COH1/""\t"</span>Bacteria<span class="st">"\t"</span>Firmicutes<span class="st">"\t"</span>Bacilli<span class="st">"\t"</span>Lactobacillales<span class="st">"\t"</span>Streptococcaceae<span class="st">"\t"</span>Streptococcus<span class="st">"\t""Streptococcus agalactiae COH1"</span> <span class="op">&gt;&gt;</span> taxonomy/dataset_1_taxonomy.tsv</span>
<span id="cb31-10"><a href="#cb31-10" tabindex="-1"></a></span>
<span id="cb31-11"><a href="#cb31-11" tabindex="-1"></a><span class="ex">$</span> echo <span class="at">-e</span> <span class="st">"Streptococcus_agalactiae_H36B/""\t"</span>Bacteria<span class="st">"\t"</span>Firmicutes<span class="st">"\t"</span>Bacilli<span class="st">"\t"</span>Lactobacillales<span class="st">"\t"</span>Streptococcaceae<span class="st">"\t"</span>Streptococcus<span class="st">"\t""Streptococcus agalactiae H36B"</span> <span class="op">&gt;&gt;</span> taxonomy/dataset_1_taxonomy.tsv</span>
<span id="cb31-12"><a href="#cb31-12" tabindex="-1"></a></span>
<span id="cb31-13"><a href="#cb31-13" tabindex="-1"></a><span class="ex">$</span> more taxonomy/dataset_1_taxonomy.tsv</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>#Genome folder   Kingdom Phylum  Class   Order   Family  Genus   Species Organism
Streptococcus_agalactiae_18RS21/    	Bacteria    	Firmicutes  	Bacilli Lactobacillales Strept
ococcaceae  	Streptococcus   Streptococcus agalactiae 18RS21
Streptococcus_agalactiae_515/   Bacteria    	Firmicutes  	Bacilli Lactobacillales Streptococcace
ae  	Streptococcus   Streptococcus agalactiae 515
Streptococcus_agalactiae_A909/  Bacteria    	Firmicutes  	Bacilli Lactobacillales Streptococcace
ae  	Streptococcus   Streptococcus agalactiae A909
Streptococcus_agalactiae_CJB111/    	Bacteria    	Firmicutes  	Bacilli Lactobacillales Strept
ococcaceae  	Streptococcus   Streptococcus agalactiae CJB111
Streptococcus_agalactiae_COH1/  Bacteria    	Firmicutes  	Bacilli Lactobacillales Streptococcace
ae  	Streptococcus   Streptococcus agalactiae COH1
Streptococcus_agalactiae_H36B/  Bacteria    	Firmicutes  	Bacilli Lactobacillales Streptococcace
ae  	Streptococcus   Streptococcus agalactiae H36B</code></pre>
</div>
<p>We will do the same for the <code>dataset_2_taxonomy.tsv</code>
file:</p>
<div class="codewrapper sourceCode" id="cb33">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a><span class="ex">$</span> echo <span class="at">-e</span> <span class="st">"Streptococcus_thermophilus_LMD-9/""\t"</span>Bacteria<span class="st">"\t"</span>Firmicutes<span class="st">"\t"</span>Bacilli<span class="st">"\t"</span>Lactobacillales<span class="st">"\t"</span>Streptococcaceae<span class="st">"\t"</span>Streptococcus<span class="st">"\t""Streptococcus thermophilus LMD-9"</span> <span class="op">&gt;&gt;</span> taxonomy/dataset_2_taxonomy.tsv</span>
<span id="cb33-2"><a href="#cb33-2" tabindex="-1"></a></span>
<span id="cb33-3"><a href="#cb33-3" tabindex="-1"></a><span class="ex">$</span> echo <span class="at">-e</span> <span class="st">"Streptococcus_thermophilus_LMG_18311/""\t"</span>Bacteria<span class="st">"\t"</span>Firmicutes<span class="st">"\t"</span>Bacilli<span class="st">"\t"</span>Lactobacillales<span class="st">"\t"</span>Streptococcaceae<span class="st">"\t"</span>Streptococcus<span class="st">"\t""Streptococcus thermophilus LMG 18311"</span> <span class="op">&gt;&gt;</span> taxonomy/dataset_2_taxonomy.tsv</span>
<span id="cb33-4"><a href="#cb33-4" tabindex="-1"></a></span>
<span id="cb33-5"><a href="#cb33-5" tabindex="-1"></a><span class="ex">$</span> more taxonomy/dataset_2_taxonomy.tsv</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>#Genome folder   Kingdom Phylum  Class   Order   Family  Genus   Species Organism
Streptococcus_thermophilus_LMD-9/   	Bacteria    	Firmicutes  	Bacilli Lactobacillales Strept
ococcaceae  	Streptococcus   Streptococcus thermophilus LMD-9
Streptococcus_thermophilus_LMG_18311/   Bacteria    	Firmicutes  	Bacilli Lactobacillales Strept
ococcaceae  	Streptococcus   Streptococcus thermophilus LMG 18311</code></pre>
</div>
<p>Now, we have the organization of our <code>input-folder</code>
complete</p>
<div class="codewrapper sourceCode" id="cb35">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb35-1"><a href="#cb35-1" tabindex="-1"></a><span class="ex">$</span> cd ..</span>
<span id="cb35-2"><a href="#cb35-2" tabindex="-1"></a><span class="ex">$</span> tree <span class="at">-F</span> input-folder/</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>input-folder/
├── dataset_1/
│   ├── Streptococcus_agalactiae_18RS21/
│   │   ├── AAJO01000016.1.region001.gbk
│   │   ├── AAJO01000043.1.region001.gbk
│   │   └── AAJO01000226.1.region001.gbk
│   ├── Streptococcus_agalactiae_515/
│   │   ├── AAJP01000027.1.region001.gbk
│   │   └── AAJP01000037.1.region001.gbk
│   ├── Streptococcus_agalactiae_A909/
│   │   ├── CP000114.1.region001.gbk
│   │   └── CP000114.1.region002.gbk
│   ├── Streptococcus_agalactiae_CJB111/
│   │   ├── AAJQ01000010.1.region001.gbk
│   │   └── AAJQ01000025.1.region001.gbk
│   ├── Streptococcus_agalactiae_COH1/
│   │   ├── AAJR01000002.1.region001.gbk
│   │   └── AAJR01000044.1.region001.gbk
│   └── Streptococcus_agalactiae_H36B/
│   	├── AAJS01000020.1.region001.gbk
│   	└── AAJS01000117.1.region001.gbk
├── dataset_2/
│   ├── Streptococcus_thermophilus_LMD-9/
│   │   ├── CP000419.1.region001.gbk
│   │   ├── CP000419.1.region002.gbk
│   │   ├── CP000419.1.region003.gbk
│   │   ├── CP000419.1.region004.gbk
│   │   └── CP000419.1.region005.gbk
│   └── Streptococcus_thermophilus_LMG_18311/
│   	├── NC_006448.1.region001.gbk
│   	├── NC_006448.1.region002.gbk
│   	├── NC_006448.1.region003.gbk
│   	├── NC_006448.1.region004.gbk
│   	├── NC_006448.1.region005.gbk
│   	└── NC_006448.1.region006.gbk
├── datasets.tsv
└── taxonomy/
	├── dataset_1_taxonomy.tsv
	└── dataset_2_taxonomy.tsv

11 directories, 27 files</code></pre>
</div>
<div id="exercise-2.-taxonomic-information" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="exercise-2.-taxonomic-information" class="callout-inner">
<h3 class="callout-title">Exercise 2. Taxonomic information</h3>
<div class="callout-content">
<p>If you use the head -n20 on one of your .gbk files, you will obtain
an output like this</p>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>LOCUS   	AAJO01000169.1      	2501 bp	DNA 	linear   UNK
DEFINITION  Streptococcus agalactiae 18RS21
ACCESSION   AAJO01000169.1
KEYWORDS	.
SOURCE  	Streptococcus agalactiae 18RS21.
 ORGANISM  Streptococcus agalactiae 18RS21
       	Bacteria; Terrabacteria group; Firmicutes; Bacilli;
       	Lactobacillales; Streptococcaceae; Streptococcus; Streptococcus
       	agalactiae.
FEATURES         	Location/Qualifiers
	source      	1..2501
                	/mol_type="genomic DNA"
                	/db_xref="taxon:342613"
                	/genome_md5=""
                	/project="nselem35_342613"
                	/genome_id="342613.37"
                	/organism="Streptococcus agalactiae 18RS21"
	CDS         	1..213
                	/db_xref="SEED:fig|342613.37.peg.1973"
                	/db_xref="GO:0003735"</code></pre>
</div>
<p>As you can see, in the gbk file we can find the taxonomic information
that we need to fill the taxonomic information of each genome. Fill in
the blank spaces to complete the command that would get the organism
information from the .gbk file.</p>
<p style="text-align: center;">
$ grep ___________ -m1 ____________ Streptococcus_agalactiae_18RS21.gbk
</p>
<p style="text-align: center;">
-B 3       -A 3       -a       ‘Streptococcus’       ‘ORGANISM’      
‘gbk’
</p>
<p>To obtain the next result:</p>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code> ORGANISM  Streptococcus agalactiae 18RS21
       	Bacteria; Terrabacteria group; Firmicutes; Bacilli;
       	Lactobacillales; Streptococcaceae; Streptococcus; Streptococcus
       	agalactiae.</code></pre>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<p>You can use the <code>grep --help</code> command to get information
about the available options for the <code>grep</code> command</p>
</div>
</div>
</div>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<p>We need to use the grep command to search for the word “ORGANISM” in
the file. Once located we just want to get the first instance of the
word that is why we use the -m1 flag. Then we want the 3 next lines of
the file after the instance, because that’s where the information is. So
the final command would be:</p>
<div class="codewrapper sourceCode" id="cb39">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb39-1"><a href="#cb39-1" tabindex="-1"></a><span class="ex">$</span> grep <span class="st">'ORGANISM'</span> <span class="at">-m1</span> <span class="at">-A3</span> Streptococcus_agalactiae_18RS21.gbk</span></code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="running-big-slice">Running BiG-SLiCE<a class="anchor" aria-label="anchor" href="#running-big-slice"></a></h3>
<p>We are ready to run <code>BiG-SLICE</code>. After we type the next
line of command, it will take close to 3 minutes long to end the
process:</p>
<div class="codewrapper sourceCode" id="cb40">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb40-1"><a href="#cb40-1" tabindex="-1"></a><span class="ex">$</span> bigslice <span class="at">-i</span> input-folder output-bigslice</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>...
...
BiG-SLiCE run complete!</code></pre>
</div>
<p>In order to see the results, we need to download the
<code>output-bigslice</code> to our local computer. We will use the
<code>scp</code> command to accomplish this.</p>
<p>Open another terminal and <strong>without connecting it to the remote
computer</strong> (<em>i.e.</em> in your own computer), move to a
directory where you know you can save the folder. In this example we
will copy it to the <code>Documents</code> folder of our computer.</p>
<div class="codewrapper sourceCode" id="cb42">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb42-1"><a href="#cb42-1" tabindex="-1"></a><span class="ex">$</span> cd Documents/</span>
<span id="cb42-2"><a href="#cb42-2" tabindex="-1"></a><span class="ex">$</span> scp <span class="at">-r</span> virtual-computerbetterlab@###.###.###.##/home/virtual-User/dc_workshop/results/bigslice/output-bigslice .</span>
<span id="cb42-3"><a href="#cb42-3" tabindex="-1"></a><span class="ex">$</span> ls <span class="at">-F</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>output-bigslice/</code></pre>
</div>
<p>The <code>BiG-SLiCE</code> prepared the code so that in each output
folder we will obtain the output files and some scripts that will
generate a way to visualize the data.</p>
<div class="codewrapper sourceCode" id="cb44">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb44-1"><a href="#cb44-1" tabindex="-1"></a><span class="ex">$</span> ls output-bigslice/</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>app  LICENSE.txt  requirements.txt  result  start_server.sh</code></pre>
</div>
<p>The file <code>requirements.txt</code> contains the names of two
softwares that need to be installed to use the visualization:
<code>SQLite3</code> and <code>flask</code>. We will install these with
the next command:</p>
<div class="codewrapper sourceCode" id="cb46">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb46-1"><a href="#cb46-1" tabindex="-1"></a><span class="ex">$</span> pip install <span class="at">-r</span> output-bigslice/requirements.txt</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>...
...
Successfully installed pysqlite3</code></pre>
</div>
<p>After the installation, we can use the <code>start_server.sh</code>
to generate a</p>
<div class="codewrapper sourceCode" id="cb48">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb48-1"><a href="#cb48-1" tabindex="-1"></a><span class="ex">$</span> bash output-bigslice/start_server.sh</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code> * Serving Flask app "/mnt/d/documentos/sideprojects/sc-mineria/bigSlice/output-bigslice/app/run.py"
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on http://0.0.0.0:5000/ (Press CTRL+C to quit)</code></pre>
</div>
<p>The result looks a little intimidating, but if you obtained a result
that looks like the above one do not worry. Next, we need to open an
internet browser. In a new tab type the next line.</p>
<pre class="url"><code>http://localhost:5000/</code></pre>
<p>As a result, we will obtain a web-page that looks like this:</p>
<p><a href="fig/02-04-05.png">
<img src="fig/02-04-05.png" alt="BiG SLiCE web page output displaying the results obtained from the example run.
                                  	A left panel presents the information generated, composed of seven tabs;
                                  	Summary, Datasets, Runs, Reports, Help, Feedback, and About.
                                  	The rest is a Summary of all datasets and runs included in the output.
                                  	Appearing firstly, the Datasets provided as input, organized as a table with five rows;
                                  	name, #genomes, total, with taxonomy, and description.
                                  	Next, the information about the Runs, also organized as a table with the following rows;
                                  	name, started, finished, status, threshold, #BGCs, and #GCFs.." class="figure"></a></p>
<p>Here we have a summary of the obtained results. In the left part, we
see a set of 7 fields. 3 of them have the information that we generated:
<code>Summary</code>, <code>Datasets</code>, and <code>Runs</code>. The
other ones are for information regarding the software, one of them is to
give feedback to the developers. You can acces to the
<code>Datasets</code> and <code>Runs</code> page also from the options
that we see in the main part of this summary page. The first part of
this summary describes the datasets that we provided as input. The
second one is the information of the process (when we ran the program)
that <code>BiG-SLiCE</code> carry out with our data.</p>
<p>If we click into the <code>Runs</code> field on the left or the
<code>run-0001</code> blue button on the <strong>Runs</strong> main
section, we will jump into the page that gives us the information
regarding the BGCs GCFs that were obtained:</p>
<p><a href="fig/02-04-06.png">
<img src="fig/02-04-06.png" alt="BiG SLiCE web page output displaying the information obtained from the Run-0001.
                                  	Firstly, it is shown an Overview from the Run. Indicating the Status, when was it Started and Finished,
                                  	as well as the Parameters, the HMM models and the Input.
                                  	Next, the Statistics about the run are shown as two bar-plots.
                                  	The left one plots the average distance to GCF models, whilst the right one shows the amount of BGCs in GCFs." class="figure"></a></p>
<p>In this new page we have two bar-plots that give us information
regarding if the BGCs that we sumministrated were classified as
fragmented or complete by <code>AntiSMASH</code> (according to the
<code>AntiSMASH</code> algorithm, a BGC can be classified as fragmented
if it is found at the edge of a contig), and how many families were
obtained with singletons (composed of just one BGC) and of 2-3 BGCs.</p>
<p>If we go to the bottom of this page, we will found a table of all the
GCFs with information of the number of BGCs that compose them, the
representative BGC class inside each family, and which genus has more
BGCs that belong to each family.</p>
<p>Click the <code>+</code> symbol of the GCF_7, and then on the
<code>View</code> button that is displayed. This will take us to a page
with statistics of the BGCs that are part of this family.</p>
<p>Inside this section, we can found a great amount of information
regarding the GCF_7 BGC’s. If we jump to section called
<code>Members</code>, we can click on the <code>view BGC arrowers</code>
button to get a visualization of the domains that are part of each of
the genes of these BGCs.</p>
<p><a href="fig/02-04-07.png">
<img src="fig/02-04-07.png" alt="BiG SLiCE web page output displaying detailed information regarding the BGCs from GCF_7.
                                  	The Arrowers show a gene arrow visualization of the domains that are part of each of the genes of the BGCs belonging to GCF_7." class="figure"></a></p>
<p>As we can see, this tool can be used to clusterize homologous BGCs
that share a structure. With this knowledge, we can try some hypothesis
or even build new ones in the light of this information.</p>
<div id="discussion" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<span class="callout-header">Discussion</span>
<div id="discussion" class="callout-inner">
<h3 class="callout-title">Discussion</h3>
<div class="callout-content">
<p>Considering the results in the runs tab, we can see that there are a
lot of Gene Cluster Families that have only one cluster associated to it
(GCF_1, GCF_2, GCF_4, GCF_5, GCF_6, GCF_8, GCF_9, GCF_13, and GCF_15).
<br> What does this tell us about the metabolites diversity of the
Streptococcus sample?</p>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<p>We can see that there are some unique gene cluster families, which
means that there are an organism that has that unique gene cluster and
therefore it has some metabolite that the others don´t and that may be
advantageous in its environment. Since there are many singleton Gene
Cluster Families, we can say that there is a big diversity from the
sample because many of the organisms in the sample have unique
“abilities” (metabolites).</p>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
<div id="discussion-1" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<span class="callout-header">Discussion</span>
<div id="discussion-1" class="callout-inner">
<h3 class="callout-title">Discussion</h3>
<div class="callout-content">
<p>Suppose we get the following results from a sample of 6
organisms.</p>
<p>|——————-+——————————————————–| | GCF | #core members |
|:—————–:+:——————————————————:| | GCF_1 | 6 |
|——————-+——————————————————–| | GCF_2 | 5 |
|——————-+——————————————————–| | GCF_3 | 5 |
|——————-+——————————————————–| | GCF_4 | 6 |
|——————-+——————————————————–|</p>
<p>What can we infer about the metabolite diversity?</p>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4"> Show me the solution </h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" data-bs-parent="#accordionSolution4" aria-labelledby="headingSolution4">
<div class="accordion-body">
<p>As most of the organisms have a gene cluster in the gene cluster
family, we can infer that most of them have the same secondary
metabolites, which means that there is little metabolite diversity
between them.</p>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="big-fam">BiG-FAM<a class="anchor" aria-label="anchor" href="#big-fam"></a></h2>
<hr class="half-width"><p>To demonstrate the functionality of <code>BiG-SLICE</code>, the
authors gathered close to 1.2 million of BGCs. All this information,
they put it on a great database that they called <code>BiG-FAM</code>.
When you get to the main <a href="https://bigfam.bioinformatics.nl/home" class="external-link">page</a>, you will face a
structure that looks like the one we saw on the <code>BiG-SLICE</code>
results:</p>
<p><a href="fig/02-04-08.png">
<img src="fig/02-04-08.png" alt="BiG-FAM main page showing an introduction as well as a graphical representation of the database.
                                  	A left panel displays the available options; Home, GCFs, BGCs, Query, Statistics, Help, Feedback, and About. " class="figure"></a></p>
<p>One of the options on the left is <code>Query</code>. If we click
here, we will move to a new page where we can insert an
<code>antiSMASH job ID</code>. This will run a comparison of all the
BGCs found on that antiSMASH run on a single genome, against the immense
database.</p>
<p>We will use the <code>antiSMASH job ID</code> from the analysis made
on <em>Streptococcus agalactiae</em> A909. The
<code>antiSMASH job ID</code> can be found in the <code>URL</code>
direction of this page in the next format:
taxon-aaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee.</p>
<p>We will use the job ID that we obtained:</p>
<p>bacteria-f931deb5-509b-4f28-a210-212456c5139b</p>
<p>And click the <code>submit</code> option.</p>
<p><a href="fig/02-04-09.png">
<img src="fig/02-04-09.png" alt="BiG-FAM query page with the option for inserting an antiSMASH job and submitting it.
                                  	Below it is described how much of the job is Pending, Processing, and Processed." class="figure"></a></p>
<p>This will generate a result page that indicates with which BGCs from
the database, the BGCs from the query genome are related.</p>
<p><a href="fig/02-04-10.png">
<img src="fig/02-04-10.png" alt="BiG-FAM result page indicating firstly an overview of the job; the query which was submitted,
                                  	its status, as well as the time at which it was started and finished.
                                  	Next, a table indicating the BGCs from the database which are related with the query BGCs.
                                  	This is organized as a table with seven rows; query BGC, class, length (kb), completeness, best hit, distance and details" class="figure"></a></p>
<div id="keypoints2" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<ul><li>
<code>BiG-SLiCE</code> and <code>BiG-FAM</code> are softwares that
are useful to compare the metabolic diversity of bacterial lineages
between each other and against a big database</li>
<li>An input-folder containing the BGCs from antiSMASH and the taxonomic
information of each genome is needed to run <code>BiG-SLiCE</code>
</li>
<li>The results from the antiSMASH web-tool are needed to run
<code>BiG-FAM</code>
</li>
<li>Gene Cluster Families can help us to compare the metabolic
capabilities of a set of bacterial lineages</li>
<li>We can use <code>BiG-FAM</code> to compare a BGC against the whole
database and predict its Gene Cluster Family</li>
</ul></div>
</div>
</div>
</section></div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="04-bigscape.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="06-corason.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="04-bigscape.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: BGC Similarity
        </a>
        <a class="chapter-link float-end" href="06-corason.html" rel="next">
          Next: Finding Variation on...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://github.com/carpentries-incubator/genome-mining/edit/main/episodes/05-bigslice-bigfam.md" class="external-link">Edit on GitHub</a>

	
        | <a href="https://github.com/carpentries-incubator/genome-mining/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/carpentries-incubator/genome-mining/" class="external-link">Source</a></p>
        <p>
        <a href="https://github.com/carpentries-incubator/genome-mining/blob/main/" class="external-link">Cite</a>
        | <a href="mailto:axel.ramos3737@gmail.com">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
		</div>
		<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.18.5" class="external-link">sandpaper (0.18.5)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.9" class="external-link">pegboard (0.7.9)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.9" class="external-link">varnish (1.0.9)</a></p>
		</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "LearningResource",
  "@id": "https://carpentries-incubator.github.io/genome-mining/05-bigslice-bigfam.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/LearningResource/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "software, data, lesson, The Carpentries",
  "name": "Homologous BGC Clusterization",
  "creativeWorkStatus": "active",
  "url": "https://carpentries-incubator.github.io/genome-mining/05-bigslice-bigfam.html",
  "identifier": "https://carpentries-incubator.github.io/genome-mining/05-bigslice-bigfam.html",
  "dateCreated": "2021-06-20",
  "dateModified": "2026-02-20",
  "datePublished": "2026-02-20"
}

  </script><script>
		feather.replace();
	</script><!-- Matomo --><script>
          var _paq = window._paq = window._paq || [];
          /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
          _paq.push(["setDocumentTitle", document.domain + "/" + document.title]);
          _paq.push(["setDomains", ["*.lessons.carpentries.org","*.datacarpentry.github.io","*.datacarpentry.org","*.librarycarpentry.github.io","*.librarycarpentry.org","*.swcarpentry.github.io", "*.carpentries.github.io"]]);
          _paq.push(["setDoNotTrack", true]);
          _paq.push(["disableCookies"]);
          _paq.push(["trackPageView"]);
          _paq.push(["enableLinkTracking"]);
          (function() {
              var u="https://matomo.carpentries.org/";
              _paq.push(["setTrackerUrl", u+"matomo.php"]);
              _paq.push(["setSiteId", "1"]);
              var d=document, g=d.createElement("script"), s=d.getElementsByTagName("script")[0];
              g.async=true; g.src="https://matomo.carpentries.org/matomo.js"; s.parentNode.insertBefore(g,s);
          })();
        </script><!-- End Matomo Code --></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

