<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Genome Mining in Prokaryotes: Metabolomics workshop</title><meta name="viewport" content="width=device-width, initial-scale=1"><script src="assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="assets/styles.css"><script src="assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="favicons/incubator/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="favicons/incubator/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="favicons/incubator/favicon-16x16.png"><link rel="manifest" href="favicons/incubator/site.webmanifest"><link rel="mask-icon" href="favicons/incubator/safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" media="(prefers-color-scheme: light)" content="white"><meta name="theme-color" media="(prefers-color-scheme: dark)" content="black"></head><body>
    <header id="top" class="navbar navbar-expand-md top-nav incubator"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="incubator-logo" alt="Lesson Description" src="assets/images/incubator-logo.svg"><span class="badge text-bg-danger">
          <abbr title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.">
            <a href="https://docs.carpentries.org/resources/curriculum/lesson-life-cycle.html" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
              Pre-Alpha
            </a>
            <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text"><li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul></li>
      </div>


      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Learner View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='instructor/09-metabolomics.html';">Instructor View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Lesson Description" src="assets/images/incubator-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Genome Mining in Prokaryotes
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Genome Mining in Prokaryotes
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="reference.html#glossary">Glossary</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="profiles.html">Learner Profiles</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><li><a class="dropdown-item" href="discuss.html">Discussion</a></li><li><a class="dropdown-item" href="reference.html">Glossary</a></li>
          </ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Genome Mining in Prokaryotes
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 91%" class="percentage">
    91%
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: 91%" aria-valuenow="91" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text"><li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul></li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Learner View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="instructor/09-metabolomics.html">Instructor View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->

            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Setup</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="01-introduction.html">1. Introduction to Genome Mining</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="02-antismash.html">2. Secondary metabolite biosynthetic gene cluster identification</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="03-antiSMASHdb-MiBiG.html">3. Genome Mining Databases</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="04-bigscape.html">4. BGC Similarity Networks</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="05-bigslice-bigfam.html">5. Homologous BGC Clusterization</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="06-corason.html">6. Finding Variation on Genomic Vicinities</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="07-evomining.html">7. Evolutionary Genome Mining</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="08-GatorGC.html">8. GATOR-GC: Genomic Assessment Tool for Orthologous Regions and Gene Clusters</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
      <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapsecurrent" aria-expanded="true" aria-controls="flush-collapsecurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        9. Metabolomics workshop
        </span>
      </button>
    </div><!--/div.accordion-header-->

    <div id="flush-collapsecurrent" class="accordion-collapse collapse show" aria-labelledby="flush-headingcurrent" data-bs-parent="#accordionFlushcurrent">
      <div class="accordion-body">
        <ul><li><a href="#introduction">Introduction</a></li>
<li><a href="#creating-a-gnps-account">Creating a GNPS account</a></li>
<li><a href="#download-mzmine-v3.9">Download MZMine v3.9</a></li>
<li><a href="#download-cytoscape">Download Cytoscape</a></li>
<li><a href="#download-the-metabolomics-dataset">Download the metabolomics dataset</a></li>
<li><a href="#import-the-dataset-and-use-the-batch-file-to-analyze-your-data">Import the dataset, and use the batch file to analyze your data</a></li>
<li><a href="#analysis-using-mzmine">Analysis using MZMine</a></li>
<li><a href="#explore-the-structure-of-our-data">Explore the structure of our data</a></li>
<li><a href="#analyze-the-final-output-from-the-analysis">Analyze the final output from the analysis</a></li>
<li><a href="#remove-media-blanks-ms-spectra">Remove media blanks MS spectra</a></li>
<li><a href="#export-feature-lists-in-gnps-format">Export Feature lists in GNPS format</a></li>
<li><a href="#create-a-molecular-network">Create a molecular network</a></li>
<li><a href="#statistics-analysis-using-fbmn-stats-guide-web-server">Statistics analysis using FBMN STATS guide web server</a></li>
<li><a href="#visualize-the-network-using-cytoscape">Visualize the network using Cytoscape</a></li>
<li><a href="#references">References</a></li>
        </ul></div><!--/div.accordion-body-->
    </div><!--/div.accordion-collapse-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush11">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading11">
        <a href="10-OtherResources.html">10. Other Resources</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="reference.html#glossary">Glossary</a>
                      </li>
                      <li>
                        <a href="profiles.html">Learner Profiles</a>
                      </li>
                      <li><a href="discuss.html">Discussion</a></li><li><a href="reference.html">Glossary</a></li>
                    </ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources"><a href="aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">

            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="08-GatorGC.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="10-OtherResources.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="08-GatorGC.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: GATOR-GC: Genomic
        </a>
        <a class="chapter-link float-end" href="10-OtherResources.html" rel="next">
          Next: Other Resources...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>Metabolomics workshop</h1>
        <p>Last updated on 2026-02-20 |

        <a href="https://github.com/carpentries-incubator/genome-mining/edit/main/episodes/09-metabolomics.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>



        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>How can I evaluate the similarity between MS spectra?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>Understand how GNPS molecular networks work.</li>
<li>Use raw mzML metabolomic data for annotation.</li>
<li>Visualize and explore molecular networks</li>
</ul></div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a></h2>
<hr class="half-width"><p>In the previous sections, we explored the biosynthetic potential of
various strains by analyzing the BGCs within their genomes. We utilized
BiG-SCAPE to create BGC networks, which compare all the BGCs detected by
antiSMASH to determine their relatedness. Similarly, using metabolomics
we can investigate the metabolites produced by selected strains in the
laboratory. While crude extracts can be made to analyze the metabolites
of specific strains, it is also possible to study the metabolites
present in complex samples, such as soil, marine environments, host
organisms, or stool samples. There are two approaches to annotating
metabolomics data. The first comprises using the information obtained by
LC-MS, comparing the exact mass and UV spectra of each detected
metabolite through natural product databases. The introduction of the
global natural products social molecular networking <a href="https://gnps.ucsd.edu/ProteoSAFe/static/gnps-splash.jsp" class="external-link">GNPS</a>
platform for molecular networking (M. Wang et al., 2016), significantly
influenced dereplication techniques. Molecular networking groups
metabolites into molecular families (MFs), thereby improving the
annotation process of unknown metabolites.</p>
<p><a href="fig/Molecular-network.png">
<img src="fig/Molecular-network.png" alt="GNPS output can be directly visualized in the GNPS webpage, or using other visualization tools such as [Cytoscape](https://cytoscape.org/) " width="900" class="figure"></a></p>
</section><section><h2 class="section-heading" id="creating-a-gnps-account">Creating a GNPS account<a class="anchor" aria-label="anchor" href="#creating-a-gnps-account"></a></h2>
<hr class="half-width"><p>Before starting this tutorial, it will be useful to have a GNPS
account. For that go to the <a href="https://gnps.ucsd.edu/ProteoSAFe/static/gnps-splash.jsp" class="external-link">GNPS</a>
webpage and select Create a new account.</p>
<p><a href="fig/create_GNPSaccount.jpg">
<img src="fig/create_GNPSaccount.jpg" alt="Create an account in GNPS" width="800" class="figure"></a></p>
<p>Then fill in the following information:</p>
<ul><li>Username</li>
<li>Name</li>
<li>Organization</li>
<li>Email</li>
<li>Password</li>
</ul><p>Wait for a confirmation email, and you now have a GNPS account.</p>
</section><section><h2 class="section-heading" id="download-mzmine-v3-9">Download MZMine v3.9<a class="anchor" aria-label="anchor" href="#download-mzmine-v3-9"></a></h2>
<hr class="half-width"><ol style="list-style-type: decimal"><li><p>First, go to the MZMine 3.9 release <a href="https://github.com/mzmine/mzmine/releases/tag/v3.9.0" class="external-link">MZMine
3.9</a></p></li>
<li><p>Select the installable file depending on your computer</p></li>
</ol><p><a href="fig/Download_MZMine.jpg">
<img src="fig/Download_MZMine.jpg" alt="MZmine 3, an MS data analysis platform " width="500" class="figure"></a></p>
<ol start="3" style="list-style-type: decimal"><li>Double-click on the file, and install the software</li>
</ol></section><section><h2 class="section-heading" id="download-cytoscape">Download Cytoscape<a class="anchor" aria-label="anchor" href="#download-cytoscape"></a></h2>
<hr class="half-width"><p>Cytoscape is a software frequently used to visualize networks, such
as BGC networks or molecular networks.</p>
<ol style="list-style-type: decimal"><li><p>Go to <a href="https://cytoscape.org/download.html" class="external-link">Cytoscape
webpage</a></p></li>
<li><p>Click on download Cytoscape for your operating system</p></li>
<li><p>Install Cytoscape in your computer.</p></li>
</ol></section><section><h2 class="section-heading" id="download-the-metabolomics-dataset">Download the metabolomics dataset<a class="anchor" aria-label="anchor" href="#download-the-metabolomics-dataset"></a></h2>
<hr class="half-width"><p>First, we need to go to Zenodo and download all the mzML raw data
collected from the described strains. <a href="https://zenodo.org/api/records/13352458/files-archive" class="external-link uri">https://zenodo.org/api/records/13352458/files-archive</a></p>
<p>After downloading the compressed file, we need to decompress it and
store the files in a folder on our computer.</p>
</section><section><h2 class="section-heading" id="import-the-dataset-and-use-the-batch-file-to-analyze-your-data">Import the dataset, and use the batch file to analyze your data<a class="anchor" aria-label="anchor" href="#import-the-dataset-and-use-the-batch-file-to-analyze-your-data"></a></h2>
<hr class="half-width"><p>This data was collected from crude extracts from two marine
<em>Streptomyces</em>: <em>Streptomyces</em> sp. H-KF8, and
<em>Streptomyces</em> sp. Vc74B-19. Two media were used, ISP2 and ISP2
prepared with artificial seawater (ASW), to evaluate the effect of
replicating the natural environment from which these strains were
isolated.</p>
<p><a href="fig/Datasets.png">
<img src="fig/Datasets.png" alt="Data collection from *Streptomyces* sp. H-KF8, and *Streptomyces* sp. Vc74B-19." width="800" class="figure"></a></p>
<p>We downloaded 18 LC-MS/MS-derived files in mzML format. This data was
collected by Dr. Mauricio Caraballo-Rodriguez in the Dorrestein Lab, at
the University of California San Diego. There are files from each
strain, in ISP2 and ISP2-ASW, besides the crude extracts from the
culture media. The data is in triplicates.</p>
<p>Besides mzML files, there is a file metadata_table.tsv, that contains
all the relevant information from this dataset. Includes the names of
the samples, relevant data collection, and taxonomic information.</p>
<p>In addition, there is information relevant to the analysis, such as
the names of the strains, the media used for culturing, and the
antimicrobial activity. All this information is included in the format
ATTRIBUTE_*</p>
<p><a href="fig/Metadata.jpg">
<img src="fig/Metadata.jpg" alt="Data collection from *Streptomyces* sp. H-KF8, and *Streptomyces* sp. Vc74B-19." width="800" class="figure"></a></p>
<p>At last, there is a file named MZMine_FBMN_batch.xml that collects
all the information necessary for the analysis using MZMine</p>
</section><section><h2 class="section-heading" id="analysis-using-mzmine">Analysis using MZMine<a class="anchor" aria-label="anchor" href="#analysis-using-mzmine"></a></h2>
<hr class="half-width"><p>Load batch file</p>
<p>Open MZMine3, click on “Open”, and then in “Batch Mode”</p>
<p><a href="fig/batch_01.jpg">
<img src="fig/batch_01.jpg" alt="Load batch file" width="400" class="figure"></a></p>
<p>Here you should select load, and search for your downloaded files on
your computer. Then select the MZMine_FBMN_batch.xml file In
confirmation, you should select Replace the batch steps.</p>
<p><a href="fig/batch_02.jpg">
<img src="fig/batch_02.jpg" alt="Load batch file" width="600" class="figure"></a></p>
<p>Then double-click on import MS data</p>
<p><a href="fig/batch_03.jpg">
<img src="fig/batch_03.jpg" alt="Load batch file" width="800" class="figure"></a></p>
<p>Select from your computer the 18 mzML files from this dataset</p>
<p><a href="fig/batch_04.jpg">
<img src="fig/batch_04.jpg" alt="Load batch file" width="800" class="figure"></a></p>
<p>After this, every file should be included in the batch-processing
mode. Select OK afterward so the files begin to process in the
meantime.</p>
<p><a href="fig/batch_05.jpg">
<img src="fig/batch_05.jpg" alt="Load batch file" width="800" class="figure"></a></p>
<p>Briefly, MZMine is now detecting all the masses present in your
samples, grouping them, and then aligning them, so you can know in which
sample each detected spectrum is present</p>
<p>You should note, that the parameters used are specific for this
dataset, you might need to change some of the values when analyzing your
own samples.</p>
<p>For more information: Nothias, LF., Petras, D., Schmid, R. et
al. Feature-based molecular networking in the GNPS analysis environment.
Nat Methods 17, 905–908 (2020). <a href="https://doi.org/10.1038/s41592-020-0933-6" class="external-link uri">https://doi.org/10.1038/s41592-020-0933-6</a></p>
</section><section><h2 class="section-heading" id="explore-the-structure-of-our-data">Explore the structure of our data<a class="anchor" aria-label="anchor" href="#explore-the-structure-of-our-data"></a></h2>
<hr class="half-width"><p>In the “MS data files” from MZMine you can observe all the 18
LC-MS/MS files in mzML format that we loaded</p>
<p><a href="fig/MZMine_datafiles.jpg">
<img src="fig/MZMine_datafiles.jpg" alt="Datasets" width="400" class="figure"></a></p>
<p>Let’s inspect what the two of these files look like. We could select
one file from <em>Streptomyces</em> sp. H-KF8 in ISP2, and one in
ISP2-ASW We can select both files, then right-click and select “Show
chromatograms”</p>
<p><a href="fig/chromatogram_01.jpg">
<img src="fig/chromatogram_01.jpg" alt="Chromatogram" width="400" class="figure"></a></p>
<p>Here we can select the mass range that we want to observe. Since we
want to see all the spectra detected, click “Auto range”. It will
automatically will select masses ranging from 100 <em>m/z</em> to almost
3,749 <em>m/z</em> Click “OK” then</p>
<p><a href="fig/chromatogram_02.jpg">
<img src="fig/chromatogram_02.jpg" alt="Chromatogram" width="500" class="figure"></a></p>
<p>The software will display the Total Ion Chromatogram (TIC) from both
samples. In this case, strain H-KF8 is displayed in pink when cultured
in ISP2-ASW, and in black when cultured in ISP2</p>
<p><a href="fig/chromatogram_03.jpg">
<img src="fig/chromatogram_03.jpg" alt="Chromatogram" width="800" class="figure"></a></p>
<p>We can select a section of the chromatogram to inspect the
differences of the metabolomic profiles of these samples</p>
<p><a href="fig/chromatogram_04.jpg">
<img src="fig/chromatogram_04.jpg" alt="Chromatogram" width="800" class="figure"></a></p>
<p>We can observe that several spectra are produced exclusively by
strain H-KF8 in ISP2-ASW</p>
<p><a href="fig/chromatogram_05.jpg">
<img src="fig/chromatogram_05.jpg" alt="Chromatogram" width="800" class="figure"></a></p>
</section><section><h2 class="section-heading" id="analyze-the-final-output-from-the-analysis">Analyze the final output from the analysis<a class="anchor" aria-label="anchor" href="#analyze-the-final-output-from-the-analysis"></a></h2>
<hr class="half-width"><p>After processing all files. We should look at the feature lists tab
from MZMine. There we can observe that we have a file called “Aligned
feature list 13C gaps”. Double-click on that</p>
<p><a href="fig/feat_list_01.jpg">
<img src="fig/feat_list_01.jpg" alt="Feature list" width="400" class="figure"></a></p>
<p>Here we can observe the feature list, where each row is one detected
MS spectra with its <em>m/z</em> and retention time (RT). Each column is
one of the 18 samples. If MS spectra are detected in a sample, then the
height of the peak is displayed in the table</p>
<p><a href="fig/feat_list_02.jpg">
<img src="fig/feat_list_02.jpg" alt="Feature List" width="800" class="figure"></a></p>
</section><section><h2 class="section-heading" id="remove-media-blanks-ms-spectra">Remove media blanks MS spectra<a class="anchor" aria-label="anchor" href="#remove-media-blanks-ms-spectra"></a></h2>
<hr class="half-width"><p>Now we want to remove all the MS spectra that are part of the culture
media and not produced by our strains.</p>
<p>For that, we need to go to “Feature List Methods”, then click on
“Feature List Filtering”, and then on “Feature List Blank
Subtraction”</p>
<p><a href="fig/blank_01.jpg">
<img src="fig/blank_01.jpg" alt="Blank substraction" width="600" class="figure"></a></p>
<p>In the “Blank/Control raw data files” section, we need to select
“Specific raw data files”, and then select</p>
<p><a href="fig/blank_02.jpg">
<img src="fig/blank_02.jpg" alt="Blank substraction" width="600" class="figure"></a></p>
<p>Here we need to select all the samples that belong to the crude
extracts from the Media. There are 6 in total. Press OK afterward</p>
<p><a href="fig/blank_03.jpg">
<img src="fig/blank_03.jpg" alt="Blank substraction" width="600" class="figure"></a></p>
<p>Now we have two Feature lists</p>
<ol style="list-style-type: decimal"><li>Aligned feature list 13C gaps. That is the original feature list
including media MS spectra</li>
<li>Aligned feature list 13C gaps subtracted. Feature list with media
blanks removed</li>
</ol><p><a href="fig/blank_04.jpg">
<img src="fig/blank_04.jpg" alt="Blank substraction" width="400" class="figure"></a></p>
</section><section><h2 class="section-heading" id="export-feature-lists-in-gnps-format">Export Feature lists in GNPS format<a class="anchor" aria-label="anchor" href="#export-feature-lists-in-gnps-format"></a></h2>
<hr class="half-width"><p>We are going to export both Feature lists.</p>
<p>First, select “Aligned feature list 13C gaps”. Then go to “Feature
List Methods”, “Export Feature List”, and select Molecular “networking
files”</p>
<p><a href="fig/export_01.jpg">
<img src="fig/export_01.jpg" alt="Export files" width="600" class="figure"></a></p>
<p>Then click “Select”, and in “File name” write the name that you want
your files to be named. In this case, I selected
“GM_workshop_Featurelist_complete”. So I know that this file is from the
Latin American genome mining workshop and that the feature list includes
the media blank MS spectra. then press “save”</p>
<p><a href="fig/export_02.jpg">
<img src="fig/export_02.jpg" alt="Export files" width="600" class="figure"></a></p>
<p>Make sure that in Filter rows you select “MS2 or ION IDENTITY”, so
only MS spectra with MS2 are selected.</p>
<p><a href="fig/export_03.jpg">
<img src="fig/export_03.jpg" alt="Export files" width="600" class="figure"></a></p>
<p>Then, in your selected folder, you should have two files</p>
<ol style="list-style-type: decimal"><li>GM_workshop_Featurelist_complete_quant.csv</li>
</ol><p>This file is a table that includes all the feature lists in your
samples. Again, each row is an MS spectrum, and each column is each of
the 18 samples.</p>
<ol start="2" style="list-style-type: decimal"><li>GM_workshop_Featurelist_complete_quant.mgf</li>
</ol><p>This file contains the information on each spectrum. Contains the
parent mass in <em>m/z</em>, and the <em>m/z</em> values of each
fragment from that spectra, with the peak intensity of each
spectrum.</p>
<p><a href="fig/export_04.jpg">
<img src="fig/export_04.jpg" alt="Export files" width="600" class="figure"></a></p>
<p>Now we need to repeat the export step but with the media blanks
removed. This time the files will be named
“GM_workshop_Featurelist_filtered” so we can know that there is no MS
spectra that are originally from the culture media.</p>
<p>After this, we have 4 files. And we are done with the processing
steps in MZMine 3</p>
</section><section><h2 class="section-heading" id="create-a-molecular-network">Create a molecular network<a class="anchor" aria-label="anchor" href="#create-a-molecular-network"></a></h2>
<hr class="half-width"><p>Go into <a href="https://gnps.ucsd.edu/ProteoSAFe/static/gnps-splash.jsp" class="external-link">GNPS</a>
webpage</p>
<p>and login using your username and password</p>
<p>Then we should go to “Advanced Analysis Tools”, and Select “Analyze”
in the Feature Networking</p>
<p><a href="fig/FBMN_01.jpg">
<img src="fig/FBMN_01.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>Then we should write a title for our network. We could use something
like “GM_workshop_FBMN_filtered”, because we are going to use the
feature list with the media blanks removed.</p>
<p>Following that, in “File Selection”, click on “Select Input File”</p>
<p><a href="fig/FBMN_02.jpg">
<img src="fig/FBMN_02.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>We now need to upload our feature lists and our metadata table. Click
on “Upload files”</p>
<p><a href="fig/FBMN_03.jpg">
<img src="fig/FBMN_03.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>Here we can create a folder in our GNPS. I created a new folder
called “LATAM_GM_workshop”. Click on that folder. Then in the “File Drag
and Drop”, drag and drop the following files:</p>
<ol style="list-style-type: decimal"><li>GM_workshop_Featurelist_filtered.mgf</li>
<li>GM_workshop_Featurelist_filtered_quant.csv</li>
<li>metadata_table.tsv</li>
</ol><p><a href="fig/FBMN_04.jpg">
<img src="fig/FBMN_04.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>Return to the “Select Input Files” after uploading your files. You
should be able to see the three uploaded files in your directory</p>
<p><a href="fig/FBMN_05.jpg">
<img src="fig/FBMN_05.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>We are going to select the “GM_workshop_Featurelist_filtered.mgf”
file as “MS2 file in MGF format”</p>
<p><a href="fig/FBMN_06.jpg">
<img src="fig/FBMN_06.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>then we are going to select
“GM_workshop_Featurelist_filtered_quant.csv” as the “Feature
Quantification Table”</p>
<p><a href="fig/FBMN_07.jpg">
<img src="fig/FBMN_07.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>Finally, we are going to select the “metadata_table.tsv” as our
“Sample Metadata Table”</p>
<p><a href="fig/FBMN_08.jpg">
<img src="fig/FBMN_08.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>In the “Selected Files” section, we should be able to see the three
files in their corresponding sections. Click on finish selection after
checking everything is ok.</p>
<p><a href="fig/FBMN_09.jpg">
<img src="fig/FBMN_09.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>After selecting the files, we need to adjust the parameters for our
network Since our data was collected using a high-resolution LC-MS/MS,
we could adjust the</p>
<ul><li>“Precursor Ion Mass Tolerance” to 0.02 Da This value affects the
clustering of nearly identical MS/MS spectra through MS-Cluster.</li>
<li>“Fragment Ion Mass Tolerance” to 0.02 Da Sets the allowable
deviation in m/z values for fragment ions when clustering MS/MS
spectra</li>
</ul><p><a href="fig/FBMN_10.jpg">
<img src="fig/FBMN_10.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>Then we need to select our thresholds to create the molecular
network</p>
<ul><li>“min pair cos” to 0.7 The minimum cosine score required between a
pair of consensus MS/MS spectra for an edge to be created in the
molecular network.</li>
<li>“Minimum matched fragment ions” to 6 Specify the minimum number of
common fragment ions that two consensus MS/MS spectra must share to be
connected by an edge in the molecular network.</li>
</ul><p><a href="fig/FBMN_11.jpg">
<img src="fig/FBMN_11.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>After this, write your email, so you know when your network is
finished. Then click “Submit”</p>
<p><a href="fig/FBMN_12.jpg">
<img src="fig/FBMN_12.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>For more information about the rest of the parameters used for
molecular networking:</p>
<ul><li><a href="https://ccms-ucsd.github.io/GNPSDocumentation/networking/" class="external-link">visit
the GNPS documentation</a></li>
<li>Also visit Aron, A.T., Gentry, E.C., McPhail, K.L. et al. <a href="https://doi.org/10.1038/s41596-020-0317-5" class="external-link">Reproducible molecular
networking of untargeted mass spectrometry data using GNPS.</a> Nat
Protoc 15, 1954–1991 (2020).</li>
</ul></section><section><h2 class="section-heading" id="statistics-analysis-using-fbmn-stats-guide-web-server">Statistics analysis using FBMN STATS guide web server<a class="anchor" aria-label="anchor" href="#statistics-analysis-using-fbmn-stats-guide-web-server"></a></h2>
<hr class="half-width"><p>Now we want to observe how different are the metabolic profiles of
our samples. For that, we are going to calculate a Principal Coordinate
Analysis (PCoA)</p>
<p>Go into <a href="https://fbmn-statsguide.gnps2.org/" class="external-link">FBMN STATS
guide</a></p>
<p>Select “Data Preparation”</p>
<p><a href="fig/FBMN-STATS_01.jpg">
<img src="fig/FBMN-STATS_01.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>In file origin select “Quantification table and metadata files”.</p>
<ul><li>In the Quantification Table section, select the MZMine output file
“GM_workshop_Featurelist_complete_quant.csv”. This table contains the
information of the two strains and the media. We want to observe how
different our samples’ metabolomic profiles are, in comparison with the
culture media.</li>
<li>in the Meta Data Table section, include our metadata file
“metadata_table.tsv”.</li>
</ul><p><a href="fig/FBMN-STATS_02.jpg">
<img src="fig/FBMN-STATS_02.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>After loading your files, click “Submit Data for Statistics!”.</p>
<p><a href="fig/FBMN-STATS_03.jpg">
<img src="fig/FBMN-STATS_03.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>Check that your data have been properly submitted by checking that
you have the “Data preparation was successful!”.</p>
<p><a href="fig/FBMN-STATS_04.jpg">
<img src="fig/FBMN-STATS_04.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>Now we should go to the “PERMANOVA &amp; PCoA” section.</p>
<p><a href="fig/FBMN-STATS_05.jpg">
<img src="fig/FBMN-STATS_05.jpg" alt="FBMN" width="300" class="figure"></a></p>
<p>Select Principal Coordinate Analysis.</p>
<p><a href="fig/FBMN-STATS_06.jpg">
<img src="fig/FBMN-STATS_06.jpg" alt="FBMN" width="600" class="figure"></a></p>
<p>Now we could change how we want to color our samples. Select
“attribute for multivariate analysis” ATTRIBUTE_media. This will color
our samples according to the media (ISP2 or ISP2-ASW)</p>
<p><a href="fig/FBMN-STATS_07.jpg">
<img src="fig/FBMN-STATS_07.jpg" alt="FBMN" width="600" class="figure"></a></p>
<p>We can observe that there is a difference between the samples
prepared with ISP2 and ISP2-ASW</p>
<p><a href="fig/FBMN-STATS_08.jpg">
<img src="fig/FBMN-STATS_08.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>Now if we color our samples by strain, selecting ATTRIBUTE_strain in
“attribute for multivariate analysis”</p>
<p>We can observe that <em>Streptomyces</em> sp. Vc74B-19 metabolic
profile is quite different than the ones from <em>Streptomyces</em> sp.
H-KF8 and the culture media.</p>
<p>Also, it is possible to observe that <em>Streptomyces</em> sp. H-KF8
metabolomic profile differs more from the media when cultured in
ISP2-ASW</p>
<p><a href="fig/FBMN-STATS_09.jpg">
<img src="fig/FBMN-STATS_09.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>There are several statistics that you can do using FBMN STATS guide,
you can check the preprint <a href="https://doi.org/10.26434/CHEMRXIV-2023-WWBT0" class="external-link">here</a>.</p>
</section><section><h2 class="section-heading" id="visualize-the-network-using-cytoscape">Visualize the network using Cytoscape<a class="anchor" aria-label="anchor" href="#visualize-the-network-using-cytoscape"></a></h2>
<hr class="half-width"><p>We need to check if our GNPS network is done processing. You might
have received an email, otherwise, go to your GNPS account and click on
“Jobs”</p>
<p><a href="fig/cytoscape_01.jpg">
<img src="fig/cytoscape_01.jpg" alt="FBMN" width="600" class="figure"></a></p>
<p>If your network is Done, then click on it. Afterward, click on
“Direct Cytoscape Preview/Download”</p>
<p><a href="fig/cytoscape_02.jpg">
<img src="fig/cytoscape_02.jpg" alt="FBMN" width="600" class="figure"></a></p>
<p>If your network is not Done, then we could use a previously computed
network, that holds the same data that we used in this workshop</p>
<p><a href="https://cytoscape.gnps2.org/process?task=568cc93ed2e84f20b2b94bd186b48ee8#%7B%7D" class="external-link">Cytoscape</a></p>
<p>Click on “Download Cytoscape File”, and save it in your computer</p>
<p><a href="fig/cytoscape_03.jpg">
<img src="fig/cytoscape_03.jpg" alt="FBMN" width="600" class="figure"></a></p>
<p>Open the network on Cytoscape. We want to change the style of the
network and color the nodes according to the strains. Click on
“Style”</p>
<p><a href="fig/cytoscape_04.jpg">
<img src="fig/cytoscape_04.jpg" alt="FBMN" width="600" class="figure"></a></p>
<p>Select the pie chart icon on Image/Chart. This will help us figure
out the abundance of each MS spectra, depending on the metadata
parameter that we choose</p>
<p><a href="fig/cytoscape_05.jpg">
<img src="fig/cytoscape_05.jpg" alt="FBMN" width="600" class="figure"></a></p>
<p>In this case, we will select the two strains, and also the media
Although we removed all the MS spectra present in the media, there might
still be some nodes with MS spectra detected in some of the media
samples.</p>
<p><a href="fig/cytoscape_06.jpg">
<img src="fig/cytoscape_06.jpg" alt="FBMN" width="600" class="figure"></a></p>
<p>If we click “Apply”, we can observe the nodes present in each strain,
and the media. In this case, the light blue means the MS spectra
detected in <em>Streptomyces</em> sp. Vc74B-19</p>
<p>In this Molecular Family, we can observe some MS spectra with
annotations. Some are similar to Urdamycinone B, Dehydroxyaquayamycin,
and other Angucycline-related compounds.</p>
<p><a href="fig/cytoscape_07.jpg">
<img src="fig/cytoscape_07.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>We can color now the nodes depending on the media where they are
detected. Select the pie chart again in “Image/Chart”, and now select
ISP2 and ISP2-ASW. This will color ISP2 in red, and ISP2-ASW in light
blue</p>
<p><a href="fig/cytoscape_08.jpg">
<img src="fig/cytoscape_08.jpg" alt="FBMN" width="600" class="figure"></a></p>
<p>We can observe that some nodes are detected mostly in ISP2. However,
several angucycline-related compounds are detected almost exclusively in
ISP2-ASW</p>
<p><a href="fig/cytoscape_09.jpg">
<img src="fig/cytoscape_09.jpg" alt="FBMN" width="800" class="figure"></a></p>
</section><section><h2 class="section-heading" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
<hr class="half-width"><ul><li>Wang, M., Carver, J., Phelan, V. et al. Sharing and community
curation of mass spectrometry data with Global Natural Products Social
Molecular Networking. Nat Biotechnol 34, 828–837 (2016). <a href="https://doi.org/10.1038/nbt.3597" class="external-link uri">https://doi.org/10.1038/nbt.3597</a>.</li>
<li>Schmid, R., Heuckeroth, S., Korf, A. et al. Integrative analysis of
multimodal mass spectrometry data in MZmine 3. Nat Biotechnol 41,
447–449 (2023). <a href="https://doi.org/10.1038/s41587-023-01690-2" class="external-link uri">https://doi.org/10.1038/s41587-023-01690-2</a>.</li>
<li>Nothias, LF., Petras, D., Schmid, R. et al. Feature-based molecular
networking in the GNPS analysis environment. Nat Methods 17, 905–908
(2020). <a href="https://doi.org/10.1038/s41592-020-0933-6" class="external-link uri">https://doi.org/10.1038/s41592-020-0933-6</a>
</li>
<li>Aron, A.T., Gentry, E.C., McPhail, K.L. et al. Reproducible
molecular networking of untargeted mass spectrometry data using GNPS.
Nat Protoc 15, 1954–1991 (2020). <a href="https://doi.org/10.1038/s41596-020-0317-5" class="external-link uri">https://doi.org/10.1038/s41596-020-0317-5</a>
</li>
<li>Shah, A. K. P., Walter, A., Ottosson, F., Russo, F., Navarro-Díaz,
M., Boldt, J., Kalinski, J.-C., Kontou, E. E., Elofson, J., Polyzois,
A., González-Marín, C., Farrell, S., Aggerbeck, M. R., Pruksatrakul, T.,
Chan, N., Wang, Y., Pöchhacker, M., Brungs, C., Cámara, B., … Petras, D.
(2023). The Hitchhiker’s Guide to Statistical Analysis of Feature-based
Molecular Networks from Non-Targeted Metabolomics Data. ChemRxiv, 1–83.
<a href="https://doi.org/10.26434/CHEMRXIV-2023-WWBT0" class="external-link uri">https://doi.org/10.26434/CHEMRXIV-2023-WWBT0</a>
</li>
</ul><p>In the future it will include:</p>
<ul><li>
<a href="https://www.tfleao.com/general-8" class="external-link uri">https://www.tfleao.com/general-8</a>,</li>
<li><a href="https://pairedomicsdata.bioinformatics.nl/" class="external-link">Paired
omics</a></li>
<li>In the future it will include: SIRIUS: Kai Dührkop, Markus
Fleischauer, Marcus Ludwig, Alexander A. Aksenov, Alexey V. Melnik,
Marvin Meusel, Pieter C. Dorrestein, Juho Rousu, and Sebastian Böcker,
SIRIUS 4: Turning tandem mass spectra into metabolite structure
information. Nature Methods 16, 299–302, 2019.</li>
</ul><div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<ul><li>Data is generated using Liquid Chromatography coupled to a tandem
mass spectrometer (LC-MS/MS or MS2).</li>
<li>Dereplication is the process of identifying previously known
compounds.</li>
<li>Molecular networking is a computational method that organizes MS2
data based on spectral similarity, allowing us to infer relationships
between chemical structures</li>
<li>Feature-Based Molecular Networking (FBMN) enhances classical
molecular networking by integrating relative quantitative data, enabling
more robust metabolomics statistical analysis.</li>
</ul></div>
</div>
</div>
</section></div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="08-GatorGC.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="10-OtherResources.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="08-GatorGC.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: GATOR-GC: Genomic
        </a>
        <a class="chapter-link float-end" href="10-OtherResources.html" rel="next">
          Next: Other Resources...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://github.com/carpentries-incubator/genome-mining/edit/main/episodes/09-metabolomics.md" class="external-link">Edit on GitHub</a>

	
        | <a href="https://github.com/carpentries-incubator/genome-mining/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/carpentries-incubator/genome-mining/" class="external-link">Source</a></p>
        <p>
        <a href="https://github.com/carpentries-incubator/genome-mining/blob/main/" class="external-link">Cite</a>
        | <a href="mailto:axel.ramos3737@gmail.com">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
		</div>
		<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.18.5" class="external-link">sandpaper (0.18.5)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.9" class="external-link">pegboard (0.7.9)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.9" class="external-link">varnish (1.0.9)</a></p>
		</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "LearningResource",
  "@id": "https://carpentries-incubator.github.io/genome-mining/09-metabolomics.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/LearningResource/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "software, data, lesson, The Carpentries",
  "name": "Metabolomics workshop",
  "creativeWorkStatus": "active",
  "url": "https://carpentries-incubator.github.io/genome-mining/09-metabolomics.html",
  "identifier": "https://carpentries-incubator.github.io/genome-mining/09-metabolomics.html",
  "dateCreated": "2021-06-20",
  "dateModified": "2026-02-20",
  "datePublished": "2026-02-20"
}

  </script><script>
		feather.replace();
	</script><!-- Matomo --><script>
          var _paq = window._paq = window._paq || [];
          /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
          _paq.push(["setDocumentTitle", document.domain + "/" + document.title]);
          _paq.push(["setDomains", ["*.lessons.carpentries.org","*.datacarpentry.github.io","*.datacarpentry.org","*.librarycarpentry.github.io","*.librarycarpentry.org","*.swcarpentry.github.io", "*.carpentries.github.io"]]);
          _paq.push(["setDoNotTrack", true]);
          _paq.push(["disableCookies"]);
          _paq.push(["trackPageView"]);
          _paq.push(["enableLinkTracking"]);
          (function() {
              var u="https://matomo.carpentries.org/";
              _paq.push(["setTrackerUrl", u+"matomo.php"]);
              _paq.push(["setSiteId", "1"]);
              var d=document, g=d.createElement("script"), s=d.getElementsByTagName("script")[0];
              g.async=true; g.src="https://matomo.carpentries.org/matomo.js"; s.parentNode.insertBefore(g,s);
          })();
        </script><!-- End Matomo Code --></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

