<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<title>Genome Mining in Prokaryotes: All Images</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<script src="assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="assets/styles.css">
<script src="assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="favicons/incubator/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicons/incubator/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="favicons/incubator/favicon-16x16.png">
<link rel="manifest" href="favicons/incubator/site.webmanifest">
<link rel="mask-icon" href="favicons/incubator/safari-pinned-tab.svg" color="#5bbad5">
<meta name="msapplication-TileColor" content="#da532c">
<meta name="theme-color" media="(prefers-color-scheme: light)" content="white">
<meta name="theme-color" media="(prefers-color-scheme: dark)" content="black">
</head>
<body>
    <header id="top" class="navbar navbar-expand-md top-nav incubator"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="incubator-logo" alt="Lesson Description" src="assets/images/incubator-logo.svg"><span class="badge text-bg-danger">
          <abbr title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.">
            <a href="https://docs.carpentries.org/resources/curriculum/lesson-life-cycle.html" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
              Pre-Alpha
            </a>
            <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text">
<li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul>
</li>
      </div>


      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Learner View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1">
<li><button class="dropdown-item" type="button" onclick="window.location.href='instructor/images.html';">Instructor View</button></li>
        </ul>
</div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Lesson Description" src="assets/images/incubator-logo-sm.svg">
</div>
    <div class="lesson-title-md">
      Genome Mining in Prokaryotes
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0">
<li class="nav-item">
          <span class="lesson-title">
            Genome Mining in Prokaryotes
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="reference.html#glossary">Glossary</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="profiles.html">Learner Profiles</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown">
<li><a class="dropdown-item" href="discuss.html">Discussion</a></li>
<li><a class="dropdown-item" href="reference.html">Glossary</a></li>
          </ul>
</li>
      </ul>
</div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div>
<!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Genome Mining in Prokaryotes
</div>

<aside class="col-md-12 lesson-progress"><div style="width: %" class="percentage">
    %
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: %" aria-valuenow="" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text">
<li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul>
</li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Learner View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="instructor/images.html">Instructor View</a>
                      </div>
                    </div>
                  </div>
<!--/div.accordion-item-->
                </div>
<!--/div.accordion-flush-->
              </div>
<!--div.sidenav-view-selector -->
            </div>
<!--/div.col -->

            <hr>
</div>
<!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Setup</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="01-introduction.html">1. Introduction to Genome Mining</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="02-antismash.html">2. Secondary metabolite biosynthetic gene cluster identification</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="03-antiSMASHdb-MiBiG.html">3. Genome Mining Databases</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="04-bigscape.html">4. BGC Similarity Networks</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="05-bigslice-bigfam.html">5. Homologous BGC Clusterization</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="06-corason.html">6. Finding Variation on Genomic Vicinities</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="07-evomining.html">7. Evolutionary Genome Mining</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="08-GatorGC.html">8. GATOR-GC: Genomic Assessment Tool for Orthologous Regions and Gene Clusters</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush10">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading10">
        <a href="09-metabolomics.html">9. Metabolomics workshop</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush11">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading11">
        <a href="10-OtherResources.html">10. Other Resources</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width">
<div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul>
<li>
                        <a href="key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="reference.html#glossary">Glossary</a>
                      </li>
                      <li>
                        <a href="profiles.html">Learner Profiles</a>
                      </li>
                      <li><a href="discuss.html">Discussion</a></li>
<li><a href="reference.html">Glossary</a></li>
                    </ul>
</div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources">
<a href="aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none">
<div class="d-grid gap-1">

            </div>
          </div>
<!-- /div.accordion -->
        </div>
<!-- /div.sidebar-inner -->
      </nav>
</div>
<!-- /div.sidebar -->
  </div>
<!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-extra.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <main id="main-content" class="main-content"><div class="container lesson-content">
        
        
<section id="01-introduction"><h2 class="section-heading"><a href="01-introduction.html">Introduction to Genome Mining</a></h2>
<hr class="half-width">
<h3 id="01-introduction-figure-1">Figure 1</h3>
<p aria-hidden="true">Image 1 of 1: ‘Complete pipeline of genome mining. From a single genome, this example obtains their BGC and compares them with other BGC from related genomes’</p>
<a href="fig/Chapter1Fig1.png">
<img src="fig/Introduction01.png" alt="Complete pipeline of genome mining. From a single genome, this example obtains their BGC and compares them with other BGC from related genomes" width="60%" height="60%" class="figure"></a><hr>
<h3 id="01-introduction-figure-2">Figure 2</h3>
<p aria-hidden="true">Image 1 of 1: ‘Natural products can be produced by bacteria, fungi, plants and animals’</p>
<a href="fig/productos%20naturales.png">
<img src="fig/productos%20naturales.png" alt="Natural products can be produced by bacteria, fungi, plants and animals" class="figure"></a><hr>
<h3 id="01-introduction-figure-3">Figure 3</h3>
<p aria-hidden="true">Image 1 of 1: ‘BGC arrange example’</p>
<a href="fig/08-20-24_BGC_arrange.png">
<img src="fig/08-20-24_BGC_arrange.png" alt="BGC arrange example" class="figure"></a><hr>
<h3 id="01-introduction-figure-4">Figure 4</h3>
<p aria-hidden="true">Image 1 of 1: ‘NRPS animation of fakeomycin’</p>
<a href="fig/NRPS-animation-Mullowney_16-9_0.08.gif">
<img src="fig/NRPS-animation-Mullowney_16-9_0.08.gif" alt="NRPS animation of fakeomycin" class="figure"></a><hr>
<h3 id="01-introduction-figure-5">Figure 5</h3>
<p aria-hidden="true">Image 1 of 1: ‘MIBiG layout of the Chloramphenicol gene cluster from _Streptomyces venezuelae_ comprising 17 genes’</p>
<a href="fig/episode1-fig1.PNG">
<img src="fig/episode1-fig1.PNG" alt="MIBiG layout of the Chloramphenicol gene cluster from _Streptomyces venezuelae_ comprising 17 genes" class="figure"></a><hr></section><section id="02-antismash"><h2 class="section-heading"><a href="02-antismash.html">Secondary metabolite biosynthetic gene cluster identification</a></h2>
<hr class="half-width"></section><section id="03-antiSMASHdb-MiBiG"><h2 class="section-heading"><a href="03-antiSMASHdb-MiBiG.html">Genome Mining Databases</a></h2>
<hr class="half-width">
<h3 id="03-antiSMASHdb-MiBiG-figure-1">Figure 1</h3>
<p aria-hidden="true">Image 1 of 1: ‘MIBiG website homepage highlighting the search tool’</p>
<a href="fig/MIBiG_search.png">
<img src="fig/MIBiG_search.png" alt="MIBiG website homepage highlighting the search tool" class="figure"></a><hr>
<h3 id="03-antiSMASHdb-MiBiG-figure-2">Figure 2</h3>
<p aria-hidden="true">Image 1 of 1: ‘MIBiG website query page’</p>
<a href="fig/MIBiG_query.png">
<img src="fig/MIBiG_query.png" alt="MIBiG website query page" class="figure"></a><hr>
<h3 id="03-antiSMASHdb-MiBiG-figure-3">Figure 3</h3>
<p aria-hidden="true">Image 1 of 1: ‘MIBiG website displaying the results from the simple search Streptococcus’</p>
<a href="fig/MIBiG_results.png">
<img src="fig/MIBiG_results.png" alt="MIBiG website displaying the results from the simple search Streptococcus" class="figure"></a><hr>
<h3 id="03-antiSMASHdb-MiBiG-figure-4">Figure 4</h3>
<p aria-hidden="true">Image 1 of 1: ‘antiSMASH website homepage’</p>
<a href="fig/antiSMASH_db.png">
<img src="fig/antiSMASH_db.png" alt="antiSMASH website homepage" class="figure"></a><hr>
<h3 id="03-antiSMASHdb-MiBiG-figure-5">Figure 5</h3>
<p aria-hidden="true">Image 1 of 1: ‘antiSMASH website query page’</p>
<a href="fig/antiSMASH_search.png">
<img src="fig/antiSMASH_search.png" alt="antiSMASH website query page" class="figure"></a><hr>
<h3 id="03-antiSMASHdb-MiBiG-figure-6">Figure 6</h3>
<p aria-hidden="true">Image 1 of 1: ‘antiSMASH website displaying the results from the simple search Streptococcus’</p>
<a href="fig/antiSMASH_query.png">
<img src="fig/antiSMASH_query.png" alt="antiSMASH website displaying the results from the simple search Streptococcus" class="figure"></a><hr>
<h3 id="03-antiSMASHdb-MiBiG-figure-7">Figure 7</h3>
<p aria-hidden="true">Image 1 of 1: ‘a dataframe variable the content of the Streptococcus predicted BGC’</p>
<a href="fig/21-08-24-chapter9-table.png">
<img src="fig/21-08-24-chapter9-table.png" alt="a dataframe variable the content of the Streptococcus predicted BGC" width="800" class="figure"></a><hr>
<h3 id="03-antiSMASHdb-MiBiG-figure-8">Figure 8</h3>
<p aria-hidden="true">Image 1 of 1: ‘the content of the ocurrences grouped by species column’</p>
<a href="fig/21-08-24-chapter9-occurences.png">
<img src="fig/21-08-24-chapter9-occurences.png" alt="the content of the ocurrences grouped by species column" width="500" class="figure"></a><hr>
<h3 id="03-antiSMASHdb-MiBiG-figure-9">Figure 9</h3>
<p aria-hidden="true">Image 1 of 1: ‘visualization of the BGC content on a heatmap.’</p>
<a href="fig/21-08-24-chapter9-heatmap.png">
<img src="fig/21-08-24-chapter9-heatmap.png" alt="visualization of the BGC content on a heatmap." width="800" class="figure"></a><hr>
<h3 id="03-antiSMASHdb-MiBiG-figure-10">Figure 10</h3>
<p aria-hidden="true">Image 1 of 1: ‘visualization of the BGC content of S. agalactiae. on a sctterplot’</p>
<a href="fig/21-08-24-chapter9.dotplot.png">
<img src="fig/21-08-24-chapter9.dotplot.png" alt="visualization of the BGC content of S. agalactiae. on a sctterplot" width="800" class="figure"></a><hr>
<h3 id="03-antiSMASHdb-MiBiG-figure-11">Figure 11</h3>
<p aria-hidden="true">Image 1 of 1: ‘visualization of the BGC content on a scatterplot’</p>
<a href="fig/21-08-24-chapter9.dotplot2.png">
<img src="fig/21-08-24-chapter9.dotplot2.png" alt="visualization of the BGC content on a scatterplot" width="800" class="figure"></a><hr>
<h3 id="03-antiSMASHdb-MiBiG-figure-12">Figure 12</h3>
<p aria-hidden="true">Image 1 of 1: ‘filtered heatmap ’</p>
<a href="fig/21-08-24-chapter9-pivot-filtered.png">
<img src="fig/21-08-24-chapter9-pivot-filtered.png" alt="filtered heatmap " width="800" class="figure"></a><hr></section><section id="04-bigscape"><h2 class="section-heading"><a href="04-bigscape.html">BGC Similarity Networks</a></h2>
<hr class="half-width">
<h3 id="04-bigscape-figure-1">Figure 1</h3>
<p aria-hidden="true">Image 1 of 1: ‘BIG-SCAPE output as visualized in the web page.
                                  	The overview page is displayed. At the left is the Run Information,
                                  	indicating the date and time at which the analysis was started and completed, as well as the parameters of the run.
                                  	Next is displayed the Input Data, specifying the total number of genomes and the total BGCs,
                                  	in this example 8 and 23, respectively. There are two pie charts, one representing the BGC per Genome
                                  	and the other the BGC per Class. At the right is the Network Overview,
                                  	which allows selection between mix and the different BGC classes.
                                  	From the mix overview it displays the Number of Families, Average number of BGCs per family,
                                  	Max number of BGCs in a family and the Families with MIBiG Reference BGCs.’</p>
<a href="fig/02_05_01.png">
<img src="fig/02_05_01.png" alt="BIG-SCAPE output as visualized in the web page.
                                  	The overview page is displayed. At the left is the Run Information,
                                  	indicating the date and time at which the analysis was started and completed, as well as the parameters of the run.
                                  	Next is displayed the Input Data, specifying the total number of genomes and the total BGCs,
                                  	in this example 8 and 23, respectively. There are two pie charts, one representing the BGC per Genome
                                  	and the other the BGC per Class. At the right is the Network Overview,
                                  	which allows selection between mix and the different BGC classes.
                                  	From the mix overview it displays the Number of Families, Average number of BGCs per family,
                                  	Max number of BGCs in a family and the Families with MIBiG Reference BGCs." class="figure"></a><hr>
<h3 id="04-bigscape-figure-2">Figure 2</h3>
<p aria-hidden="true">Image 1 of 1: ‘BIG-SCAPE output as visualized in the web page.
                                  	The overview page displaying the clustered heatmap of the presence/absence of the GCFs,
                                  	each class organized as a column at the base of the table, in each genome,
                                  	which are organized as rows at the right side of the table. Presence is depicted in red with 1 and absence in beige with 0.’</p>
<a href="fig/02_05_02.png">
<img src="fig/02_05_02.png" alt="BIG-SCAPE output as visualized in the web page.
                                  	The overview page displaying the clustered heatmap of the presence/absence of the GCFs,
                                  	each class organized as a column at the base of the table, in each genome,
                                  	which are organized as rows at the right side of the table. Presence is depicted in red with 1 and absence in beige with 0." class="figure"></a><hr>
<h3 id="04-bigscape-figure-3">Figure 3</h3>
<p aria-hidden="true">Image 1 of 1: ‘BIG-SCAPE similarity network of the complete mix of BGCs
                                  	obtained from the run. A network is represented for each GCF, each dot represents a BGC.
                                  	In this example there are a total of 23 BGCs, of which 6 are singletons, there are 28 links and 11 families.
                                  	Clicking over any of the dots shows the GCF at the right side and allows you to explore it further.’</p>
<a href="fig/02_05_03.png">
<img src="fig/02_05_03.png" alt="BIG-SCAPE similarity network of the complete mix of BGCs
                                  	obtained from the run. A network is represented for each GCF, each dot represents a BGC.
                                  	In this example there are a total of 23 BGCs, of which 6 are singletons, there are 28 links and 11 families.
                                  	Clicking over any of the dots shows the GCF at the right side and allows you to explore it further." class="figure"></a><hr>
<h3 id="04-bigscape-figure-4">Figure 4</h3>
<p aria-hidden="true">Image 1 of 1: ‘BIG-SCAPE output displaying a tree of phylogenetic distances
                                  	among the BGCs comprised in a GCF. The example shows the GCF 10, comprised by six members.
                                  	Each BGC is represented in the tree by an arrow diagram of the genes and the protein domains in the genes
                                  	corresponding to that cluster.’</p>
<a href="fig/02_05_04.png">
<img src="fig/02_05_04.png" alt="BIG-SCAPE output displaying a tree of phylogenetic distances
                                  	among the BGCs comprised in a GCF. The example shows the GCF 10, comprised by six members.
                                  	Each BGC is represented in the tree by an arrow diagram of the genes and the protein domains in the genes
                                  	corresponding to that cluster." class="figure"></a><hr></section><section id="05-bigslice-bigfam"><h2 class="section-heading"><a href="05-bigslice-bigfam.html">Homologous BGC Clusterization</a></h2>
<hr class="half-width">
<h3 id="05-bigslice-bigfam-figure-1">Figure 1</h3>
<p aria-hidden="true">Image 1 of 1: ‘Three biomolecules are depicted in three different colors; blue, red and purple. These molecules are present in BGCs of diverse bacterial lineages and in turn grouped into Gene Cluster Families (GCFs). GF1 contains three domains related to the blue molecule and three from the red molecule. GF2 contains four domains of each of the molecules, blue, red and purple. GCF3 and GCF4 contain each three domains associated with the purple biomolecule but different from each other. Every GCF produces a different metabolite, here represented as weapons or tools.’</p>
<figure><img src="fig/02-04-02.png" alt="Three biomolecules are depicted in three different colors; blue, red and purple. These molecules are present in BGCs of diverse bacterial lineages and in turn grouped into Gene Cluster Families (GCFs). GF1 contains three domains related to the blue molecule and three from the red molecule. GF2 contains four domains of each of the molecules, blue, red and purple. GCF3 and GCF4 contain each three domains associated with the purple biomolecule but different from each other. Every GCF produces a different metabolite, here represented as weapons or tools." class="figure mx-auto d-block"></figure><hr>
<h3 id="05-bigslice-bigfam-figure-2">Figure 2</h3>
<p aria-hidden="true">Image 1 of 1: ‘Example of tsv table composed by five columns and two rows.
                                  	The first row contains the title for each column; # Dataset name, Path to folder, Path to taxonomy, Description.’</p>
<a href="fig/02-04-03.png">
<img src="fig/02-04-03.png" alt="Example of tsv table composed by five columns and two rows.
                                  	The first row contains the title for each column; # Dataset name, Path to folder, Path to taxonomy, Description." class="figure"></a><hr>
<h3 id="05-bigslice-bigfam-figure-3">Figure 3</h3>
<p aria-hidden="true">Image 1 of 1: ‘Example of the structure of the input-folder.
                                  	The content of the directory input-folder is displayed in a tree-like format, listing the files and directories inside it.’</p>
<a href="fig/02-04-04.png">
<img src="fig/02-04-04.png" alt="Example of the structure of the input-folder.
                                  	The content of the directory input-folder is displayed in a tree-like format, listing the files and directories inside it." class="figure"></a><hr>
<h3 id="05-bigslice-bigfam-figure-4">Figure 4</h3>
<p aria-hidden="true">Image 1 of 1: ‘BiG SLiCE web page output displaying the results obtained from the example run.
                                  	A left panel presents the information generated, composed of seven tabs;
                                  	Summary, Datasets, Runs, Reports, Help, Feedback, and About.
                                  	The rest is a Summary of all datasets and runs included in the output.
                                  	Appearing firstly, the Datasets provided as input, organized as a table with five rows;
                                  	name, #genomes, total, with taxonomy, and description.
                                  	Next, the information about the Runs, also organized as a table with the following rows;
                                  	name, started, finished, status, threshold, #BGCs, and #GCFs..’</p>
<a href="fig/02-04-05.png">
<img src="fig/02-04-05.png" alt="BiG SLiCE web page output displaying the results obtained from the example run.
                                  	A left panel presents the information generated, composed of seven tabs;
                                  	Summary, Datasets, Runs, Reports, Help, Feedback, and About.
                                  	The rest is a Summary of all datasets and runs included in the output.
                                  	Appearing firstly, the Datasets provided as input, organized as a table with five rows;
                                  	name, #genomes, total, with taxonomy, and description.
                                  	Next, the information about the Runs, also organized as a table with the following rows;
                                  	name, started, finished, status, threshold, #BGCs, and #GCFs.." class="figure"></a><hr>
<h3 id="05-bigslice-bigfam-figure-5">Figure 5</h3>
<p aria-hidden="true">Image 1 of 1: ‘BiG SLiCE web page output displaying the information obtained from the Run-0001.
                                  	Firstly, it is shown an Overview from the Run. Indicating the Status, when was it Started and Finished,
                                  	as well as the Parameters, the HMM models and the Input.
                                  	Next, the Statistics about the run are shown as two bar-plots.
                                  	The left one plots the average distance to GCF models, whilst the right one shows the amount of BGCs in GCFs.’</p>
<a href="fig/02-04-06.png">
<img src="fig/02-04-06.png" alt="BiG SLiCE web page output displaying the information obtained from the Run-0001.
                                  	Firstly, it is shown an Overview from the Run. Indicating the Status, when was it Started and Finished,
                                  	as well as the Parameters, the HMM models and the Input.
                                  	Next, the Statistics about the run are shown as two bar-plots.
                                  	The left one plots the average distance to GCF models, whilst the right one shows the amount of BGCs in GCFs." class="figure"></a><hr>
<h3 id="05-bigslice-bigfam-figure-6">Figure 6</h3>
<p aria-hidden="true">Image 1 of 1: ‘BiG SLiCE web page output displaying detailed information regarding the BGCs from GCF_7.
                                  	The Arrowers show a gene arrow visualization of the domains that are part of each of the genes of the BGCs belonging to GCF_7.’</p>
<a href="fig/02-04-07.png">
<img src="fig/02-04-07.png" alt="BiG SLiCE web page output displaying detailed information regarding the BGCs from GCF_7.
                                  	The Arrowers show a gene arrow visualization of the domains that are part of each of the genes of the BGCs belonging to GCF_7." class="figure"></a><hr>
<h3 id="05-bigslice-bigfam-figure-7">Figure 7</h3>
<p aria-hidden="true">Image 1 of 1: ‘BiG-FAM main page showing an introduction as well as a graphical representation of the database.
                                  	A left panel displays the available options; Home, GCFs, BGCs, Query, Statistics, Help, Feedback, and About. ’</p>
<a href="fig/02-04-08.png">
<img src="fig/02-04-08.png" alt="BiG-FAM main page showing an introduction as well as a graphical representation of the database.
                                  	A left panel displays the available options; Home, GCFs, BGCs, Query, Statistics, Help, Feedback, and About. " class="figure"></a><hr>
<h3 id="05-bigslice-bigfam-figure-8">Figure 8</h3>
<p aria-hidden="true">Image 1 of 1: ‘BiG-FAM query page with the option for inserting an antiSMASH job and submitting it.
                                  	Below it is described how much of the job is Pending, Processing, and Processed.’</p>
<a href="fig/02-04-09.png">
<img src="fig/02-04-09.png" alt="BiG-FAM query page with the option for inserting an antiSMASH job and submitting it.
                                  	Below it is described how much of the job is Pending, Processing, and Processed." class="figure"></a><hr>
<h3 id="05-bigslice-bigfam-figure-9">Figure 9</h3>
<p aria-hidden="true">Image 1 of 1: ‘BiG-FAM result page indicating firstly an overview of the job; the query which was submitted,
                                  	its status, as well as the time at which it was started and finished.
                                  	Next, a table indicating the BGCs from the database which are related with the query BGCs.
                                  	This is organized as a table with seven rows; query BGC, class, length (kb), completeness, best hit, distance and details’</p>
<a href="fig/02-04-10.png">
<img src="fig/02-04-10.png" alt="BiG-FAM result page indicating firstly an overview of the job; the query which was submitted,
                                  	its status, as well as the time at which it was started and finished.
                                  	Next, a table indicating the BGCs from the database which are related with the query BGCs.
                                  	This is organized as a table with seven rows; query BGC, class, length (kb), completeness, best hit, distance and details" class="figure"></a><hr></section><section id="06-corason"><h2 class="section-heading"><a href="06-corason.html">Finding Variation on Genomic Vicinities</a></h2>
<hr class="half-width">
<h3 id="06-corason-figure-1">Figure 1</h3>
<p aria-hidden="true">Image 1 of 1: ‘CORASON's workflow for sorting phylogenetically BGCs. Given a query gene in a reference BGC and a genomic annotated database,
                                     CORASON firstly searches for query gene homologues, it filters out all genomic vicinities
                                     not related to the reference BGC. Then, CORASON infers a phylogenetic tree and
                                     calculates the frequency of occurrence for each gene family from the reference BGC.
                                     Using the same reference BGC, if a new query gene is selected, CORASON visualizes
                                     a new phylogeny with families containing the same molecular modifications.’</p>
<a href="fig/corason.png">
<img src="fig/corason.png" alt="CORASON's workflow for sorting phylogenetically BGCs. Given a query gene in a reference BGC and a genomic annotated database,
                                     CORASON firstly searches for query gene homologues, it filters out all genomic vicinities
                                     not related to the reference BGC. Then, CORASON infers a phylogenetic tree and
                                     calculates the frequency of occurrence for each gene family from the reference BGC.
                                     Using the same reference BGC, if a new query gene is selected, CORASON visualizes
                                     a new phylogeny with families containing the same molecular modifications." class="figure"></a><hr>
<h3 id="06-corason-figure-2">Figure 2</h3>
<p aria-hidden="true">Image 1 of 1: ‘CORASON phylogenetic svg reconstruction using cpsG as query gene and _S. agalactiae_ 1000006 as query cluster.
                                    At the bottom, it is displayed the frequency of occurrence for each gene family from the reference BGC,
                                    each with a different color.’</p>
<a href="fig/ViewOutput.jpg">
<img src="fig/ViewOutput.jpg" alt="CORASON phylogenetic svg reconstruction using cpsG as query gene and _S. agalactiae_ 1000006 as query cluster.
                                    At the bottom, it is displayed the frequency of occurrence for each gene family from the reference BGC,
                                    each with a different color." class="figure"></a><hr>
<h3 id="06-corason-figure-3">Figure 3</h3>
<p aria-hidden="true">Image 1 of 1: ‘CORASON phylogenetic svg reconstruction using cpsG as query gene and _S. agalactiae_ 1000006 as query cluster.
                                    At the bottom, it is displayed the frequency of occurrence for each gene family from the reference BGC,
                                    each with a different color.’</p>
<a href="fig/Joined.svg">
<img src="fig/Joined.svg" alt="CORASON phylogenetic svg reconstruction using cpsG as query gene and _S. agalactiae_ 1000006 as query cluster.
                                    At the bottom, it is displayed the frequency of occurrence for each gene family from the reference BGC,
                                    each with a different color." class="figure"></a><hr></section><section id="07-evomining"><h2 class="section-heading"><a href="07-evomining.html">Evolutionary Genome Mining</a></h2>
<hr class="half-width">
<h3 id="07-evomining-figure-1">Figure 1</h3>
<p aria-hidden="true">Image 1 of 1: ‘a) EvoMining expansion-and-recruitment pipeline. A group of grey stacked cylinders representing genomes in a database (DB).
                                   	Homologues and expansions of seed enzymes, represented as an orange arrow, from the enzyme DB
                                   	are searched by blastp in the genome DB.
                                   	The outcome is integrated as the expanded enzyme families (EFs) within the genome DB.
                                   	Bidirectional best hits (BBH) of seed enzymes, red arrows, are marked as conserved metabolism.
                                   	The EFs are amplified after being compared against a DB of natural products (NP) biosynthetic enzymes,
                                   	represented by a blue cylinder, to find recruitments defined as enzymes of the family that are part of a MIBiG BGC.
                                   	b) The genome DB, represented by the gray stacked cylinders, is searched as previously described.
                                   	Additionally, antiSMASH predictions, cyan arrows, can be added by the user.
                                   	antiSMASH enzyme predictions that are at the same time marked in red are defined as transition enzymes, purple arrows.
                                   	c) EvoMining phylogenetic reconstruction and visualization. On the left side, a phylogenetic reconstruction of an EF is shown.
                                   	On the right side it is shown the EvoMining tree displaying the EvoMining predictions (green),
                                   	which are those extra copies closer to enzyme recruitments into BGC (blue) than to conserved metabolic enzymes (red).
                                   	antiSMASH predicted enzymes are represented in cyan, transition enzymes in black and
                                   	extra copies that are neither antiSMASH nor EvoMining predictions are left in grey.’</p>
<a href="fig/evomining.gif">
<img src="fig/evomining.gif" alt="a) EvoMining expansion-and-recruitment pipeline. A group of grey stacked cylinders representing genomes in a database (DB).
                                   	Homologues and expansions of seed enzymes, represented as an orange arrow, from the enzyme DB
                                   	are searched by blastp in the genome DB.
                                   	The outcome is integrated as the expanded enzyme families (EFs) within the genome DB.
                                   	Bidirectional best hits (BBH) of seed enzymes, red arrows, are marked as conserved metabolism.
                                   	The EFs are amplified after being compared against a DB of natural products (NP) biosynthetic enzymes,
                                   	represented by a blue cylinder, to find recruitments defined as enzymes of the family that are part of a MIBiG BGC.
                                   	b) The genome DB, represented by the gray stacked cylinders, is searched as previously described.
                                   	Additionally, antiSMASH predictions, cyan arrows, can be added by the user.
                                   	antiSMASH enzyme predictions that are at the same time marked in red are defined as transition enzymes, purple arrows.
                                   	c) EvoMining phylogenetic reconstruction and visualization. On the left side, a phylogenetic reconstruction of an EF is shown.
                                   	On the right side it is shown the EvoMining tree displaying the EvoMining predictions (green),
                                   	which are those extra copies closer to enzyme recruitments into BGC (blue) than to conserved metabolic enzymes (red).
                                   	antiSMASH predicted enzymes are represented in cyan, transition enzymes in black and
                                   	extra copies that are neither antiSMASH nor EvoMining predictions are left in grey." class="figure"></a><hr>
<h3 id="07-evomining-figure-2">Figure 2</h3>
<p aria-hidden="true">Image 1 of 1: ‘EvoMining phylogenetic reconstruction providing evolutionary insights into the metabolic origin
and the fate of members of diverse EF from the Streptococcus example.
Seed enzymes are labeled in orange. The most conserved copies or central metabolism copies are marked in red.
Enzyme copies recruited into specialized metabolism, contained in MIBiG, are labeled in blue.
Enzyme copies that are closer to blue enzyme recruitments than to red conserved enzymes are labeled in green
and represent EvoMining Hits. Extra copies with an unknown metabolic fate are shown in gray.’</p>
<figure><img src="../fig/tree.png" alt="EvoMining phylogenetic reconstruction providing evolutionary insights into the metabolic origin
and the fate of members of diverse EF from the Streptococcus example.
Seed enzymes are labeled in orange. The most conserved copies or central metabolism copies are marked in red.
Enzyme copies recruited into specialized metabolism, contained in MIBiG, are labeled in blue.
Enzyme copies that are closer to blue enzyme recruitments than to red conserved enzymes are labeled in green
and represent EvoMining Hits. Extra copies with an unknown metabolic fate are shown in gray." class="figure mx-auto d-block"></figure><hr>
<h3 id="07-evomining-figure-3">Figure 3</h3>
<p aria-hidden="true">Image 1 of 1: ‘Select the path to download file’</p>
<a href="../fig/ExportFiles1.jpg">
<img src="../fig/ExportFiles1.jpg" alt="Select the path to download file" class="figure"></a><hr>
<h3 id="07-evomining-figure-4">Figure 4</h3>
<p aria-hidden="true">Image 1 of 1: ‘Select the path to download file’</p>
<a href="fig/ExportFiles2.png">
<img src="fig/ExportFiles2.png" alt="Select the path to download file" class="figure"></a><hr>
<h3 id="07-evomining-figure-5">Figure 5</h3>
<p aria-hidden="true">Image 1 of 1: ‘MicroReact visualization of the EvoMining run Streptococcus example.
                                             	At the left a bar-chart with the EF in the X axis and the number of entries in the Y axis.
                                             	At the right, the EvoMining phylogenetic tree using the same color code as the chart.
                                             	Right of the tree the legend indicating the colors by metabolism; central metabolism enzymes in red,
                                             	expansion enzymes in gray, recruited enzymes contained in MIBiG in blue,
                                             	secondary metabolism enzymes (EvoMining hits) are marked in green,
                                             	and seed enzymes are colored in orange. Below appears the metadata from the run,
                                             	organized in a five row table including Id, metabolism, genome, function and copies.’</p>
<a href="fig/EvoMiningMicroReact.png">
<img src="fig/EvoMiningMicroReact.png" alt="MicroReact visualization of the EvoMining run Streptococcus example.
                                             	At the left a bar-chart with the EF in the X axis and the number of entries in the Y axis.
                                             	At the right, the EvoMining phylogenetic tree using the same color code as the chart.
                                             	Right of the tree the legend indicating the colors by metabolism; central metabolism enzymes in red,
                                             	expansion enzymes in gray, recruited enzymes contained in MIBiG in blue,
                                             	secondary metabolism enzymes (EvoMining hits) are marked in green,
                                             	and seed enzymes are colored in orange. Below appears the metadata from the run,
                                             	organized in a five row table including Id, metabolism, genome, function and copies." class="figure"></a><hr></section><section id="08-GatorGC"><h2 class="section-heading"><a href="08-GatorGC.html">GATOR-GC: Genomic Assessment Tool for Orthologous Regions and Gene Clusters</a></h2>
<hr class="half-width">
<h3 id="08-GatorGC-figure-1">Figure 1</h3>
<p aria-hidden="true">Image 1 of 1: ‘GATOR methods’</p>
<a href="fig/gator_gc_logo.jpg">
<img src="fig/gator_gc_logo.jpg" alt="GATOR methods" style="width: 60%; height: auto;" class="figure"></a><hr>
<h3 id="08-GatorGC-figure-2">Figure 2</h3>
<p aria-hidden="true">Image 1 of 1: ‘GATOR methods’</p>
<a href="fig/gator_gc_alogorithm.png">
<img src="fig/gator_gc_alogorithm.png" alt="GATOR methods" class="figure"></a><hr>
<h3 id="08-GatorGC-figure-3">Figure 3</h3>
<p aria-hidden="true">Image 1 of 1: ‘Heatmap resulting from cpsg gator-gc analysis’</p>
<a href="fig/heatmap.svg">
<img src="fig/heatmap.svg" alt="Heatmap resulting from cpsg gator-gc analysis" class="figure"></a><hr>
<h3 id="08-GatorGC-figure-4">Figure 4</h3>
<p aria-hidden="true">Image 1 of 1: ‘Conservation plot for the first window’</p>
<a href="fig/coservation_window_1.svg%20%20%20%20">
<img src="fig/coservation_window_1.svg" alt="Conservation plot for the first window" class="figure"></a><hr>
<h3 id="08-GatorGC-figure-5">Figure 5</h3>
<p aria-hidden="true">Image 1 of 1: ‘Fisrt window_1_neighborhood plot for the first window’</p>
<a href="fig/window_1_neighborhood.svg">
<img src="fig/window_1_neighborhood.svg" alt="Fisrt window_1_neighborhood plot for the first window" class="figure"></a><hr></section><section id="09-metabolomics"><h2 class="section-heading"><a href="09-metabolomics.html">Metabolomics workshop</a></h2>
<hr class="half-width">
<h3 id="09-metabolomics-figure-1">Figure 1</h3>
<p aria-hidden="true">Image 1 of 1: ‘GNPS output can be directly visualized in the GNPS webpage, or using other visualization tools such as [Cytoscape](https://cytoscape.org/) ’</p>
<a href="fig/Molecular-network.png">
<img src="fig/Molecular-network.png" alt="GNPS output can be directly visualized in the GNPS webpage, or using other visualization tools such as [Cytoscape](https://cytoscape.org/) " width="900" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-2">Figure 2</h3>
<p aria-hidden="true">Image 1 of 1: ‘Create an account in GNPS’</p>
<a href="fig/create_GNPSaccount.jpg">
<img src="fig/create_GNPSaccount.jpg" alt="Create an account in GNPS" width="800" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-3">Figure 3</h3>
<p aria-hidden="true">Image 1 of 1: ‘MZmine 3, an MS data analysis platform ’</p>
<a href="fig/Download_MZMine.jpg">
<img src="fig/Download_MZMine.jpg" alt="MZmine 3, an MS data analysis platform " width="500" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-4">Figure 4</h3>
<p aria-hidden="true">Image 1 of 1: ‘Data collection from *Streptomyces* sp. H-KF8, and *Streptomyces* sp. Vc74B-19.’</p>
<a href="fig/Datasets.png">
<img src="fig/Datasets.png" alt="Data collection from *Streptomyces* sp. H-KF8, and *Streptomyces* sp. Vc74B-19." width="800" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-5">Figure 5</h3>
<p aria-hidden="true">Image 1 of 1: ‘Data collection from *Streptomyces* sp. H-KF8, and *Streptomyces* sp. Vc74B-19.’</p>
<a href="fig/Metadata.jpg">
<img src="fig/Metadata.jpg" alt="Data collection from *Streptomyces* sp. H-KF8, and *Streptomyces* sp. Vc74B-19." width="800" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-6">Figure 6</h3>
<p aria-hidden="true">Image 1 of 1: ‘Load batch file’</p>
<a href="fig/batch_01.jpg">
<img src="fig/batch_01.jpg" alt="Load batch file" width="400" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-7">Figure 7</h3>
<p aria-hidden="true">Image 1 of 1: ‘Load batch file’</p>
<a href="fig/batch_02.jpg">
<img src="fig/batch_02.jpg" alt="Load batch file" width="600" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-8">Figure 8</h3>
<p aria-hidden="true">Image 1 of 1: ‘Load batch file’</p>
<a href="fig/batch_03.jpg">
<img src="fig/batch_03.jpg" alt="Load batch file" width="800" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-9">Figure 9</h3>
<p aria-hidden="true">Image 1 of 1: ‘Load batch file’</p>
<a href="fig/batch_04.jpg">
<img src="fig/batch_04.jpg" alt="Load batch file" width="800" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-10">Figure 10</h3>
<p aria-hidden="true">Image 1 of 1: ‘Load batch file’</p>
<a href="fig/batch_05.jpg">
<img src="fig/batch_05.jpg" alt="Load batch file" width="800" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-11">Figure 11</h3>
<p aria-hidden="true">Image 1 of 1: ‘Datasets’</p>
<a href="fig/MZMine_datafiles.jpg">
<img src="fig/MZMine_datafiles.jpg" alt="Datasets" width="400" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-12">Figure 12</h3>
<p aria-hidden="true">Image 1 of 1: ‘Chromatogram’</p>
<a href="fig/chromatogram_01.jpg">
<img src="fig/chromatogram_01.jpg" alt="Chromatogram" width="400" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-13">Figure 13</h3>
<p aria-hidden="true">Image 1 of 1: ‘Chromatogram’</p>
<a href="fig/chromatogram_02.jpg">
<img src="fig/chromatogram_02.jpg" alt="Chromatogram" width="500" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-14">Figure 14</h3>
<p aria-hidden="true">Image 1 of 1: ‘Chromatogram’</p>
<a href="fig/chromatogram_03.jpg">
<img src="fig/chromatogram_03.jpg" alt="Chromatogram" width="800" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-15">Figure 15</h3>
<p aria-hidden="true">Image 1 of 1: ‘Chromatogram’</p>
<a href="fig/chromatogram_04.jpg">
<img src="fig/chromatogram_04.jpg" alt="Chromatogram" width="800" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-16">Figure 16</h3>
<p aria-hidden="true">Image 1 of 1: ‘Chromatogram’</p>
<a href="fig/chromatogram_05.jpg">
<img src="fig/chromatogram_05.jpg" alt="Chromatogram" width="800" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-17">Figure 17</h3>
<p aria-hidden="true">Image 1 of 1: ‘Feature list’</p>
<a href="fig/feat_list_01.jpg">
<img src="fig/feat_list_01.jpg" alt="Feature list" width="400" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-18">Figure 18</h3>
<p aria-hidden="true">Image 1 of 1: ‘Feature List’</p>
<a href="fig/feat_list_02.jpg">
<img src="fig/feat_list_02.jpg" alt="Feature List" width="800" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-19">Figure 19</h3>
<p aria-hidden="true">Image 1 of 1: ‘Blank substraction’</p>
<a href="fig/blank_01.jpg">
<img src="fig/blank_01.jpg" alt="Blank substraction" width="600" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-20">Figure 20</h3>
<p aria-hidden="true">Image 1 of 1: ‘Blank substraction’</p>
<a href="fig/blank_02.jpg">
<img src="fig/blank_02.jpg" alt="Blank substraction" width="600" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-21">Figure 21</h3>
<p aria-hidden="true">Image 1 of 1: ‘Blank substraction’</p>
<a href="fig/blank_03.jpg">
<img src="fig/blank_03.jpg" alt="Blank substraction" width="600" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-22">Figure 22</h3>
<p aria-hidden="true">Image 1 of 1: ‘Blank substraction’</p>
<a href="fig/blank_04.jpg">
<img src="fig/blank_04.jpg" alt="Blank substraction" width="400" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-23">Figure 23</h3>
<p aria-hidden="true">Image 1 of 1: ‘Export files’</p>
<a href="fig/export_01.jpg">
<img src="fig/export_01.jpg" alt="Export files" width="600" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-24">Figure 24</h3>
<p aria-hidden="true">Image 1 of 1: ‘Export files’</p>
<a href="fig/export_02.jpg">
<img src="fig/export_02.jpg" alt="Export files" width="600" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-25">Figure 25</h3>
<p aria-hidden="true">Image 1 of 1: ‘Export files’</p>
<a href="fig/export_03.jpg">
<img src="fig/export_03.jpg" alt="Export files" width="600" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-26">Figure 26</h3>
<p aria-hidden="true">Image 1 of 1: ‘Export files’</p>
<a href="fig/export_04.jpg">
<img src="fig/export_04.jpg" alt="Export files" width="600" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-27">Figure 27</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/FBMN_01.jpg">
<img src="fig/FBMN_01.jpg" alt="FBMN" width="800" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-28">Figure 28</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/FBMN_02.jpg">
<img src="fig/FBMN_02.jpg" alt="FBMN" width="800" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-29">Figure 29</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/FBMN_03.jpg">
<img src="fig/FBMN_03.jpg" alt="FBMN" width="800" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-30">Figure 30</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/FBMN_04.jpg">
<img src="fig/FBMN_04.jpg" alt="FBMN" width="800" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-31">Figure 31</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/FBMN_05.jpg">
<img src="fig/FBMN_05.jpg" alt="FBMN" width="800" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-32">Figure 32</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/FBMN_06.jpg">
<img src="fig/FBMN_06.jpg" alt="FBMN" width="800" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-33">Figure 33</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/FBMN_07.jpg">
<img src="fig/FBMN_07.jpg" alt="FBMN" width="800" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-34">Figure 34</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/FBMN_08.jpg">
<img src="fig/FBMN_08.jpg" alt="FBMN" width="800" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-35">Figure 35</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/FBMN_09.jpg">
<img src="fig/FBMN_09.jpg" alt="FBMN" width="800" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-36">Figure 36</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/FBMN_10.jpg">
<img src="fig/FBMN_10.jpg" alt="FBMN" width="800" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-37">Figure 37</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/FBMN_11.jpg">
<img src="fig/FBMN_11.jpg" alt="FBMN" width="800" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-38">Figure 38</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/FBMN_12.jpg">
<img src="fig/FBMN_12.jpg" alt="FBMN" width="800" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-39">Figure 39</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/FBMN-STATS_01.jpg">
<img src="fig/FBMN-STATS_01.jpg" alt="FBMN" width="800" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-40">Figure 40</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/FBMN-STATS_02.jpg">
<img src="fig/FBMN-STATS_02.jpg" alt="FBMN" width="800" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-41">Figure 41</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/FBMN-STATS_03.jpg">
<img src="fig/FBMN-STATS_03.jpg" alt="FBMN" width="800" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-42">Figure 42</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/FBMN-STATS_04.jpg">
<img src="fig/FBMN-STATS_04.jpg" alt="FBMN" width="800" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-43">Figure 43</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/FBMN-STATS_05.jpg">
<img src="fig/FBMN-STATS_05.jpg" alt="FBMN" width="300" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-44">Figure 44</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/FBMN-STATS_06.jpg">
<img src="fig/FBMN-STATS_06.jpg" alt="FBMN" width="600" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-45">Figure 45</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/FBMN-STATS_07.jpg">
<img src="fig/FBMN-STATS_07.jpg" alt="FBMN" width="600" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-46">Figure 46</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/FBMN-STATS_08.jpg">
<img src="fig/FBMN-STATS_08.jpg" alt="FBMN" width="800" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-47">Figure 47</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/FBMN-STATS_09.jpg">
<img src="fig/FBMN-STATS_09.jpg" alt="FBMN" width="800" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-48">Figure 48</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/cytoscape_01.jpg">
<img src="fig/cytoscape_01.jpg" alt="FBMN" width="600" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-49">Figure 49</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/cytoscape_02.jpg">
<img src="fig/cytoscape_02.jpg" alt="FBMN" width="600" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-50">Figure 50</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/cytoscape_03.jpg">
<img src="fig/cytoscape_03.jpg" alt="FBMN" width="600" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-51">Figure 51</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/cytoscape_04.jpg">
<img src="fig/cytoscape_04.jpg" alt="FBMN" width="600" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-52">Figure 52</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/cytoscape_05.jpg">
<img src="fig/cytoscape_05.jpg" alt="FBMN" width="600" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-53">Figure 53</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/cytoscape_06.jpg">
<img src="fig/cytoscape_06.jpg" alt="FBMN" width="600" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-54">Figure 54</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/cytoscape_07.jpg">
<img src="fig/cytoscape_07.jpg" alt="FBMN" width="800" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-55">Figure 55</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/cytoscape_08.jpg">
<img src="fig/cytoscape_08.jpg" alt="FBMN" width="600" class="figure"></a><hr>
<h3 id="09-metabolomics-figure-56">Figure 56</h3>
<p aria-hidden="true">Image 1 of 1: ‘FBMN’</p>
<a href="fig/cytoscape_09.jpg">
<img src="fig/cytoscape_09.jpg" alt="FBMN" width="800" class="figure"></a><hr></section><section id="10-OtherResources"><h2 class="section-heading"><a href="10-OtherResources.html">Other Resources</a></h2>
<hr class="half-width"></section>
</div>
    </main>
</div>
<!-- END  : inst/pkgdown/templates/content-extra.html -->

      </div>
<!--/div.row-->
      		<footer class="row footer mx-md-3"><hr>
<div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://github.com/carpentries-incubator/genome-mining/edit/main/README.md" class="external-link">Edit on GitHub</a>

	
        | <a href="https://github.com/carpentries-incubator/genome-mining/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/carpentries-incubator/genome-mining/" class="external-link">Source</a></p>
        <p>
        <a href="https://github.com/carpentries-incubator/genome-mining/blob/main/" class="external-link">Cite</a>
        | <a href="mailto:axel.ramos3737@gmail.com">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
		</div>
		<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.18.5" class="external-link">sandpaper (0.18.5)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.9" class="external-link">pegboard (0.7.9)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.9" class="external-link">varnish (1.0.9)</a></p>
		</div>
		</footer>
</div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "LearningResource",
  "@id": "https://carpentries-incubator.github.io/genome-mining/images.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/LearningResource/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "software, data, lesson, The Carpentries",
  "name": "All Images",
  "creativeWorkStatus": "active",
  "url": "https://carpentries-incubator.github.io/genome-mining/images.html",
  "identifier": "https://carpentries-incubator.github.io/genome-mining/images.html",
  "dateCreated": "2021-06-20",
  "dateModified": "2026-02-20",
  "datePublished": "2026-02-20"
}

  </script><script>
		feather.replace();
	</script><!-- Matomo --><script>
          var _paq = window._paq = window._paq || [];
          /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
          _paq.push(["setDocumentTitle", document.domain + "/" + document.title]);
          _paq.push(["setDomains", ["*.lessons.carpentries.org","*.datacarpentry.github.io","*.datacarpentry.org","*.librarycarpentry.github.io","*.librarycarpentry.org","*.swcarpentry.github.io", "*.carpentries.github.io"]]);
          _paq.push(["setDoNotTrack", true]);
          _paq.push(["disableCookies"]);
          _paq.push(["trackPageView"]);
          _paq.push(["enableLinkTracking"]);
          (function() {
              var u="https://matomo.carpentries.org/";
              _paq.push(["setTrackerUrl", u+"matomo.php"]);
              _paq.push(["setSiteId", "1"]);
              var d=document, g=d.createElement("script"), s=d.getElementsByTagName("script")[0];
              g.async=true; g.src="https://matomo.carpentries.org/matomo.js"; s.parentNode.insertBefore(g,s);
          })();
        </script><!-- End Matomo Code -->
</body>
</html><!-- END:   inst/pkgdown/templates/layout.html-->

