<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Genome Mining in Prokaryotes: BGC Similarity Networks</title><meta name="viewport" content="width=device-width, initial-scale=1"><script src="../assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="../assets/styles.css"><script src="../assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../favicons/incubator/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicons/incubator/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="../favicons/incubator/favicon-16x16.png"><link rel="manifest" href="../favicons/incubator/site.webmanifest"><link rel="mask-icon" href="../favicons/incubator/safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" media="(prefers-color-scheme: light)" content="white"><meta name="theme-color" media="(prefers-color-scheme: dark)" content="black"></head><body>
    <header id="top" class="navbar navbar-expand-md top-nav incubator"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="incubator-logo" alt="Lesson Description" src="../assets/images/incubator-logo.svg"><span class="badge text-bg-danger">
          <abbr title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.">
            <a href="https://docs.carpentries.org/resources/curriculum/lesson-life-cycle.html" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
              Pre-Alpha
            </a>
            <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text"><li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul></li>
      </div>


      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='../04-bigscape.html';">Learner View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Lesson Description" src="../assets/images/incubator-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Genome Mining in Prokaryotes
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Genome Mining in Prokaryotes
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><li><a class="dropdown-item" href="data.html">Data</a></li><hr><li><a class="dropdown-item" href="discuss.html">Discussion</a></li><li><a class="dropdown-item" href="reference.html">Glossary</a></li>
          </ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="../instructor/aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Genome Mining in Prokaryotes
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 20%" class="percentage">
    20%
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: 20%" aria-valuenow="20" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text"><li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul></li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../04-bigscape.html">Learner View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->

            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="01-introduction.html">1. Introduction to Genome Mining</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="02-antismash.html">2. Secondary metabolite biosynthetic gene cluster identification</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="03-antiSMASHdb-MiBiG.html">3. Genome Mining Databases</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
      <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapsecurrent" aria-expanded="true" aria-controls="flush-collapsecurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        4. BGC Similarity Networks
        </span>
      </button>
    </div><!--/div.accordion-header-->

    <div id="flush-collapsecurrent" class="accordion-collapse collapse show" aria-labelledby="flush-headingcurrent" data-bs-parent="#accordionFlushcurrent">
      <div class="accordion-body">
        <ul><li><a href="#introduction">Introduction</a></li>
<li><a href="#preparing-the-input">Preparing the input</a></li>
<li><a href="#running-big-scape">Running BiG-SCAPE</a></li>
<li><a href="#viewing-the-results">Viewing the results</a></li>
<li><a href="#cytoscape-visualization-of-the-results">Cytoscape visualization of the results</a></li>
        </ul></div><!--/div.accordion-body-->
    </div><!--/div.accordion-collapse-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="05-bigslice-bigfam.html">5. Homologous BGC Clusterization</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="06-corason.html">6. Finding Variation on Genomic Vicinities</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="07-evomining.html">7. Evolutionary Genome Mining</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="08-GatorGC.html">8. GATOR-GC: Genomic Assessment Tool for Orthologous Regions and Gene Clusters</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush10">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading10">
        <a href="09-metabolomics.html">9. Metabolomics workshop</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush11">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading11">
        <a href="10-OtherResources.html">10. Other Resources</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <li><a href="data.html">Data</a></li><hr><li><a class="dropdown-item" href="discuss.html">Discussion</a></li><li><a class="dropdown-item" href="reference.html">Glossary</a></li>
                    </ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources"><a href="../instructor/aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">

            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/03-antiSMASHdb-MiBiG.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/05-bigslice-bigfam.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/03-antiSMASHdb-MiBiG.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Genome Mining
        </a>
        <a class="chapter-link float-end" href="../instructor/05-bigslice-bigfam.html" rel="next">
          Next: Homologous BGC...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>BGC Similarity Networks</h1>
        <p>Last updated on 2026-02-20 |

        <a href="https://github.com/carpentries-incubator/genome-mining/edit/main/episodes/04-bigscape.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>



        <p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 45 minutes</p>

        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>How can I measure similarity between BGCs?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>Understand how BiG-SCAPE measures similarity between BGCs.</li>
<li>Take an antiSMASH output to perform a BiG-SCAPE analysis.</li>
<li>Interpret BiG-SCAPE similarity networks and GCF phylogenies.</li>
</ul></div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a></h2>
<hr class="half-width"><p>In the previous section, we learned how to study the BGCs encoded by
each of the genomes of our analyses. In case you are interested in the
study of a certain BGC or a certain strain, this may be enough. However,
sometimes the aim is to compare the biosynthetic potential of tens or
hundreds of genomes. To perform this kind of analysis, we will use
BiG-SCAPE (Navarro-Muñoz et al., 2019), a workflow that compares all the
BGCs detected by antiSMASH to find their relatedness. BiG-SCAPE will
search for Pfam domains (Mistry et al., 2021) in the protein sequences
of each BGC. Then, the Pfam domains will be linearized and compared,
creating different similarity networks and scoring the similarity of
each pair of clusters. Based on a cutoff value for this score, the
diverse BGCs will be classified on Gene Cluster Families (GCFs) to
facilitate their study. A single GCF is supposed to encompass BGCs that
produce chemically related metabolites (molecular families). Lower
cutoffs would create families of BGCs that produce identical compounds,
while higher cutoffs would create families of more loosely related
compounds.</p>
</section><section><h2 class="section-heading" id="preparing-the-input">Preparing the input<a class="anchor" aria-label="anchor" href="#preparing-the-input"></a></h2>
<hr class="half-width"><p>In each of the antiSMASH output directories, we will find a single
<code>.gbk</code> file for each BGC, which includes “region” within its
filename. Thus, we will copy all these files to the new directory.</p>
<p>Move into the directory that has the antiSMASH results of each
genome:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">$</span> conda deactivate  </span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="ex">$</span> conda activate /miniconda3/envs/bigscape/  </span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="ex">$</span> cd ~/pan_workshop/results/antismash/</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="ex">$</span> ls</span></code></pre>
</div>
<p>Since we will put together in the same directory many files with
similar names, we want to make sure that nothing is left behind or
overwritten. For this, we will count all the <code>gbk</code> files of
all the genomes.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="ex">$</span> ls Streptococcus_agalactiae_<span class="pp">*</span>/<span class="pp">*</span>region<span class="pp">*</span>gbk <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code><span><span class="fl">16</span></span></code></pre>
</div>
<p>Since the names are somewhat cryptic, they could be repeated, so we
will rename the <code>gbks</code> in such a way that they include the
genome name.</p>
<p>Copy the following script to a file named
<code>change-names.sh</code> using <code>nano</code>:</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="co"># This script is to rename the antiSMASH gbks for them to include the species and strain names, taken from the directory name.</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="co"># The argument it requires is the name of the directory with the AntiSMASH output, which must NOT contain a slash at the end.</span></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="co"># Usage for one AntiSMASH output directory:</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="co">#     sh change-names.sh &lt;folder&gt;</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a><span class="co"># Usage for multiple AntiSMASH output directory:</span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a><span class="co">#     for species in &lt;output-directory-pattern*&gt;</span></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a><span class="co">#    	 do</span></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a><span class="co">#    		 sh change-names.sh $species</span></span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a><span class="co">#    	 done</span></span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a></span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a></span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a></span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-1</span> <span class="st">"</span><span class="va">$1</span><span class="st">"</span>/<span class="pp">*</span>region<span class="pp">*</span>gbk <span class="kw">|</span> <span class="cf">while</span> <span class="bu">read</span> <span class="va">line</span> <span class="co"># enlist the gbks of all regions in the directory and start a while loop</span></span>
<span id="cb4-16"><a href="#cb4-16" tabindex="-1"></a> <span class="cf">do</span></span>
<span id="cb4-17"><a href="#cb4-17" tabindex="-1"></a>    <span class="va">dir</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">$line</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-d</span><span class="st">'/'</span> <span class="at">-f1</span><span class="va">)</span> <span class="co"># save the directory name in a variable</span></span>
<span id="cb4-18"><a href="#cb4-18" tabindex="-1"></a>    <span class="va">file</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">$line</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-d</span><span class="st">'/'</span> <span class="at">-f2</span><span class="va">)</span> <span class="co"># save the file name in a variable</span></span>
<span id="cb4-19"><a href="#cb4-19" tabindex="-1"></a>	<span class="cf">for</span> directory <span class="kw">in</span> <span class="va">$dir</span> <span class="co"># start a for loop</span></span>
<span id="cb4-20"><a href="#cb4-20" tabindex="-1"></a>    	<span class="cf">do</span></span>
<span id="cb4-21"><a href="#cb4-21" tabindex="-1"></a>   		 <span class="bu">cd</span> <span class="va">$directory</span> <span class="co"># enter the directory</span></span>
<span id="cb4-22"><a href="#cb4-22" tabindex="-1"></a>      	<span class="va">newfile</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">$dir</span>-<span class="va">$file)</span> <span class="co"># make a new variable that fuses the directory name with the file name</span></span>
<span id="cb4-23"><a href="#cb4-23" tabindex="-1"></a>			 	<span class="bu">echo</span> <span class="st">"Renaming"</span> <span class="va">$file</span> <span class="st">" to"</span> <span class="va">$newfile</span> <span class="co"># print a message showing the old and new file names</span></span>
<span id="cb4-24"><a href="#cb4-24" tabindex="-1"></a>			 	<span class="fu">mv</span> <span class="va">$file</span> <span class="va">$newfile</span> <span class="co"># rename</span></span>
<span id="cb4-25"><a href="#cb4-25" tabindex="-1"></a>			 	<span class="bu">cd</span> .. <span class="co"># return to main directory before it begins again</span></span>
<span id="cb4-26"><a href="#cb4-26" tabindex="-1"></a>		 <span class="cf">done</span></span>
<span id="cb4-27"><a href="#cb4-27" tabindex="-1"></a> <span class="cf">done</span></span></code></pre>
</div>
<p>Run the script for all the directory:</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="ex">$</span> for species in Streptococcus_agalactiae_<span class="pp">*</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="op">&gt;</span> do</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="op">&gt;</span>   	 sh <span class="ex">change-names.sh</span> <span class="va">$species</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="op">&gt;</span>    done</span></code></pre>
</div>
<p>Now make a directory for all your BiG-SCAPE analyses and inside it
make a directory that contains all of the <code>gbks</code> of all of
your genomes. This one will be the input for BiG-SCAPE. (For convenience
<code>bigscape/</code> will be inside <code>antismash/</code> while we
run BiG-SCAPE).</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="ex">$</span> mkdir <span class="at">-p</span> bigscape/bgcs_gbks/</span></code></pre>
</div>
<p>Now copy all the region <code>gbks</code>to this new directory, and
look at its content:</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="ex">$</span> cp Streptococcus_agalactiae_<span class="pp">*</span>/<span class="pp">*</span>region<span class="pp">*</span>gbk bigscape/bgcs_gbks/</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="ex">$</span> ls bigscape/bgcs_gbks/</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code><span><span class="va">Streptococcus_agalactiae_18RS21_prokka</span><span class="op">-</span><span class="va">AAJO01000016.1.region001.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_18RS21_prokka</span><span class="op">-</span><span class="va">AAJO01000043.1.region001.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_2603V_prokka</span><span class="op">-</span><span class="va">NC_004116.1.region001.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_2603V_prokka</span><span class="op">-</span><span class="va">NC_004116.1.region002.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_515_prokka</span><span class="op">-</span><span class="va">NZ_CP051004.1.region001.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_515_prokka</span><span class="op">-</span><span class="va">NZ_CP051004.1.region002.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_A909_prokka</span><span class="op">-</span><span class="va">NC_007432.1.region001.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_A909_prokka</span><span class="op">-</span><span class="va">NC_007432.1.region002.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_CJB111_prokka</span><span class="op">-</span><span class="va">NZ_AAJQ01000010.region001.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_CJB111_prokka</span><span class="op">-</span><span class="va">NZ_AAJQ01000025.region001.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_COH1_prokka</span><span class="op">-</span><span class="va">NZ_HG939456.1.region001.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_COH1_prokka</span><span class="op">-</span><span class="va">NZ_HG939456.1.region002.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_H36B_prokka</span><span class="op">-</span><span class="va">AAJS01000020.1.region001.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_H36B_prokka</span><span class="op">-</span><span class="va">AAJS01000117.1.region001.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_NEM316_prokka</span><span class="op">-</span><span class="va">NC_004368.1.region001.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_NEM316_prokka</span><span class="op">-</span><span class="va">NC_004368.1.region002.gbk</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="running-big-scape">Running BiG-SCAPE<a class="anchor" aria-label="anchor" href="#running-big-scape"></a></h2>
<hr class="half-width"><p>BiG-SCAPE can be executed in different ways, depending on the
installation mode that you applied. You could call the program through
<code>bigscape</code>, <code>run_bigscape</code>or
<code>run_bigscape.py</code>. Here, based on our installation (see <a href="index.html#setup">Setup</a>), we will use <code>bigscape</code>.</p>
<p>The options that we will use are described in the help page:</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="ex">-i</span> INPUTDIR, <span class="at">--inputdir</span> INPUTDIR</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>                    	<span class="ex">Input</span> directory of gbk files, if left empty, all gbk</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>                    	<span class="ex">files</span> in current and lower directories will be used.</span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a></span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a><span class="ex">-o</span> OUTPUTDIR, <span class="at">--outputdir</span> OUTPUTDIR</span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a>                    	<span class="ex">Output</span> directory, this will contain all output data</span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a>                    	<span class="ex">files.</span></span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a></span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a><span class="ex">--mix</span>             	By default, BiG-SCAPE separates the analysis according</span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a>                   	<span class="ex">to</span> the BGC product <span class="er">(</span><span class="ex">PKS</span> Type I, NRPS, RiPPs, etc.<span class="kw">)</span> <span class="ex">and</span></span>
<span id="cb9-11"><a href="#cb9-11" tabindex="-1"></a>                   	<span class="ex">will</span> create network directories for each class. Toggle</span>
<span id="cb9-12"><a href="#cb9-12" tabindex="-1"></a>                   	<span class="ex">to</span> include an analysis mixing all classes</span>
<span id="cb9-13"><a href="#cb9-13" tabindex="-1"></a>                  	 </span>
<span id="cb9-14"><a href="#cb9-14" tabindex="-1"></a><span class="ex">--hybrids-off</span>     	Toggle to also add BGCs with hybrid predicted products</span>
<span id="cb9-15"><a href="#cb9-15" tabindex="-1"></a>                    	<span class="ex">from</span> the PKS/NRPS Hybrids and Others classes to each</span>
<span id="cb9-16"><a href="#cb9-16" tabindex="-1"></a>                    	<span class="ex">subclass</span> <span class="er">(</span><span class="ex">e.g.</span> a <span class="st">'terpene-nrps'</span> BGC from Others would</span>
<span id="cb9-17"><a href="#cb9-17" tabindex="-1"></a>                    	<span class="ex">be</span> added to the Terpene and NRPS classes<span class="kw">)</span></span>
<span id="cb9-18"><a href="#cb9-18" tabindex="-1"></a></span>
<span id="cb9-19"><a href="#cb9-19" tabindex="-1"></a><span class="ex">--mode</span> <span class="dt">{global</span><span class="op">,</span><span class="dt">glocal</span><span class="op">,</span><span class="dt">auto}</span></span>
<span id="cb9-20"><a href="#cb9-20" tabindex="-1"></a>                    	<span class="ex">Alignment</span> mode for each pair of gene clusters.</span>
<span id="cb9-21"><a href="#cb9-21" tabindex="-1"></a>                    	<span class="st">'global'</span><span class="ex">:</span> the whole list of domains of each BGC are</span>
<span id="cb9-22"><a href="#cb9-22" tabindex="-1"></a>                    	<span class="ex">compared</span><span class="kw">;</span> <span class="st">'glocal'</span><span class="ex">:</span> Longest Common Subcluster mode.</span>
<span id="cb9-23"><a href="#cb9-23" tabindex="-1"></a>                    	<span class="ex">Redefine</span> the subset of the domains used to calculate</span>
<span id="cb9-24"><a href="#cb9-24" tabindex="-1"></a>                    	<span class="ex">distance</span> by trying to find the longest slice of common</span>
<span id="cb9-25"><a href="#cb9-25" tabindex="-1"></a>                    	<span class="ex">domain</span> content per gene in both BGCs, then expand each</span>
<span id="cb9-26"><a href="#cb9-26" tabindex="-1"></a>                    	<span class="ex">slice.</span> <span class="st">'auto'</span>: use glocal when at least one of the</span>
<span id="cb9-27"><a href="#cb9-27" tabindex="-1"></a>                    	<span class="ex">BGCs</span> in each pair has the <span class="st">'contig_edge'</span> annotation</span>
<span id="cb9-28"><a href="#cb9-28" tabindex="-1"></a>                    	<span class="ex">from</span> antiSMASH v4+, otherwise use global mode on that</span>
<span id="cb9-29"><a href="#cb9-29" tabindex="-1"></a>                    	<span class="ex">pair</span></span></code></pre>
</div>
<p>We will use the option <code>--mix</code> to have an analysis of all
of the BGCs together besides the analyses of the BGCs separated by
class. The <code>--hybrids-off</code> option will prevent from having
the same BGC twice (in the case of hybrid BGCs that could belong to two
classes) in our results. And since none of the BGCs is on a contig edge,
we could use the global mode. However, frequently, when analyzing draft
genomes, this is not the case. Thus, the auto mode will be the most
appropriate, which will use the global mode to align domains except for
those cases in which the BGC is located near to a contig end, for which
the glocal mode is automatically selected. For this run we will use only
the default cutoff value (0.3).</p>
<p>Now we are ready to run BiG-SCAPE:</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="ex">$</span> bigscape <span class="at">-i</span> bigscape/bgcs_gbks/ <span class="at">-o</span> bigscape/output_220124 <span class="at">--mix</span> <span class="at">--hybrids-off</span> <span class="at">--mode</span> auto <span class="at">--pfam_dir</span> /miniconda3/envs/bigscape/BiG-SCAPE-1.1.5/</span></code></pre>
</div>
<div id="output-names" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<span class="callout-header">Callout</span>
<div id="output-names" class="callout-inner">
<h3 class="callout-title">Output names</h3>
<div class="callout-content">
<p>Most of the times you will need to re-run a software once you have
explored the results; maybe you want to change parameters or add more
samples. For this reason, you will need a flexible way to organize your
project, including the names of the outputs. Usually, a good idea is to
put the date in the name of the folder and keep track of the analyses
you are running in your research notes.</p>
</div>
</div>
</div>
<p>Once the process is finished, you will find in your terminal screen
some basic results, such as the number of BGCs included in each type of
network. In the output folder you will find all the data.<br>
Since BiG-SCAPE does not produce a file with the run information, it is
useful to copy the text printed on the screen to a file. Use
<code>nano</code> to paste all the text that BiG-SCAPE generated in a
file named <code>bigscape.log</code> inside the folder
<code>output_100722/logs/</code>.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="ex">$</span> nano bigscape/output_220124/logs/bigscape.log</span></code></pre>
</div>
<p>Take a look at the BiG-SCAPE outputs:</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="ex">$</span> ls <span class="at">-F</span> bigscape/output_220124/</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>cache/  html_content/  index.html*  logs/  network_files/  SVG/</code></pre>
</div>
<p>To keep ordered your directories, move your <code>bigscape/</code>
directory to <code>results/</code>:</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="ex">$</span> mv bigscape/ ../</span></code></pre>
</div>
<p>And move the file <code>change-names</code> to folder
<code>scripts/</code> in the <code>pan_workshop/</code>directory to save
our little script:</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="ex">$</span> mv change-names.sh ~/pan_workshop/scripts/</span></code></pre>
</div>
</section><section><h2 class="section-heading" id="viewing-the-results">Viewing the results<a class="anchor" aria-label="anchor" href="#viewing-the-results"></a></h2>
<hr class="half-width"><p>An easy way to prospect your results is by opening the
<code>index.html</code> file with a browser (Firefox, Chrome, Safari,
etc.). In order to do this, you need to go to your local machine and
locate in the folder where you want to store your BiG-SCAPE results. Now
we compress the folder <code>output_100622/</code> and download it to
your local machine.</p>
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="ex">$</span> cd ~/pan_workshop/results/bigscape/</span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a><span class="ex">$</span> zip <span class="at">-r</span> output_220124.zip output_220124</span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a><span class="ex">$</span> ls</span></code></pre>
</div>
<p>In the JupyterHub we navigate to the folders and go to
<code>/pan_workshop/results/bigscape</code> and select the one we just
created <code>output_220124.zip</code> and download it.</p>
<p>On your local computer we unzip the files and open
<code>output_220124/index.html</code> with a browser:</p>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a><span class="ex">$</span> firefox output_100622/index.html <span class="co"># If you have Firefox available, otherwise open it using the GUI.</span></span></code></pre>
</div>
<p>There are diverse sections in the visualization. The following image
shows the overview page. At the left there is the run information and
there are pie chart representations of the number of BGCs per genome and
per class. At the right there is the Network Overview for each of the
BGC classes found and the mix category. They show the Number of
Families, Average number of BGCs per family, Max number of BGCs in a
family and the Families with MIBiG Reference BGCs. You can click on the
name of the class to see its Network Overview.</p>
<p><a href="../fig/02_05_01.png">
<img src="../fig/02_05_01.png" alt="BIG-SCAPE output as visualized in the web page.
                                  	The overview page is displayed. At the left is the Run Information,
                                  	indicating the date and time at which the analysis was started and completed, as well as the parameters of the run.
                                  	Next is displayed the Input Data, specifying the total number of genomes and the total BGCs,
                                  	in this example 8 and 23, respectively. There are two pie charts, one representing the BGC per Genome
                                  	and the other the BGC per Class. At the right is the Network Overview,
                                  	which allows selection between mix and the different BGC classes.
                                  	From the mix overview it displays the Number of Families, Average number of BGCs per family,
                                  	Max number of BGCs in a family and the Families with MIBiG Reference BGCs." class="figure"></a></p>
<p>Below, there is a clustered heatmap of the presence/absence of the
GCFs in each genome for each class. You can customize this heatmap and
select the clustering methods, or the number of GCFs represented.</p>
<p><a href="../fig/02_05_02.png">
<img src="../fig/02_05_02.png" alt="BIG-SCAPE output as visualized in the web page.
                                  	The overview page displaying the clustered heatmap of the presence/absence of the GCFs,
                                  	each class organized as a column at the base of the table, in each genome,
                                  	which are organized as rows at the right side of the table. Presence is depicted in red with 1 and absence in beige with 0." class="figure"></a></p>
<p>Clicking any of the class names of the upper left bar displays a
similarity network of BGCs. It may take some time to load the network. A
single network is represented for each GCF of each Gene Cluster Clan
(GCC), which may comprise several GCFs. Each dot represents a BGC. These
dots with bold circles are already described BGCs that have been
recruited from the MiBIG database (Kautsar et al., 2020) because of its
similarity with some BGC of the analysis.</p>
<p><a href="../fig/02_05_03.png">
<img src="../fig/02_05_03.png" alt="BIG-SCAPE similarity network of the complete mix of BGCs
                                  	obtained from the run. A network is represented for each GCF, each dot represents a BGC.
                                  	In this example there are a total of 23 BGCs, of which 6 are singletons, there are 28 links and 11 families.
                                  	Clicking over any of the dots shows the GCF at the right side and allows you to explore it further." class="figure"></a></p>
<p>When you click on a BGC (dot), it appears its GCF at the right. You
can click on the GCF name to see the phylogenetic distances among all
the BGCs comprised by a single GCF. Within the tree there is an arrow
diagram of the genes in the BGCs and the protein domains in the genes.
These trees are useful to prioritize the search of secondary
metabolites, for example, by focusing on the most divergent BGC clade or
those that are distant to already described BGCs.</p>
<p><a href="../fig/02_05_04.png">
<img src="../fig/02_05_04.png" alt="BIG-SCAPE output displaying a tree of phylogenetic distances
                                  	among the BGCs comprised in a GCF. The example shows the GCF 10, comprised by six members.
                                  	Each BGC is represented in the tree by an arrow diagram of the genes and the protein domains in the genes
                                  	corresponding to that cluster." class="figure"></a></p>
<div id="discussion-1-reading-the-gcf-networks" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="discussion-1-reading-the-gcf-networks" class="callout-inner">
<h3 class="callout-title">Discussion 1: Reading the GCF networks</h3>
<div class="callout-content">
<p>What can you conclude about the diversity of BGCs between <em>S.
agalactiae</em> and <em>S. thermophilus</em>? Are they equally diverse?
Do they share GCFs?</p>
<p>Digging deeper: Why do you think the strain <em>S. agalactiae</em>
H36B has a BGC that is not part of the other GCF in the same class?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p><em>S. agalactiae</em> and <em>S. thermophilus</em> seem to have no
diversity in common because there is no GCF with members from both
species. All strains of <em>S.agalactiae</em> seem to have a very
similar set of BGCs. While <em>S. thermophilus</em> has several BGCs
that are not related to any other BGC.</p>
<p>Digging deeper: The BGC of the class PKSother in <em>S.
agalactiae</em> H3B6 has only 3 genes, which are also present in the
BGCs of the other GCF in this class. It is possible that these genes are
in a fragmented part of the assembled genome, but are actually part of a
BGC that is similar to the other ones. And it is also possible that
these 3 genes are present in the genome without being part of a bigger
BGC.</p>
</div>
</div>
</div>
</div>
<div id="exercise-1-using-the-text-output" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="exercise-1-using-the-text-output" class="callout-inner">
<h3 class="callout-title">Exercise 1: Using the text output</h3>
<div class="callout-content">
<p>Use one of the commands from the following list to make a reduced
version of the <code>Network_Annotations_full.tsv</code>, it should only
contain the information of the type of product, the BGC class of each
BGC, and its name. And save it in a file.</p>
<p><code>grep</code> <code>cut</code> <code>ls</code> <code>cat</code>
<code>mv</code></p>
<p><strong>Tip</strong>: The file is inside
<code>network_files/</code>.</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p>Take a look at the content of the file:</p>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="ex">$</span> head <span class="at">-n</span> 3 network_files/2022-06-10_21-27-26_auto/Network_Annotations_Full.tsv</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>BGC    Accession ID    Description    Product Prediction    BiG-SCAPE class    Organism    Taxonomy
Streptococcus_agalactiae_18RS21-AAJO01000016.1.region001    AAJO01000016.1    Streptococcus agalactiae
18RS21    arylpolyene    Others    Streptococcus agalactiae 18RS21    Bacteria,Terrabacteria
group,Firmicutes,Bacilli,Lactobacillales,Streptococcaceae,Streptococcus,Streptococcus agalactiae
Streptococcus_agalactiae_18RS21-AAJO01000043.1.region001    AAJO01000043.1    Streptococcus agalactiae
18RS21    T3PKS    PKSother    Streptococcus agalactiae 18RS21    Bacteria,Terrabacteria
group,Firmicutes,Bacilli,Lactobacillales,Streptococcaceae,Streptococcus,Streptococcus agalactiae</code></pre>
</div>
<p>This table is difficult to read because of the amount of information
it has. The first line has the names of the columns; <code>BGC</code>,
<code>Product Prediction</code> and <code>BiG-SCAPE class</code> are the
ones we are interested in. So we will extract those.</p>
<div class="codewrapper sourceCode" id="cb20">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a><span class="fu">cut</span> <span class="at">-f</span> 1,4,5 network_files/2022-06-10_21-27-26_auto/Network_Annotations_Full.tsv <span class="op">&gt;</span> type_of_BGC.tsv</span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a><span class="fu">cat</span> type_of_BGCs.tsv</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>BGC    Product Prediction    BiG-SCAPE class
Streptococcus_agalactiae_18RS21-AAJO01000016.1.region001    arylpolyene    Others
Streptococcus_agalactiae_18RS21-AAJO01000043.1.region001    T3PKS    PKSother
Streptococcus_agalactiae_515-AAJP01000027.1.region001    arylpolyene    Others
Streptococcus_agalactiae_515-AAJP01000037.1.region001    T3PKS    PKSother
.
.
.</code></pre>
</div>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="cytoscape-visualization-of-the-results">Cytoscape visualization of the results<a class="anchor" aria-label="anchor" href="#cytoscape-visualization-of-the-results"></a></h2>
<hr class="half-width"><p>You can also customize and re-renderize the similarity networks of
your results with Cytoscape (<a href="https://cytoscape.org/" class="external-link uri">https://cytoscape.org/</a>). To do so, you will need some
files included in the output directory of BiG-SCAPE. Both are located in
the same folder. You can choose any folder included in the
Network_files/(date)hybrids_auto/ directory, depending on your interest.
The “Mix/” folder represents the complete network, including all the
BGCs of the analysis. There, you will need the files “mix_c0.30.network”
and “mix_clans_0.30_0.70.tsv”. When you upload the “.network” file, it
is required that you select as “source” the first column and as “target”
the second one. Then, you can upload the “.tsv” and just select the
first column as “source”. Finally, you need to click on Tools -&gt;
merge -&gt; Networks -&gt; Union to combine both GCFs and singletons.
Now you can change colors, labels, etc. according to your specific
requirements.</p>
<div class="section level3">
<h3 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h3>
<ul><li><p>Navarro-Muñoz, J.C., Selem-Mojica, N., Mullowney, M.W. et al. “A
computational framework to explore large-scale biosynthetic diversity”.
Nature Chemical Biology (2019).</p></li>
<li><p>Mistry, J., Chuguransky, S., Williams, L., Qureshi, M., Salazar,
G. A., Sonnhammer, E. L., … &amp; Bateman, A. (2021). Pfam: The protein
families database in 2021. Nucleic Acids Research, 49(D1),
D412-D419.</p></li>
<li><p>Kautsar, S. A., Blin, K., Shaw, S., Navarro-Muñoz, J. C.,
Terlouw, B. R., van der Hooft, J. J., … &amp; Medema, M. H. (2020).
MIBiG 2.0: a repository for biosynthetic gene clusters of known
function. Nucleic acids research, 48(D1), D454-D458.</p></li>
</ul><div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<ul><li>BGC similarity is measured by BiG-SCAPE according to protein domain
content, adjacency and sequence identity.</li>
<li>The <code>gbks</code> of the regions identified by antiSMASH are the
input for BiG-SCAPE.</li>
<li>BiG-SCAPE delivers BGCs similarity networks with which it delimits
Gene Cluster Families and creates a phylogeny of the BGCs in each
GCF.</li>
</ul></div>
</div>
</div>
</div>
</section></div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/03-antiSMASHdb-MiBiG.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/05-bigslice-bigfam.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/03-antiSMASHdb-MiBiG.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Genome Mining
        </a>
        <a class="chapter-link float-end" href="../instructor/05-bigslice-bigfam.html" rel="next">
          Next: Homologous BGC...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://github.com/carpentries-incubator/genome-mining/edit/main/episodes/04-bigscape.md" class="external-link">Edit on GitHub</a>

	
        | <a href="https://github.com/carpentries-incubator/genome-mining/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/carpentries-incubator/genome-mining/" class="external-link">Source</a></p>
        <p>
        <a href="https://github.com/carpentries-incubator/genome-mining/blob/main/" class="external-link">Cite</a>
        | <a href="mailto:axel.ramos3737@gmail.com">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
		</div>
		<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.18.5" class="external-link">sandpaper (0.18.5)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.9" class="external-link">pegboard (0.7.9)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.9" class="external-link">varnish (1.0.9)</a></p>
		</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "LearningResource",
  "@id": "https://carpentries-incubator.github.io/genome-mining/instructor/04-bigscape.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/LearningResource/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "software, data, lesson, The Carpentries",
  "name": "BGC Similarity Networks",
  "creativeWorkStatus": "active",
  "url": "https://carpentries-incubator.github.io/genome-mining/instructor/04-bigscape.html",
  "identifier": "https://carpentries-incubator.github.io/genome-mining/instructor/04-bigscape.html",
  "dateCreated": "2021-06-20",
  "dateModified": "2026-02-20",
  "datePublished": "2026-02-20"
}

  </script><script>
		feather.replace();
	</script><!-- Matomo --><script>
          var _paq = window._paq = window._paq || [];
          /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
          _paq.push(["setDocumentTitle", document.domain + "/" + document.title]);
          _paq.push(["setDomains", ["*.lessons.carpentries.org","*.datacarpentry.github.io","*.datacarpentry.org","*.librarycarpentry.github.io","*.librarycarpentry.org","*.swcarpentry.github.io", "*.carpentries.github.io"]]);
          _paq.push(["setDoNotTrack", true]);
          _paq.push(["disableCookies"]);
          _paq.push(["trackPageView"]);
          _paq.push(["enableLinkTracking"]);
          (function() {
              var u="https://matomo.carpentries.org/";
              _paq.push(["setTrackerUrl", u+"matomo.php"]);
              _paq.push(["setSiteId", "1"]);
              var d=document, g=d.createElement("script"), s=d.getElementsByTagName("script")[0];
              g.async=true; g.src="https://matomo.carpentries.org/matomo.js"; s.parentNode.insertBefore(g,s);
          })();
        </script><!-- End Matomo Code --></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

