<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<title>Genome Mining in Prokaryotes: All in One View</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<script src="../assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="../assets/styles.css">
<script src="../assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../favicons/incubator/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicons/incubator/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="../favicons/incubator/favicon-16x16.png">
<link rel="manifest" href="../favicons/incubator/site.webmanifest">
<link rel="mask-icon" href="../favicons/incubator/safari-pinned-tab.svg" color="#5bbad5">
<meta name="msapplication-TileColor" content="#da532c">
<meta name="theme-color" media="(prefers-color-scheme: light)" content="white">
<meta name="theme-color" media="(prefers-color-scheme: dark)" content="black">
</head>
<body>
    <header id="top" class="navbar navbar-expand-md top-nav incubator"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="incubator-logo" alt="Lesson Description" src="../assets/images/incubator-logo.svg"><span class="badge text-bg-danger">
          <abbr title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.">
            <a href="https://docs.carpentries.org/resources/curriculum/lesson-life-cycle.html" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
              Pre-Alpha
            </a>
            <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text">
<li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul>
</li>
      </div>


      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1">
<li><button class="dropdown-item" type="button" onclick="window.location.href='../aio.html';">Learner View</button></li>
        </ul>
</div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Lesson Description" src="../assets/images/incubator-logo-sm.svg">
</div>
    <div class="lesson-title-md">
      Genome Mining in Prokaryotes
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0">
<li class="nav-item">
          <span class="lesson-title">
            Genome Mining in Prokaryotes
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown">
<li><a class="dropdown-item" href="data.html">Data</a></li>
<hr>
<li><a class="dropdown-item" href="discuss.html">Discussion</a></li>
<li><a class="dropdown-item" href="reference.html">Glossary</a></li>
          </ul>
</li>
      </ul>
</div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="../instructor/aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div>
<!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Genome Mining in Prokaryotes
</div>

<aside class="col-md-12 lesson-progress"><div style="width: %" class="percentage">
    %
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: %" aria-valuenow="" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text">
<li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul>
</li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../aio.html">Learner View</a>
                      </div>
                    </div>
                  </div>
<!--/div.accordion-item-->
                </div>
<!--/div.accordion-flush-->
              </div>
<!--div.sidenav-view-selector -->
            </div>
<!--/div.col -->

            <hr>
</div>
<!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="01-introduction.html">1. Introduction to Genome Mining</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="02-antismash.html">2. Secondary metabolite biosynthetic gene cluster identification</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="03-antiSMASHdb-MiBiG.html">3. Genome Mining Databases</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="04-bigscape.html">4. BGC Similarity Networks</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="05-bigslice-bigfam.html">5. Homologous BGC Clusterization</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="06-corason.html">6. Finding Variation on Genomic Vicinities</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="07-evomining.html">7. Evolutionary Genome Mining</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="08-GatorGC.html">8. GATOR-GC: Genomic Assessment Tool for Orthologous Regions and Gene Clusters</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush10">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading10">
        <a href="09-metabolomics.html">9. Metabolomics workshop</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush11">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading11">
        <a href="10-OtherResources.html">10. Other Resources</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width">
<div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul>
<li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <li><a href="data.html">Data</a></li>
<hr>
<li><a class="dropdown-item" href="discuss.html">Discussion</a></li>
<li><a class="dropdown-item" href="reference.html">Glossary</a></li>
                    </ul>
</div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources">
<a href="../instructor/aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none">
<div class="d-grid gap-1">

            </div>
          </div>
<!-- /div.accordion -->
        </div>
<!-- /div.sidebar-inner -->
      </nav>
</div>
<!-- /div.sidebar -->
  </div>
<!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-extra.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <main id="main-content" class="main-content"><div class="container lesson-content">
        
        
<section id="aio-01-introduction"><p>Content from <a href="01-introduction.html">Introduction to Genome Mining</a></p>
<hr>
<p>Last updated on 2026-02-20 |

        <a href="https://github.com/carpentries-incubator/genome-mining/edit/main/episodes/01-introduction.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 10 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What is Genome Mining?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand that Natural products are encoded in Biosynthetic Gene
Clusters.</li>
<li>Understand that Biosynthetic Gene Clusters can be identified in the
genomic material.</li>
<li>Discuss bioinformatic’s good practices with your colleagues.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="what-is-genome-mining">What is Genome Mining?<a class="anchor" aria-label="anchor" href="#what-is-genome-mining"></a>
</h2>
<hr class="half-width">
<p><a href="../fig/Chapter1Fig1.png">
<img src="../fig/Introduction01.png" alt="Complete pipeline of genome mining. From a single genome, this example obtains their BGC and compares them with other BGC from related genomes" width="60%" height="60%" class="figure"></a></p>
<p>In bioinformatics, genomic mining is defined as the computational
analysis of nucleotide sequence data based on the comparison and
recognition of conserved patterns. Under this definition, any
computational method that involves searching for and predicting
physiological or metabolic properties is considered part of genomic
mining. The specific focus of genomic mining, when applied to natural
products (NPs), is centered on the identification of biosynthetic gene
clusters (BGCs) of NPs.</p>
<p>Natural products (NPs) are low molecular weight organic molecules
that encompass a wide and diverse range of chemical entities with
multiple biological functions. These molecules can be produced by
bacteria, fungi, plants, and animals. Natural products (NPs) thus play
various roles that can be analyzed from two main perspectives:</p>
<ul>
<li>
<strong>Biological function</strong>: This refers to the role the
molecule plays in the producing organism.</li>
<li>
<strong>Anthropocentric function</strong>: This focuses on the
utility of NPs for humans, including their use in medicine, agriculture,
and other areas.</li>
</ul>
<p><a href="../fig/productos%20naturales.png">
<img src="../fig/productos%20naturales.png" alt="Natural products can be produced by bacteria, fungi, plants and animals" class="figure"></a></p>
<p>Currently, more than 126,000 NPs are known to originate from various
sources and are classified into six main groups. These groups are
defined based on their chemical structure, the enzymes involved in their
synthesis, the precursors used, and the final modifications they
undergo.</p>
<table class="table">
<colgroup>
<col width="0%">
<col width="99%">
<col width="0%">
</colgroup>
<thead><tr class="header">
<th><strong>Class</strong></th>
<th><strong>Description</strong></th>
<th><strong>Example</strong></th>
</tr></thead>
<tbody>
<tr class="odd">
<td><strong>Polyketides (PKs)</strong></td>
<td>Polyketides are organic molecules characterized by a repetitive
chain of ketone (&gt;C=O) and methylene (&gt;CH2) groups. Their
biosynthesis is similar to fatty acid synthesis, which is crucial to
their diversity. These versatile molecules are found in bacteria, fungi,
plants, and marine organisms. Notable examples include erythromycin, an
antibiotic used to treat respiratory infections, and lovastatin, a
lipid-lowering drug that reduces cholesterol. In summary, polyketides
have a structure based on the repetition of ketone and methylene groups,
with significant medical and pharmacological applications.</td>
<td><strong>Erythromycin</strong></td>
</tr>
<tr class="even">
<td><strong>Non-Ribosomal Peptides (NRPs)</strong></td>
<td>NRPs are a family of natural products that differ from ribosomal
peptides due to their non-linear synthesis. Their main structure is
based on non-ribosomal modules, which include three key domains:
Adenylation, acyl carrier, and condensation domain. Relevant examples
include Daptomycin, an antibiotic widely used in various infections. Its
NRP structure contributes to its antimicrobial activity. Cyclosporine is
an immunosuppressant used in organ transplants. Its NRP structure is
crucial to its function. In summary, NRPs feature a non-linear modular
structure and play a significant role in medicine and chemical
biology.</td>
<td><strong>Daptomycin</strong></td>
</tr>
<tr class="odd">
<td><strong>Ribosomally synthesized and Post-translationally modified
Peptides (RiPPs)</strong></td>
<td>RiPPs are a diverse class of natural products of ribosomal origin.
These peptides are produced in the ribosomes and then undergo enzymatic
modifications after their synthesis. Historically, RiPPs were studied
individually, but in 2013, uniform naming guidelines were established
for these natural products. RiPPs include examples such as Microcin J25,
an antibacterial RiPP produced by <em>Escherichia coli</em>, and Nisin,
an antimicrobial RiPP produced by <em>Lactococcus lactis</em>. Their
diversity and applications continue to be the subject of ongoing
research.</td>
<td><strong>Nisin</strong></td>
</tr>
<tr class="even">
<td><strong>Saccharides</strong></td>
<td>Saccharides produced in bacterial secondary metabolism are molecules
synthesized by various mechanisms and then undergo subsequent enzymatic
modifications based on the addition of carbohydrates. Two notable
examples are Kanamycin, an aminoglycoside produced by <em>Streptomyces
kanamyceticus</em>, used to treat bacterial infections, and
Streptomycin, another aminoglycoside produced by <em>Streptomyces
griseus</em>, which was one of the first antibiotics used to treat
tuberculosis. Although their use has decreased due to bacterial
resistance, they remain important in some cases.</td>
<td><strong>Kanamycin</strong></td>
</tr>
<tr class="odd">
<td><strong>Terpenes</strong></td>
<td>Traditionally, terpenes have been considered derivatives of
2-methyl-butadiene, better known as isoprene. Although they have been
related to isoprene, terpenes do not directly derive from isoprene, as
it has never been found as a natural product. The true precursor of
terpenes is mevalonic acid, which comes from acetyl coenzyme A. Terpenes
originate through the enzymatic polymerization of two or more isoprene
units, assembled and modified in many different ways. Most terpenes have
multicentric structures that differ not only in functional group but
also in their basic carbon skeleton. These compounds are the main
constituent of the essential oils of some plants and flowers, such as
lemon and orange trees. They serve various functions in plants, such as
being part of chlorophyll, carotenoid pigments, gibberellin and abscisic
acid hormones, and increasing the fixation of proteins to cell membranes
through isoprenylation. Additionally, terpenes are used in traditional
medicine, aromatherapy, and as potential pharmacological agents. Two
notable examples of terpenes are limonene, present in citrus peels and
used in perfumery and cleaning products, and hopanoids, pentacyclic
compounds similar to sterols, whose main function is to confer rigidity
to the plasma membrane in prokaryotes.</td>
<td><strong>Diplopterol</strong></td>
</tr>
<tr class="even">
<td><strong>Alkaloids</strong></td>
<td>Alkaloids are a class of natural, basic organic compounds that
contain at least one nitrogen atom. They are mainly derived from amino
acids and are characterized by their water solubility in acidic pH and
solubility in organic solvents at alkaline pH. These compounds are found
in various organisms, such as plants, fungi, and bacteria. Alkaloids
have a wide range of pharmacological activities, such as Quinine, used
against malaria, and Ephedrine, a bronchodilator. Additionally, some
alkaloids, such as BE-54017, have an unusual structure, as it presents
an indenotryptoline skeleton, rarely observed in other bis-indoles;
however, its specific application is not well documented.</td>
<td><strong>BE-54017</strong></td>
</tr>
</tbody>
</table></section><section><h2 class="section-heading" id="genome-mining-aims-to-find-bgcs">Genome mining aims to find BGCs<a class="anchor" aria-label="anchor" href="#genome-mining-aims-to-find-bgcs"></a>
</h2>
<hr class="half-width">
<p>Natural products are encoded in <a href="https://en.wikipedia.org/wiki/Metabolic_gene_cluster" class="external-link">Biosynthetic
Gene Clusters</a> (BGCs) in Bacteria. These BGCs are clusters of genes
generally placed together in the same genome region. These include the
genes encoding the biosynthetic enzymes and those related to the
metabolite’s transport or resistance against antibacterial metabolites.
Most BGCs are composed of several types of genes, so genome mining is
based on the identification of these genes in a genome.</p>
<p><a href="../fig/08-20-24_BGC_arrange.png">
<img src="../fig/08-20-24_BGC_arrange.png" alt="BGC arrange example" class="figure"></a></p>
<p>Each class of biosynthetic gene clusters (BGCs) is distinguished by
the types of essential and accessory genes it contains. The most common
classes of BGCs for natural products include polyketide synthases
(PKSs), non-ribosomal peptide synthetases (NRPSs), ribosomally
synthesized and post-translationally modified peptides (RiPPs), and
terpenes. For example, non-ribosomal peptides (NRPs) are a class of
metabolites characterized by the assembly of amino acid residues or
their derivatives, with non-ribosomal peptide synthetases being the
enzymes responsible for assembling these molecules. NRPSs are large
enzymes organized into modules and domains, similar to other common
classes of BGCs. Below, we show you an animation created by <a href="https://www.michaelmullowney.com/gallery" class="external-link">Michael W. Mullowney</a>
of the biosynthesis of a fictitious NP called “fakeomycin”, which is of
the NRPS class with a cyclization domain.</p>
<p><a href="../fig/NRPS-animation-Mullowney_16-9_0.08.gif">
<img src="../fig/NRPS-animation-Mullowney_16-9_0.08.gif" alt="NRPS animation of fakeomycin" class="figure"></a></p>
<p><a href="https://en.wikipedia.org/wiki/Genome_mining" class="external-link">Genome
mining</a> consists in analyzing genomes with specialized algorithms
designed to find some BGCs. Chemists in the last century diligently
characterized some of these clusters. We have extensive databases that
contain information about which genes belong to which BGCs and some
control sets of genes that do not. The use of genome mining
methodologies facilitates the prioritization of BGCs for the search of
novel metabolites. Since the era of next-generation sequencing, genomes
have been explored as a source for discovering new BGCs.</p>
</section><section><h2 class="section-heading" id="chloramphenicol-is-a-known-antibiotic-produced-in-a-bgc">Chloramphenicol is a known antibiotic produced in a BGC<a class="anchor" aria-label="anchor" href="#chloramphenicol-is-a-known-antibiotic-produced-in-a-bgc"></a>
</h2>
<hr class="half-width">
<p>For example, let’s look into the BGC responsible for chloramphenicol
biosynthesis. This is a BGC described for the first time in a
<em>Streptomyces venezuelae</em> genome.</p>
<p><a href="../fig/episode1-fig1.PNG">
<img src="../fig/episode1-fig1.PNG" alt="MIBiG layout of the Chloramphenicol gene cluster from _Streptomyces venezuelae_ comprising 17 genes" class="figure"></a> <a href="https://mibig.secondarymetabolites.org/repository/BGC0000893/index.html#r1c1" class="external-link">Explore
the BGC</a></p>
<p><a href="https://en.wikipedia.org/wiki/Genome_mining" class="external-link">Genome Mining
Wikipedia</a></p>
<div id="exercise-1-sort-the-steps-to-identify-bgcs-similar-to-clavulanic-acid" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="exercise-1-sort-the-steps-to-identify-bgcs-similar-to-clavulanic-acid" class="callout-inner">
<h3 class="callout-title">Exercise 1: Sort the Steps to Identify BGCs
Similar to Clavulanic Acid</h3>
<div class="callout-content">
<p>Below is a list of steps in disarray that are part of the process of
identifying BGCs similar to clavulanic acid. Your task is to logically
order them to establish a coherent methodology that allows the effective
identification of such BGCs.</p>
<p><strong>Disordered Steps:</strong></p>
<ol style="list-style-type: lower-alpha">
<li><p>Annotate the genes within the identified BGCs to predict their
function.</p></li>
<li><p>Compare the identified BGCs against databases of known BGCs to
find similarities to clavulanic acid.</p></li>
<li><p>Extract DNA from samples of interest, such as microorganism-rich
soils or specific bacterial cultures.</p></li>
<li><p>Conduct phylogenetic analysis of the BGCs to explore their
evolutionary relationship.</p></li>
<li><p>Use bioinformatics tools to assemble DNA sequences and detect
potential BGCs.</p></li>
<li><p>Sequence the extracted DNA using next-generation sequencing (NGS)
techniques.</p></li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<ol style="list-style-type: decimal">
<li><ol start="3" style="list-style-type: lower-alpha">
<li>Extract DNA from samples of interest, such as microorganism-rich
soils or specific bacterial cultures.</li>
</ol></li>
<li><ol start="6" style="list-style-type: lower-alpha">
<li>Sequence the extracted DNA using next-generation sequencing (NGS)
techniques.</li>
</ol></li>
<li><ol start="5" style="list-style-type: lower-alpha">
<li>Use bioinformatics tools to assemble DNA sequences and detect
potential BGCs.</li>
</ol></li>
<li><ol style="list-style-type: lower-alpha">
<li>Annotate the genes within the identified BGCs to predict their
function.</li>
</ol></li>
<li><ol start="4" style="list-style-type: lower-alpha">
<li>Conduct phylogenetic analysis of the BGCs to explore their
evolutionary relationship.</li>
</ol></li>
<li><ol start="2" style="list-style-type: lower-alpha">
<li>Compare the identified BGCs against databases of known BGCs to find
similarities to clavulanic acid.</li>
</ol></li>
</ol>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="planning-a-genome-mining-project">Planning a genome mining project<a class="anchor" aria-label="anchor" href="#planning-a-genome-mining-project"></a>
</h2>
<hr class="half-width">
<p>Here we will provide tips and tricks to plan and execute a genome
mining project. Firstly, choose a set of genomes from taxa. In this
lesson we will be working with <em>S. agalactiae</em> <a href="https://zenodo.org/record/6595388#.YtD9LsFBxUJ" class="external-link">genomes</a>
(Tettelin et al., 2005). Although this genus is not know for its
potential as a Natural products producer, it is good enough to show
different approaches to genome mining. Recently, metagenomes have been
considered in genome mining studies. Here are some considerations that
might be useful as a genome miner:</p>
<ul>
<li>Think about a research question before starting to analyze the
data.</li>
<li>Remember, raw metadata should remain intact during all genome mining
processes. It could be a good idea to change its file permissions to
read-only.</li>
<li>Gather as much information as metadata of all the genomes you work
with.</li>
<li>All your intermediate steps should be considered temporal and may be
removed without risk.</li>
<li>Save your scripts using a version manager, GitHub for example.</li>
<li>Share your data in public repositories.</li>
<li>Give time to make your science repeatable and help your
community.</li>
</ul>
<div id="discussion-1-describe-your-project" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="discussion-1-describe-your-project" class="callout-inner">
<h3 class="callout-title">Discussion 1: Describe your project</h3>
<div class="callout-content">
<p>What are the advantages of documenting your work? Are there other
things you can do to make your mining work more reproducible, like
commenting on your scripts, etc.?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p>Documenting your work helps others (including your future self)
understand what you have done and facilitates troubleshooting. To make
your work more reproducible, you can detail the code logic by commenting
on your scripts, use version control systems like <a href="https://carpentries-incubator.github.io/git-novice-branch-pr/" class="external-link">Git</a>,
have backups of your scripts, share your data and code with others to
facilitate feedback, and follow standardized workflows or protocols like
those taught in this lesson. You can also name your files properly with
a consistent format, structure your directories well, and back up your
work regularly. Additionally, keeping a log (or “bitácora”) can serve as
a tutorial for your future self, making it easier to understand your
work later on.</p>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="starting-a-genome-mining-project">Starting a genome mining project<a class="anchor" aria-label="anchor" href="#starting-a-genome-mining-project"></a>
</h2>
<hr class="half-width">
<p>Once you have chosen your set of genomes, you need to annotate the
sequences. The process of genome annotation needs two steps. First, a
gene calling approach (structural annotation), which looks for CDS or
RNAs within the DNA sequences. Once these features have been detected,
you need to assign a function for each CDS (functional annotation). This
is usually done through comparison against protein databases. There are
tens of bioinformatics tools to annotate genomes, but some of the most
broadly used are; RAST (Aziz et al. 2008), and Prokka (Seeman, 2014).
Here, we will start the genome mining lesson with <em>S. agalactiae</em>
genomes already annotated by Prokka. You can download this data from
this <a href="https://zenodo.org/record/6595388#.YtD9LsFBxUJ" class="external-link">repository</a>.
The annotated genomes are written in GeneBank format (extension “.gbk”).
To learn more about the basic annotation of genomes, see the lesson
named <a href="https://carpentries-incubator.github.io/pangenomics/03-annotation-with-Prokka/index.html" class="external-link">“Pangenome
Analysis in Prokaryotes: Annotating Genomic Data”</a>. These files are
also accessible in the… <strong>Insert introduction related to the
access to the server??</strong></p>
</section><section><h2 class="section-heading" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<hr class="half-width">
<ul>
<li>Aziz, R. K., Bartels, D., Best, A. A., DeJongh, M., Disz, T.,
Edwards, R. A., … &amp; Zagnitko, O. (2008). The RAST Server: rapid
annotations using subsystems technology. BMC genomics, 9(1), 1-15.</li>
<li>Seemann, T. (2014). Prokka: rapid prokaryotic genome annotation.
Bioinformatics, 30(14), 2068-2069.</li>
<li>Tettelin, H., Masignani, V., Cieslewicz, M. J., Donati, C., Medini,
D., Ward, N. L., … &amp; Fraser, C. M. (2005). Genome analysis of
multiple pathogenic isolates of Streptococcus agalactiae: implications
for the microbial “pan-genome”. Proceedings of the National Academy of
Sciences, 102(39), 13950-13955.</li>
<li>Nett, Markus (2014). Kinghorn, A. D., ed. Genome Mining: Concept and
Strategies for Natural Product Discovery. Springer International
Publishing. pp. 199-245. ISBN 978-3-319-04900-7. <a href="../doi:10.1007/978-3-319-04900-7%5C_4" class="uri">doi:10.1007/978-3-319-04900-7\_4</a>.</li>
<li>Katz, Leonard; Baltz, Richard H. «Natural product discovery: past,
present, and future». Journal of Industrial Microbiology and
Biotechnology 43 (2-3): 155-176. ISSN 1476-5535. <a href="../doi:10.1007/s10295-015-1723-5" class="uri">doi:10.1007/s10295-015-1723-5</a>.</li>
<li>Mejía Ponce, Paulina M.(2017). Análisis filogenético de familias de
enzimas que utilizan tRNA, indicios para el descubrimiento de productos
naturales ocultos en Actinobacteria. Tesis (M.C.)–Centro de
Investigación y de Estudios Avanzados del I.P.N. Unidad Irapuato.
Laboratorio Nacional de Genómica para la Biodiversidad.
2020-08-13T02:57:32Z.</li>
</ul>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<ul>
<li>Natural products are encoded in Biosynthetic Gene Clusters
(BGCs)</li>
<li>Genome mining describes the exploitation of genomic information with
specialized algorithms intended to discover and study BGCs</li>
</ul>
</div>
</div>
</div>
</section></section><section id="aio-02-antismash"><p>Content from <a href="02-antismash.html">Secondary metabolite biosynthetic gene cluster identification</a></p>
<hr>
<p>Last updated on 2026-02-20 |

        <a href="https://github.com/carpentries-incubator/genome-mining/edit/main/episodes/02-antismash.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 30 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How can I annotate known BGC?</li>
<li>Which kind of analysis antiSMASH can perform?</li>
<li>Which file extension accepts antiSMASH?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand antiSMASH applications.</li>
<li>Perform a Minimal antiSMASH run analysis.</li>
<li>Explore several <em>Streptococcus</em> genomes by identifying the
BGCs presence and the types of secondary metabolites produced.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr class="half-width">
<p>Within microbial genomes, we can find some specific regions that take
part in the biosynthesis of secondary metabolites, these sections are
known as Biosynthetic Gene Clusters, which are relevant due to the
possible applications that they may have, for example: antimicrobials,
antitumors, cholesterol-lowering, immunosuppressants, antiprotozoal,
antihelminth, antiviral and anti-aging activities.</p>
<p>What is a natural Product?</p>
<p>antiSMASH is a pipeline based on <a href="https://www.ebi.ac.uk/training/online/courses/pfam-creating-protein-families/what-are-profile-hidden-markov-models-hmms/#:~:text=Profile%20HMMs%20are%20probabilistic%20models,the%20alignment%2C%20see%20Figure%202" class="external-link">profile
hidden Markov models</a> that allow us to identify the gene clusters
contained within the genome sequences that encode secondary metabolites
of all known broad chemical classes.</p>
</section><section><h2 class="section-heading" id="antismash-input-files">antiSMASH input files<a class="anchor" aria-label="anchor" href="#antismash-input-files"></a>
</h2>
<hr class="half-width">
<p>antiSMASH pipeline can work with three different file formats
<code>GenBank</code>, <code>FASTA</code> and <code>EMBL</code>. Both
<code>GenBank</code> and <code>EMBL</code> formats include genome
annotations, while a <code>FASTA</code> file just comprises the
nucleotides of each genomic contig.</p>
</section><section><h2 class="section-heading" id="running-antismash">Running antiSMASH<a class="anchor" aria-label="anchor" href="#running-antismash"></a>
</h2>
<hr class="half-width">
<p>The command line usage of antiSMASH is detailed in the following <a href="https://docs.antismash.secondarymetabolites.org/command_line/" class="external-link">repositories:</a></p>
<p>In summary, you will need to use your genome as the input. Then,
antiSMASH will create an output folder for each of your genomes. Within
this folder, you will find a single <code>.gbk</code> file for each of
the detected Biosynthetic Gene Clusters (we will use these files for
subsequent analyses) and a <code>.html</code> file, among others. By
opening the <code>.html</code> file, you can explore the antiSMASH
annotations.</p>
<p>You can run antiSMASH in two ways <strong>Minimal and Full-featured
run</strong>, as follows:</p>
<table class="table">
<colgroup>
<col width="14%">
<col width="85%">
</colgroup>
<thead><tr class="header">
<th>Run type</th>
<th>Command</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Minimal run</td>
<td>antismash genome.gbk</td>
</tr>
<tr class="even">
<td>Full-featured run</td>
<td>antismash –cb-general –cb-knownclusters –cb-subclusters –asf
–pfam2go –smcog-trees genome.gbk</td>
</tr>
</tbody>
</table>
<div id="exercise-1-antismash-scope" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="exercise-1-antismash-scope" class="callout-inner">
<h3 class="callout-title">Exercise 1: antiSMASH scope</h3>
<div class="callout-content">
<p>If you run an <em>Streptococcus agalactiae</em> annotated genome
using antiSMASH, what results do you expect?</p>
<ol style="list-style-type: lower-alpha">
<li><p>The substances that a cluster produces</p></li>
<li><p>A prediction of the metabolites that the clusters can
produce</p></li>
<li><p>A prediction of genes that belong to a biosynthetic
cluster</p></li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<ol style="list-style-type: lower-alpha">
<li>False. antiSMASH is not an algorithm devoted to substance
prediction.<br>
</li>
<li>False. Although antiSMASH does have some information about
metabolites produced by similar clusters, this is not its main
purpose.<br>
</li>
<li>True. antiSMASH compares domains and performs a prediction of the
genes that belong to biosynthetic gene clusters.</li>
</ol>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="run-your-own-antismash-analysis">Run your own antiSMASH analysis<a class="anchor" aria-label="anchor" href="#run-your-own-antismash-analysis"></a>
</h2>
<hr class="half-width">
<p>First, activate the GenomeMining conda environment:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">$</span> conda deactivate</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="ex">$</span> conda activate /miniconda3/envs/GenomeMining_Global</span></code></pre>
</div>
<p>Second, run the antiSMASH command shown earlier in this lesson on the
data <code>.gbk</code> or <code>.fasta</code> files. The command can be
executed either in one single file, all the files contained within a
folder or in a specific list of files. Here we show how it can be
performed in these three different cases:</p>
<div class="section level3">
<h3 id="running-antismash-in-a-single-file">Running antiSMASH in a single file<a class="anchor" aria-label="anchor" href="#running-antismash-in-a-single-file"></a>
</h3>
<p>Choose the annotated file ´agalactiae_A909_prokka.gbk´</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="ex">$</span> mkdir <span class="at">-p</span> ~/pan_workshop/results/antismash  </span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="ex">$</span> cd ~/pan_workshop/results/antismash  </span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="ex">$</span> antismash <span class="at">--genefinding-tool</span><span class="op">=</span>none ~/pan_workshop/results/annotated/Streptococcus_agalactiae_A909_prokka/Streptococcus_agalactiae_A909_prokka.gbk    </span></code></pre>
</div>
<p>To see the antiSMASH generated outcomes do the following:</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="ex">$</span> tree <span class="at">-L</span> 1 ~/pan_workshop/results/antismash/Streptococcus_agalactiae_A909_prokka</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>.
├── CP000114.1.region001.gbk
├── CP000114.1.region002.gbk
├── css
├── images
├── index.html
├── js
├── regions.js
├── Streptococcus_agalactiae_A909.gbk
├── Streptococcus_agalactiae_A909.json
├── Streptococcus_agalactiae_A909.zip
└── svg</code></pre>
</div>
</div>
<div class="section level3">
<h3 id="running-antismash-over-a-list-of-files">Running antiSMASH over a list of files<a class="anchor" aria-label="anchor" href="#running-antismash-over-a-list-of-files"></a>
</h3>
<p>Now, imagine that you want to run antiSMASH over all
<em>Streptococcus agalactiae</em> annotated genomes. Use a
<code>for</code> cycle and <code>*</code> as a wildcard to run antiSMASH
over all files that start with “S” in the annotated directory.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="ex">$</span> for gbk_file in ~/pan_workshop/results/annotated/<span class="pp">*</span>/S<span class="pp">*</span>.gbk</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="op">&gt;</span> do</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="op">&gt;</span>	antismash <span class="ex">--genefinding-tool=none</span> <span class="va">$gbk_file</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="op">&gt;</span> done</span></code></pre>
</div>
</div>
</section><section><h2 class="section-heading" id="visualizing-antismash-results">Visualizing antiSMASH results<a class="anchor" aria-label="anchor" href="#visualizing-antismash-results"></a>
</h2>
<hr class="half-width">
<p>In order to see the results after an antiSMASH run, we need to access
to the <code>index.html</code> file. Where is this file located?</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="ex">$</span> cd Streptococcus_agalactiae_A909_prokka</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="ex">$</span> pwd</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>~/pan_workshop/results/antismash/Streptococcus_agalactiae_A909_prokka</code></pre>
</div>
<p>As outcomes you should get a folder comprised mainly by the following
files:</p>
<ul>
<li>
<strong><code>.gbk</code> files</strong> For each Biosynthetic Gene
cluster region found.</li>
<li>
<strong><code>.json</code> file</strong> To know the input file
name, the antiSMASH used version and the regions data
(id,sequence_data).</li>
<li>
<strong><code>index.html</code> file</strong> To visualize the
outcomes from the analysis.</li>
</ul>
<p>In order to access these results, we can use scp protocol to download
the directory in your local computer.</p>
<p>If using <code>scp</code> , on your local machine, open a GIT bash
terminal in the <strong>destiny</strong> folder and execute the
following command:</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="ex">$</span> scp <span class="at">-r</span> user@bioinformatica.matmor.unam.mx:~/pan_workshop/results/antismash/S<span class="pp">*</span>A909.prokka/<span class="pp">*</span> ~/Downloads/.</span></code></pre>
</div>
<p>If using R-studio then in the left panel, choose the “more” option,
and “export” your file to your local computer. Decompress the
Streptococcus_agalactiae_A909.prokka.zip file.</p>
<p>Another way to download the data to your computer, you first compress
the folder and download the compressed file from JupyterHub</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="ex">$</span> cd ~/pan_workshop/results/antismash</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a><span class="ex">$</span> zip <span class="at">-r</span> Streptococcus_agalactiae_A909_prokka.zip Streptococcus_agalactiae_A909_prokka</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a><span class="ex">$</span> ls</span></code></pre>
</div>
<p>In the JupyterHub, navigate to the pan_workshop/results/antismash/
folder, select the file we just created, and press the download button
at the top</p>
<p>Once in your local machine, in the directory,
Streptococcus_agalactiae_A909.prokka, open the <code>index.html</code>
file on your local web browser.</p>
</section><section><h2 class="section-heading" id="understanding-the-output">Understanding the output<a class="anchor" aria-label="anchor" href="#understanding-the-output"></a>
</h2>
<hr class="half-width">
<p>The visualization of the results includes many different options. To
understand all the possibilities, we suggest reading the following <a href="https://docs.antismash.secondarymetabolites.org/understanding_output/" class="external-link">tutorial:</a></p>
<p>Briefly, in the “Overview” page ´.HTML´ you can find all the regions
found within every analyzed record/contig (antiSMASH inputs), and
summarized all this information in five main features:</p>
<ul>
<li>
<strong>Region:</strong> The region number.</li>
<li>
<strong>Type:</strong> Type of the product detected by
antiSMASH.</li>
<li>
<strong>From,To:</strong> The region’s location (nucleotides).</li>
<li>
<strong>Most similar known cluster:</strong> The closest compound
from th MIBiG database.</li>
<li>
<strong>Similarity:</strong> Percentage of genes within the closest
known compound that have significant BLAST hit (The last two columns
containing comparisons to the MIBiG database will only be shown if
antiSMASH was run with the KnownClusterBlast option ´–cc-mibig´).</li>
</ul></section><section><h2 class="section-heading" id="antismash-web-services">antiSMASH web services<a class="anchor" aria-label="anchor" href="#antismash-web-services"></a>
</h2>
<hr class="half-width">
<p>antiSMASH can also be used in an online server in the <a href="https://antismash.secondarymetabolites.org/#!/start" class="external-link">antiSMASH
website:</a> You will be asked to give your email. Then, the results
will be sent to you and you will be allowed to donwload a folder with
the annotations.</p>
<div class="section level4">
<h4 id="exercises-and-discussion">Exercises and discussion<a class="anchor" aria-label="anchor" href="#exercises-and-discussion"></a>
</h4>
<div id="exercise-2-ncbi-and-antismash-webserver" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="exercise-2-ncbi-and-antismash-webserver" class="callout-inner">
<h3 class="callout-title">Exercise 2: NCBI and antiSMASH webserver</h3>
<div class="callout-content">
<p>Run antiSMASH web server over <em>S. agalactiae</em> A909. First,
explore the NCBI assembly database to obtain the accession. Get the id
of your results.</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<ol style="list-style-type: decimal">
<li>Go to <a href="https://www.ncbi.nlm.nih.gov/" class="external-link">NCBI</a> and search
<em>S. agalactiae</em> A909.</li>
<li>Choose the assembly database and copy the GenBank sequence ID at the
bottom of the site.</li>
<li>Go to <a href="https://antismash.secondarymetabolites.org/#!/start" class="external-link">antiSMASH</a>
</li>
<li>Choose the accession from NCBI and paste
<code>CP000114.1</code>
</li>
<li>Run antiSMASH and paste into the collaborative document your results
id example
<code>bacteria-cbd13a47-8095-495f-957c-dcf58c261529</code>
</li>
</ol>
</div>
</div>
</div>
</div>
<div id="exercise-3-optional-exercise-run-antismash-over-thermophilus" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="exercise-3-optional-exercise-run-antismash-over-thermophilus" class="callout-inner">
<h3 class="callout-title">Exercise 3: (*Optional exercise ) Run
antiSMASH over <em>thermophilus</em>
</h3>
<div class="callout-content">
<p>Try Download and annotate <em>S. thermophilus</em> genomes strains
LMD-9 (MiBiG), and strain CIRM-BIA 65 reference genome. With the
following information, generate the script to run antiSMASH only on the
<em>S. thermophilus</em> annotated genomes.</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a> <span class="cf">done</span>  </span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>  <span class="ex">antismash</span> <span class="at">--genefinding-tool</span><span class="op">=</span>none <span class="va">$__</span>  </span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a> <span class="cf">for</span> ___ <span class="kw">in</span>  ____________  </span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a><span class="cf">do</span>  </span></code></pre>
</div>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<p>Guide to solve this problem First download genomes with
genome-download. Remember to activate the conda environment!</p>
<p>Download the annotated genomes, and then the genbank files. 0.
<code>ncbi-genome-download --formats genbank --genera "Streptococcus thermophilus"  -S "LMD-9"  -o thermophilus bacteria</code>
<code>ncbi-genome-download --formats fasta --genera "Streptococcus thermophilus"  -S "CIRM-BIA 65" -n bacteria</code>
Then, move the genomes to the directory
<code>~/pan_workshop/results/annotated/S*.gbk</code></p>
<ol start="2" style="list-style-type: decimal">
<li><p>First, we need to start the for cycle:
<code>for mygenome in ~/pan_workshop/results/annotated/S*.gbk</code><br>
note that we are using the name <code>mygenome</code> as the variable
name in the for cycle.</p></li>
<li><p>Then you need to use the reserved word <code>do</code> to start
the cycle.</p></li>
<li><p>Then you have to call antiSMASH over your variable
<code>mygenome</code>. Remember the <code>$</code> before your variable
to indicate to bash that now you are using the value of the
variable.</p></li>
<li><p>Finally, use the reserved word <code>done</code> to finish the
cycle.</p></li>
</ol>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a> <span class="cf">for</span> variable <span class="kw">in</span> ~/pan_workshop/results/annotated/t<span class="pp">*</span>.gbk  </span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a><span class="cf">do</span>  </span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a>	<span class="ex">antismash</span> <span class="at">--genefinding-tool</span><span class="op">=</span>none <span class="va">$mygenome</span>  </span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="cf">done</span>   </span></code></pre>
</div>
</div>
</div>
</div>
</div>
<div id="exercise-4-the-size-of-a-region" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="exercise-4-the-size-of-a-region" class="callout-inner">
<h3 class="callout-title">Exercise 4: The size of a region</h3>
<div class="callout-content">
<p>Sort the structure of the next commands that attempt to know the size
of a region:</p>
<p><code>SOURCE</code> <code>ORGANISM</code> <code>LOCUS</code><br><code>head</code> <code>grep</code></p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a> <span class="ex">$</span> _____  region.gbk  </span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a> <span class="ex">$</span> _____ __________ region.gbk  </span></code></pre>
</div>
<p>Apply your solution to get the size of the first region of <em>S.
agalactiae</em> A909</p>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4"> Show me the solution </h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" data-bs-parent="#accordionSolution4" aria-labelledby="headingSolution4">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="ex">$</span> cd ~/pan_workshop/results/antismash  </span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a><span class="ex">$</span> head Streptococcus_agalactiae_A909.prokka/CP000114.1.region001.gbk</span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a><span class="ex">$</span> grep LOCUS Streptococcus_agalactiae_A909.prokka/CP000114.1.region001.gbk</span></code></pre>
</div>
<p>Locus contains the information of the characteristics of the
sequence</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="ex">LOCUS</span>   	CP000114.1         	42196 bp	DNA 	linear   UNK 13-JUN-2022</span></code></pre>
</div>
<p>In this case the size of the region is 42196 bp</p>
</div>
</div>
</div>
</div>
<div id="discussion-1-bgc-classes-in-streptococcus" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="discussion-1-bgc-classes-in-streptococcus" class="callout-inner">
<h3 class="callout-title">Discussion 1: BGC classes in
<em>Streptococcus</em>
</h3>
<div class="callout-content">
<p>What BGC classes does the genus <em>Streptococcus</em> have ?</p>
</div>
</div>
</div>
<div id="accordionSolution5" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution5" aria-expanded="false" aria-controls="collapseSolution5">
  <h4 class="accordion-header" id="headingSolution5"> Show me the solution </h4>
</button>
<div id="collapseSolution5" class="accordion-collapse collapse" data-bs-parent="#accordionSolution5" aria-labelledby="headingSolution5">
<div class="accordion-body">
<p>In antiSMASH website output of A909 strain we see clusters of two
classes, T3PKS and arylpolyene. Nevertheless, this is only one
<em>Streptococcus</em> in order to infer all BGC classes in the genus we
need to run antiSMASH in more genomes.</p>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<ul>
<li>antiSMASH is a bioinformatic tool capable of identifying, annotating
and analysing secondary metabolite BGC</li>
<li>antiSMASH can be used as a web-based tool or as stand-alone
command-line tool</li>
<li>The file extensions accepted by antiSMASH are GenBank, FASTA and
EMBL</li>
</ul>
</div>
</div>
</div>
</div>
</section></section><section id="aio-03-antiSMASHdb-MiBiG"><p>Content from <a href="03-antiSMASHdb-MiBiG.html">Genome Mining Databases</a></p>
<hr>
<p>Last updated on 2026-02-20 |

        <a href="https://github.com/carpentries-incubator/genome-mining/edit/main/episodes/03-antiSMASHdb-MiBiG.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 25 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>Where can I find experimentally validated BGCs?</li>
<li>Where is information about all predicted BGCs?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Use MIBiG database as a source of experimentally tested BGC.</li>
<li>Explore antiSMASH database to learn about the distribution of
predicted BGC.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="mibig-database">MIBiG Database<a class="anchor" aria-label="anchor" href="#mibig-database"></a>
</h2>
<hr class="half-width">
<p>The Minimum Information about a Biosynthetic Gene cluster <a href="https://mibig.secondarymetabolites.org/repository" class="external-link">MIBiG</a> is a
database that facilitates consistent and systematic deposition and
retrieval of data on biosynthetic gene clusters. MIBiG provides a robust
community standard for annotations and metadata on biosynthetic gene
clusters and their molecular products. It will empower next-generation
research on the biosynthesis, chemistry and ecology of broad classes of
societally relevant bioactive secondary metabolites, guided by robust
experimental evidence and rich metadata components.</p>
<div class="section level3">
<h3 id="browsing-and-querying-in-the-mibig-database">Browsing and Querying in the MIBiG database<a class="anchor" aria-label="anchor" href="#browsing-and-querying-in-the-mibig-database"></a>
</h3>
<p>Select “Search” on the upper right corner of the menu bar</p>
<p><a href="../fig/MIBiG_search.png">
<img src="../fig/MIBiG_search.png" alt="MIBiG website homepage highlighting the search tool" class="figure"></a></p>
<p>For simple queries, such as <em>Streptococcus agalactiae</em> or
searching for a specific strain you can use the “Simple search”
function.</p>
<p><a href="../fig/MIBiG_query.png">
<img src="../fig/MIBiG_query.png" alt="MIBiG website query page" class="figure"></a></p>
<p>For complex queries, the database also provides a sophisticated query
builder that allows querying on all antiSMASH annotations. To enable
this function, click on “Build a query”</p>
</div>
<div class="section level3">
<h3 id="results">Results<a class="anchor" aria-label="anchor" href="#results"></a>
</h3>
<p><a href="../fig/MIBiG_results.png">
<img src="../fig/MIBiG_results.png" alt="MIBiG website displaying the results from the simple search Streptococcus" class="figure"></a></p>
<div id="exercise-1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="exercise-1" class="callout-inner">
<h3 class="callout-title">Exercise 1:</h3>
<div class="callout-content">
<p>Enter to <a href="https://mibig.secondarymetabolites.org/" class="external-link">MIBiG</a>
and search BGCs from <em>Streptococcus</em>. Search the BGCs that
produce the products Thermophilin 1277 and Streptolysin S. Based on the
table on MIBiG, which of these organisms has the most complete
annotation?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p><em>Streptococcus thermophilus</em> produce Thermophilin 1277 while
<em>Streptococcus pyogenes</em> M1 GAS produces Streptolysin S.
According to MIBiG metadata Streptolysin S BGC is complete while
Thermophilin 1277 is not. So Streptolysin S BGC is better annotated.</p>
</div>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="antismash-database">antiSMASH database<a class="anchor" aria-label="anchor" href="#antismash-database"></a>
</h2>
<hr class="half-width">
<p>The <a href="https://antismash-db.secondarymetabolites.org/" class="external-link">antiSMASH
database</a> provides an easy to use, up-to-date collection of annotated
BGC data. It allows to easily perform cross-genome analyses by offering
complex queries on the datasets.</p>
<div class="section level3">
<h3 id="browsing-and-querying-in-the-antismash-database">Browsing and Querying in the antiSMASH database<a class="anchor" aria-label="anchor" href="#browsing-and-querying-in-the-antismash-database"></a>
</h3>
<p>Select “Browse” on the top menu bar, alternatively you can select
“Query” in the center</p>
<p><a href="../fig/antiSMASH_db.png">
<img src="../fig/antiSMASH_db.png" alt="antiSMASH website homepage" class="figure"></a></p>
<p>For simple queries, such as “<em>Streptococcus</em>” or searching for
a specific strain you can use the “Simple search” function.</p>
<p><a href="../fig/antiSMASH_search.png">
<img src="../fig/antiSMASH_search.png" alt="antiSMASH website query page" class="figure"></a></p>
<p>For complex queries, the database also provides a sophisticated query
builder that allows querying on all antiSMASH annotations. To enable
this function, click on “Build a query”</p>
</div>
<div class="section level3">
<h3 id="results-1">Results<a class="anchor" aria-label="anchor" href="#results-1"></a>
</h3>
<p><a href="../fig/antiSMASH_query.png">
<img src="../fig/antiSMASH_query.png" alt="antiSMASH website displaying the results from the simple search Streptococcus" class="figure"></a></p>
<p>Use antiSMASH database to analyse the BGC contained in the
<em>Streptococcus</em> genomes. We’ll use Python to visualize the data.
First, import pandas, matplotlib.pyplot and seaborn libraries.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">PYTHON<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode python" tabindex="0"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="im">import</span> seaborn <span class="im">as</span> sns</span></code></pre>
</div>
<p>Secondly, store in a dataframe variable the content of the
<em>Streptococcus</em> predicted BGC downloaded from antiSMASH-db.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">PYTHON<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode python" tabindex="0"><code class="sourceCode python"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>data <span class="op">=</span> pd.read_csv(<span class="st">"https://raw.githubusercontent.com/AxelRamosGarcia/Genome-Mining/gh-pages/data/antismash_db.csv"</span>, sep<span class="op">=</span><span class="st">"</span><span class="ch">\t</span><span class="st">"</span>)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>data</span></code></pre>
</div>
<p><a href="../fig/21-08-24-chapter9-table.png">
<img src="../fig/21-08-24-chapter9-table.png" alt="a dataframe variable the content of the Streptococcus predicted BGC" width="800" class="figure"></a></p>
<p>Now, group the data by the variables Species and BGC type:</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">PYTHON<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode python" tabindex="0"><code class="sourceCode python"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>occurences <span class="op">=</span> data.groupby([<span class="st">"Species"</span>, <span class="st">"BGC type"</span>]).size().reset_index(name<span class="op">=</span><span class="st">"Occurrences"</span>)</span></code></pre>
</div>
<p>And visualize the content of the ocurrences grouped by species
column:</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">PYTHON<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode python" tabindex="0"><code class="sourceCode python"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>occurences</span></code></pre>
</div>
<p><a href="../fig/21-08-24-chapter9-occurences.png">
<img src="../fig/21-08-24-chapter9-occurences.png" alt="the content of the ocurrences grouped by species column" width="500" class="figure"></a></p>
<p>Let’s see our first visualization of the BGC content on a
heatmap.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">PYTHON<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode python" tabindex="0"><code class="sourceCode python"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>pivot <span class="op">=</span> occurences.pivot(index<span class="op">=</span><span class="st">"BGC type"</span>, columns<span class="op">=</span><span class="st">"Species"</span>, values<span class="op">=</span><span class="st">"Occurrences"</span>)</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>plt.figure(figsize<span class="op">=</span>(<span class="dv">8</span>, <span class="dv">10</span>))</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>sns.heatmap(pivot, cmap<span class="op">=</span><span class="st">"coolwarm"</span>)</span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>plt.show()</span></code></pre>
</div>
<p><a href="../fig/21-08-24-chapter9-heatmap.png">
<img src="../fig/21-08-24-chapter9-heatmap.png" alt="visualization of the BGC content on a heatmap." width="800" class="figure"></a></p>
<p>Now, let’s restrict ourselves to <em>S. agalactiae</em>.</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">PYTHON<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode python" tabindex="0"><code class="sourceCode python"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>agalactiae <span class="op">=</span> occurences[occurences[<span class="st">"Species"</span>] <span class="op">==</span> <span class="st">"agalactiae"</span>]</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>sns.scatterplot(agalactiae, x<span class="op">=</span><span class="st">"BGC type"</span>, y<span class="op">=</span><span class="st">"Occurrences"</span>)</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>plt.xticks(rotation<span class="op">=</span><span class="st">"vertical"</span>)</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>plt.show()</span></code></pre>
</div>
<p><a href="../fig/21-08-24-chapter9.dotplot.png">
<img src="../fig/21-08-24-chapter9.dotplot.png" alt="visualization of the BGC content of S. agalactiae. on a sctterplot" width="800" class="figure"></a></p>
<p>Finally, let’s restrict ourselves to BGC predicted less than 200
times.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">PYTHON<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode python" tabindex="0"><code class="sourceCode python"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>filtered <span class="op">=</span> occurences[occurences[<span class="st">"Occurrences"</span>] <span class="op">&lt;</span> <span class="dv">200</span>]</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>plt.figure(figsize<span class="op">=</span>(<span class="dv">15</span>, <span class="dv">5</span>))</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>sns.scatterplot(filtered, x<span class="op">=</span><span class="st">"BGC type"</span>, y<span class="op">=</span><span class="st">"Occurrences"</span>)</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>plt.xticks(rotation<span class="op">=</span><span class="st">"vertical"</span>)</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>plt.grid(axis<span class="op">=</span><span class="st">"y"</span>)</span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>plt.show()</span></code></pre>
</div>
<p><a href="../fig/21-08-24-chapter9.dotplot2.png">
<img src="../fig/21-08-24-chapter9.dotplot2.png" alt="visualization of the BGC content on a scatterplot" width="800" class="figure"></a></p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">PYTHON<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode python" tabindex="0"><code class="sourceCode python"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>filtered_pivot <span class="op">=</span> filtered.pivot(index<span class="op">=</span><span class="st">"BGC type"</span>, columns<span class="op">=</span><span class="st">"Species"</span>, values<span class="op">=</span><span class="st">"Occurrences"</span>)</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>plt.figure(figsize<span class="op">=</span>(<span class="dv">8</span>, <span class="dv">10</span>))</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>sns.heatmap(filtered_pivot, cmap<span class="op">=</span><span class="st">"coolwarm"</span>)</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>plt.show()</span></code></pre>
</div>
<p><a href="../fig/21-08-24-chapter9-pivot-filtered.png">
<img src="../fig/21-08-24-chapter9-pivot-filtered.png" alt="filtered heatmap " width="800" class="figure"></a></p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<ul>
<li>MIBiG provides BGCs that have been experimentally tested</li>
<li>antiSMASH database comprises predicted BGCs of each organism</li>
</ul>
</div>
</div>
</div>
</div>
</section></section><section id="aio-04-bigscape"><p>Content from <a href="04-bigscape.html">BGC Similarity Networks</a></p>
<hr>
<p>Last updated on 2026-02-20 |

        <a href="https://github.com/carpentries-incubator/genome-mining/edit/main/episodes/04-bigscape.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 45 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How can I measure similarity between BGCs?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand how BiG-SCAPE measures similarity between BGCs.</li>
<li>Take an antiSMASH output to perform a BiG-SCAPE analysis.</li>
<li>Interpret BiG-SCAPE similarity networks and GCF phylogenies.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr class="half-width">
<p>In the previous section, we learned how to study the BGCs encoded by
each of the genomes of our analyses. In case you are interested in the
study of a certain BGC or a certain strain, this may be enough. However,
sometimes the aim is to compare the biosynthetic potential of tens or
hundreds of genomes. To perform this kind of analysis, we will use
BiG-SCAPE (Navarro-Muñoz et al., 2019), a workflow that compares all the
BGCs detected by antiSMASH to find their relatedness. BiG-SCAPE will
search for Pfam domains (Mistry et al., 2021) in the protein sequences
of each BGC. Then, the Pfam domains will be linearized and compared,
creating different similarity networks and scoring the similarity of
each pair of clusters. Based on a cutoff value for this score, the
diverse BGCs will be classified on Gene Cluster Families (GCFs) to
facilitate their study. A single GCF is supposed to encompass BGCs that
produce chemically related metabolites (molecular families). Lower
cutoffs would create families of BGCs that produce identical compounds,
while higher cutoffs would create families of more loosely related
compounds.</p>
</section><section><h2 class="section-heading" id="preparing-the-input">Preparing the input<a class="anchor" aria-label="anchor" href="#preparing-the-input"></a>
</h2>
<hr class="half-width">
<p>In each of the antiSMASH output directories, we will find a single
<code>.gbk</code> file for each BGC, which includes “region” within its
filename. Thus, we will copy all these files to the new directory.</p>
<p>Move into the directory that has the antiSMASH results of each
genome:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">$</span> conda deactivate  </span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="ex">$</span> conda activate /miniconda3/envs/bigscape/  </span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="ex">$</span> cd ~/pan_workshop/results/antismash/</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="ex">$</span> ls</span></code></pre>
</div>
<p>Since we will put together in the same directory many files with
similar names, we want to make sure that nothing is left behind or
overwritten. For this, we will count all the <code>gbk</code> files of
all the genomes.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="ex">$</span> ls Streptococcus_agalactiae_<span class="pp">*</span>/<span class="pp">*</span>region<span class="pp">*</span>gbk <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code><span><span class="fl">16</span></span></code></pre>
</div>
<p>Since the names are somewhat cryptic, they could be repeated, so we
will rename the <code>gbks</code> in such a way that they include the
genome name.</p>
<p>Copy the following script to a file named
<code>change-names.sh</code> using <code>nano</code>:</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="co"># This script is to rename the antiSMASH gbks for them to include the species and strain names, taken from the directory name.</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="co"># The argument it requires is the name of the directory with the AntiSMASH output, which must NOT contain a slash at the end.</span></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="co"># Usage for one AntiSMASH output directory:</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="co">#     sh change-names.sh &lt;folder&gt;</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a><span class="co"># Usage for multiple AntiSMASH output directory:</span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a><span class="co">#     for species in &lt;output-directory-pattern*&gt;</span></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a><span class="co">#    	 do</span></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a><span class="co">#    		 sh change-names.sh $species</span></span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a><span class="co">#    	 done</span></span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a></span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a></span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a></span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-1</span> <span class="st">"</span><span class="va">$1</span><span class="st">"</span>/<span class="pp">*</span>region<span class="pp">*</span>gbk <span class="kw">|</span> <span class="cf">while</span> <span class="bu">read</span> <span class="va">line</span> <span class="co"># enlist the gbks of all regions in the directory and start a while loop</span></span>
<span id="cb4-16"><a href="#cb4-16" tabindex="-1"></a> <span class="cf">do</span></span>
<span id="cb4-17"><a href="#cb4-17" tabindex="-1"></a>    <span class="va">dir</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">$line</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-d</span><span class="st">'/'</span> <span class="at">-f1</span><span class="va">)</span> <span class="co"># save the directory name in a variable</span></span>
<span id="cb4-18"><a href="#cb4-18" tabindex="-1"></a>    <span class="va">file</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">$line</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-d</span><span class="st">'/'</span> <span class="at">-f2</span><span class="va">)</span> <span class="co"># save the file name in a variable</span></span>
<span id="cb4-19"><a href="#cb4-19" tabindex="-1"></a>	<span class="cf">for</span> directory <span class="kw">in</span> <span class="va">$dir</span> <span class="co"># start a for loop</span></span>
<span id="cb4-20"><a href="#cb4-20" tabindex="-1"></a>    	<span class="cf">do</span></span>
<span id="cb4-21"><a href="#cb4-21" tabindex="-1"></a>   		 <span class="bu">cd</span> <span class="va">$directory</span> <span class="co"># enter the directory</span></span>
<span id="cb4-22"><a href="#cb4-22" tabindex="-1"></a>      	<span class="va">newfile</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">$dir</span>-<span class="va">$file)</span> <span class="co"># make a new variable that fuses the directory name with the file name</span></span>
<span id="cb4-23"><a href="#cb4-23" tabindex="-1"></a>			 	<span class="bu">echo</span> <span class="st">"Renaming"</span> <span class="va">$file</span> <span class="st">" to"</span> <span class="va">$newfile</span> <span class="co"># print a message showing the old and new file names</span></span>
<span id="cb4-24"><a href="#cb4-24" tabindex="-1"></a>			 	<span class="fu">mv</span> <span class="va">$file</span> <span class="va">$newfile</span> <span class="co"># rename</span></span>
<span id="cb4-25"><a href="#cb4-25" tabindex="-1"></a>			 	<span class="bu">cd</span> .. <span class="co"># return to main directory before it begins again</span></span>
<span id="cb4-26"><a href="#cb4-26" tabindex="-1"></a>		 <span class="cf">done</span></span>
<span id="cb4-27"><a href="#cb4-27" tabindex="-1"></a> <span class="cf">done</span></span></code></pre>
</div>
<p>Run the script for all the directory:</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="ex">$</span> for species in Streptococcus_agalactiae_<span class="pp">*</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="op">&gt;</span> do</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="op">&gt;</span>   	 sh <span class="ex">change-names.sh</span> <span class="va">$species</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="op">&gt;</span>    done</span></code></pre>
</div>
<p>Now make a directory for all your BiG-SCAPE analyses and inside it
make a directory that contains all of the <code>gbks</code> of all of
your genomes. This one will be the input for BiG-SCAPE. (For convenience
<code>bigscape/</code> will be inside <code>antismash/</code> while we
run BiG-SCAPE).</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="ex">$</span> mkdir <span class="at">-p</span> bigscape/bgcs_gbks/</span></code></pre>
</div>
<p>Now copy all the region <code>gbks</code>to this new directory, and
look at its content:</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="ex">$</span> cp Streptococcus_agalactiae_<span class="pp">*</span>/<span class="pp">*</span>region<span class="pp">*</span>gbk bigscape/bgcs_gbks/</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="ex">$</span> ls bigscape/bgcs_gbks/</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code><span><span class="va">Streptococcus_agalactiae_18RS21_prokka</span><span class="op">-</span><span class="va">AAJO01000016.1.region001.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_18RS21_prokka</span><span class="op">-</span><span class="va">AAJO01000043.1.region001.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_2603V_prokka</span><span class="op">-</span><span class="va">NC_004116.1.region001.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_2603V_prokka</span><span class="op">-</span><span class="va">NC_004116.1.region002.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_515_prokka</span><span class="op">-</span><span class="va">NZ_CP051004.1.region001.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_515_prokka</span><span class="op">-</span><span class="va">NZ_CP051004.1.region002.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_A909_prokka</span><span class="op">-</span><span class="va">NC_007432.1.region001.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_A909_prokka</span><span class="op">-</span><span class="va">NC_007432.1.region002.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_CJB111_prokka</span><span class="op">-</span><span class="va">NZ_AAJQ01000010.region001.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_CJB111_prokka</span><span class="op">-</span><span class="va">NZ_AAJQ01000025.region001.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_COH1_prokka</span><span class="op">-</span><span class="va">NZ_HG939456.1.region001.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_COH1_prokka</span><span class="op">-</span><span class="va">NZ_HG939456.1.region002.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_H36B_prokka</span><span class="op">-</span><span class="va">AAJS01000020.1.region001.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_H36B_prokka</span><span class="op">-</span><span class="va">AAJS01000117.1.region001.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_NEM316_prokka</span><span class="op">-</span><span class="va">NC_004368.1.region001.gbk</span></span>
<span><span class="va">Streptococcus_agalactiae_NEM316_prokka</span><span class="op">-</span><span class="va">NC_004368.1.region002.gbk</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="running-big-scape">Running BiG-SCAPE<a class="anchor" aria-label="anchor" href="#running-big-scape"></a>
</h2>
<hr class="half-width">
<p>BiG-SCAPE can be executed in different ways, depending on the
installation mode that you applied. You could call the program through
<code>bigscape</code>, <code>run_bigscape</code>or
<code>run_bigscape.py</code>. Here, based on our installation (see <a href="index.html#setup">Setup</a>), we will use <code>bigscape</code>.</p>
<p>The options that we will use are described in the help page:</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="ex">-i</span> INPUTDIR, <span class="at">--inputdir</span> INPUTDIR</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>                    	<span class="ex">Input</span> directory of gbk files, if left empty, all gbk</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>                    	<span class="ex">files</span> in current and lower directories will be used.</span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a></span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a><span class="ex">-o</span> OUTPUTDIR, <span class="at">--outputdir</span> OUTPUTDIR</span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a>                    	<span class="ex">Output</span> directory, this will contain all output data</span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a>                    	<span class="ex">files.</span></span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a></span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a><span class="ex">--mix</span>             	By default, BiG-SCAPE separates the analysis according</span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a>                   	<span class="ex">to</span> the BGC product <span class="er">(</span><span class="ex">PKS</span> Type I, NRPS, RiPPs, etc.<span class="kw">)</span> <span class="ex">and</span></span>
<span id="cb9-11"><a href="#cb9-11" tabindex="-1"></a>                   	<span class="ex">will</span> create network directories for each class. Toggle</span>
<span id="cb9-12"><a href="#cb9-12" tabindex="-1"></a>                   	<span class="ex">to</span> include an analysis mixing all classes</span>
<span id="cb9-13"><a href="#cb9-13" tabindex="-1"></a>                  	 </span>
<span id="cb9-14"><a href="#cb9-14" tabindex="-1"></a><span class="ex">--hybrids-off</span>     	Toggle to also add BGCs with hybrid predicted products</span>
<span id="cb9-15"><a href="#cb9-15" tabindex="-1"></a>                    	<span class="ex">from</span> the PKS/NRPS Hybrids and Others classes to each</span>
<span id="cb9-16"><a href="#cb9-16" tabindex="-1"></a>                    	<span class="ex">subclass</span> <span class="er">(</span><span class="ex">e.g.</span> a <span class="st">'terpene-nrps'</span> BGC from Others would</span>
<span id="cb9-17"><a href="#cb9-17" tabindex="-1"></a>                    	<span class="ex">be</span> added to the Terpene and NRPS classes<span class="kw">)</span></span>
<span id="cb9-18"><a href="#cb9-18" tabindex="-1"></a></span>
<span id="cb9-19"><a href="#cb9-19" tabindex="-1"></a><span class="ex">--mode</span> <span class="dt">{global</span><span class="op">,</span><span class="dt">glocal</span><span class="op">,</span><span class="dt">auto}</span></span>
<span id="cb9-20"><a href="#cb9-20" tabindex="-1"></a>                    	<span class="ex">Alignment</span> mode for each pair of gene clusters.</span>
<span id="cb9-21"><a href="#cb9-21" tabindex="-1"></a>                    	<span class="st">'global'</span><span class="ex">:</span> the whole list of domains of each BGC are</span>
<span id="cb9-22"><a href="#cb9-22" tabindex="-1"></a>                    	<span class="ex">compared</span><span class="kw">;</span> <span class="st">'glocal'</span><span class="ex">:</span> Longest Common Subcluster mode.</span>
<span id="cb9-23"><a href="#cb9-23" tabindex="-1"></a>                    	<span class="ex">Redefine</span> the subset of the domains used to calculate</span>
<span id="cb9-24"><a href="#cb9-24" tabindex="-1"></a>                    	<span class="ex">distance</span> by trying to find the longest slice of common</span>
<span id="cb9-25"><a href="#cb9-25" tabindex="-1"></a>                    	<span class="ex">domain</span> content per gene in both BGCs, then expand each</span>
<span id="cb9-26"><a href="#cb9-26" tabindex="-1"></a>                    	<span class="ex">slice.</span> <span class="st">'auto'</span>: use glocal when at least one of the</span>
<span id="cb9-27"><a href="#cb9-27" tabindex="-1"></a>                    	<span class="ex">BGCs</span> in each pair has the <span class="st">'contig_edge'</span> annotation</span>
<span id="cb9-28"><a href="#cb9-28" tabindex="-1"></a>                    	<span class="ex">from</span> antiSMASH v4+, otherwise use global mode on that</span>
<span id="cb9-29"><a href="#cb9-29" tabindex="-1"></a>                    	<span class="ex">pair</span></span></code></pre>
</div>
<p>We will use the option <code>--mix</code> to have an analysis of all
of the BGCs together besides the analyses of the BGCs separated by
class. The <code>--hybrids-off</code> option will prevent from having
the same BGC twice (in the case of hybrid BGCs that could belong to two
classes) in our results. And since none of the BGCs is on a contig edge,
we could use the global mode. However, frequently, when analyzing draft
genomes, this is not the case. Thus, the auto mode will be the most
appropriate, which will use the global mode to align domains except for
those cases in which the BGC is located near to a contig end, for which
the glocal mode is automatically selected. For this run we will use only
the default cutoff value (0.3).</p>
<p>Now we are ready to run BiG-SCAPE:</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="ex">$</span> bigscape <span class="at">-i</span> bigscape/bgcs_gbks/ <span class="at">-o</span> bigscape/output_220124 <span class="at">--mix</span> <span class="at">--hybrids-off</span> <span class="at">--mode</span> auto <span class="at">--pfam_dir</span> /miniconda3/envs/bigscape/BiG-SCAPE-1.1.5/</span></code></pre>
</div>
<div id="output-names" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<span class="callout-header">Callout</span>
<div id="output-names" class="callout-inner">
<h3 class="callout-title">Output names</h3>
<div class="callout-content">
<p>Most of the times you will need to re-run a software once you have
explored the results; maybe you want to change parameters or add more
samples. For this reason, you will need a flexible way to organize your
project, including the names of the outputs. Usually, a good idea is to
put the date in the name of the folder and keep track of the analyses
you are running in your research notes.</p>
</div>
</div>
</div>
<p>Once the process is finished, you will find in your terminal screen
some basic results, such as the number of BGCs included in each type of
network. In the output folder you will find all the data.<br>
Since BiG-SCAPE does not produce a file with the run information, it is
useful to copy the text printed on the screen to a file. Use
<code>nano</code> to paste all the text that BiG-SCAPE generated in a
file named <code>bigscape.log</code> inside the folder
<code>output_100722/logs/</code>.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="ex">$</span> nano bigscape/output_220124/logs/bigscape.log</span></code></pre>
</div>
<p>Take a look at the BiG-SCAPE outputs:</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="ex">$</span> ls <span class="at">-F</span> bigscape/output_220124/</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>cache/  html_content/  index.html*  logs/  network_files/  SVG/</code></pre>
</div>
<p>To keep ordered your directories, move your <code>bigscape/</code>
directory to <code>results/</code>:</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="ex">$</span> mv bigscape/ ../</span></code></pre>
</div>
<p>And move the file <code>change-names</code> to folder
<code>scripts/</code> in the <code>pan_workshop/</code>directory to save
our little script:</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="ex">$</span> mv change-names.sh ~/pan_workshop/scripts/</span></code></pre>
</div>
</section><section><h2 class="section-heading" id="viewing-the-results">Viewing the results<a class="anchor" aria-label="anchor" href="#viewing-the-results"></a>
</h2>
<hr class="half-width">
<p>An easy way to prospect your results is by opening the
<code>index.html</code> file with a browser (Firefox, Chrome, Safari,
etc.). In order to do this, you need to go to your local machine and
locate in the folder where you want to store your BiG-SCAPE results. Now
we compress the folder <code>output_100622/</code> and download it to
your local machine.</p>
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="ex">$</span> cd ~/pan_workshop/results/bigscape/</span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a><span class="ex">$</span> zip <span class="at">-r</span> output_220124.zip output_220124</span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a><span class="ex">$</span> ls</span></code></pre>
</div>
<p>In the JupyterHub we navigate to the folders and go to
<code>/pan_workshop/results/bigscape</code> and select the one we just
created <code>output_220124.zip</code> and download it.</p>
<p>On your local computer we unzip the files and open
<code>output_220124/index.html</code> with a browser:</p>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a><span class="ex">$</span> firefox output_100622/index.html <span class="co"># If you have Firefox available, otherwise open it using the GUI.</span></span></code></pre>
</div>
<p>There are diverse sections in the visualization. The following image
shows the overview page. At the left there is the run information and
there are pie chart representations of the number of BGCs per genome and
per class. At the right there is the Network Overview for each of the
BGC classes found and the mix category. They show the Number of
Families, Average number of BGCs per family, Max number of BGCs in a
family and the Families with MIBiG Reference BGCs. You can click on the
name of the class to see its Network Overview.</p>
<p><a href="../fig/02_05_01.png">
<img src="../fig/02_05_01.png" alt="BIG-SCAPE output as visualized in the web page.
                                  	The overview page is displayed. At the left is the Run Information,
                                  	indicating the date and time at which the analysis was started and completed, as well as the parameters of the run.
                                  	Next is displayed the Input Data, specifying the total number of genomes and the total BGCs,
                                  	in this example 8 and 23, respectively. There are two pie charts, one representing the BGC per Genome
                                  	and the other the BGC per Class. At the right is the Network Overview,
                                  	which allows selection between mix and the different BGC classes.
                                  	From the mix overview it displays the Number of Families, Average number of BGCs per family,
                                  	Max number of BGCs in a family and the Families with MIBiG Reference BGCs." class="figure"></a></p>
<p>Below, there is a clustered heatmap of the presence/absence of the
GCFs in each genome for each class. You can customize this heatmap and
select the clustering methods, or the number of GCFs represented.</p>
<p><a href="../fig/02_05_02.png">
<img src="../fig/02_05_02.png" alt="BIG-SCAPE output as visualized in the web page.
                                  	The overview page displaying the clustered heatmap of the presence/absence of the GCFs,
                                  	each class organized as a column at the base of the table, in each genome,
                                  	which are organized as rows at the right side of the table. Presence is depicted in red with 1 and absence in beige with 0." class="figure"></a></p>
<p>Clicking any of the class names of the upper left bar displays a
similarity network of BGCs. It may take some time to load the network. A
single network is represented for each GCF of each Gene Cluster Clan
(GCC), which may comprise several GCFs. Each dot represents a BGC. These
dots with bold circles are already described BGCs that have been
recruited from the MiBIG database (Kautsar et al., 2020) because of its
similarity with some BGC of the analysis.</p>
<p><a href="../fig/02_05_03.png">
<img src="../fig/02_05_03.png" alt="BIG-SCAPE similarity network of the complete mix of BGCs
                                  	obtained from the run. A network is represented for each GCF, each dot represents a BGC.
                                  	In this example there are a total of 23 BGCs, of which 6 are singletons, there are 28 links and 11 families.
                                  	Clicking over any of the dots shows the GCF at the right side and allows you to explore it further." class="figure"></a></p>
<p>When you click on a BGC (dot), it appears its GCF at the right. You
can click on the GCF name to see the phylogenetic distances among all
the BGCs comprised by a single GCF. Within the tree there is an arrow
diagram of the genes in the BGCs and the protein domains in the genes.
These trees are useful to prioritize the search of secondary
metabolites, for example, by focusing on the most divergent BGC clade or
those that are distant to already described BGCs.</p>
<p><a href="../fig/02_05_04.png">
<img src="../fig/02_05_04.png" alt="BIG-SCAPE output displaying a tree of phylogenetic distances
                                  	among the BGCs comprised in a GCF. The example shows the GCF 10, comprised by six members.
                                  	Each BGC is represented in the tree by an arrow diagram of the genes and the protein domains in the genes
                                  	corresponding to that cluster." class="figure"></a></p>
<div id="discussion-1-reading-the-gcf-networks" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="discussion-1-reading-the-gcf-networks" class="callout-inner">
<h3 class="callout-title">Discussion 1: Reading the GCF networks</h3>
<div class="callout-content">
<p>What can you conclude about the diversity of BGCs between <em>S.
agalactiae</em> and <em>S. thermophilus</em>? Are they equally diverse?
Do they share GCFs?</p>
<p>Digging deeper: Why do you think the strain <em>S. agalactiae</em>
H36B has a BGC that is not part of the other GCF in the same class?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p><em>S. agalactiae</em> and <em>S. thermophilus</em> seem to have no
diversity in common because there is no GCF with members from both
species. All strains of <em>S.agalactiae</em> seem to have a very
similar set of BGCs. While <em>S. thermophilus</em> has several BGCs
that are not related to any other BGC.</p>
<p>Digging deeper: The BGC of the class PKSother in <em>S.
agalactiae</em> H3B6 has only 3 genes, which are also present in the
BGCs of the other GCF in this class. It is possible that these genes are
in a fragmented part of the assembled genome, but are actually part of a
BGC that is similar to the other ones. And it is also possible that
these 3 genes are present in the genome without being part of a bigger
BGC.</p>
</div>
</div>
</div>
</div>
<div id="exercise-1-using-the-text-output" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="exercise-1-using-the-text-output" class="callout-inner">
<h3 class="callout-title">Exercise 1: Using the text output</h3>
<div class="callout-content">
<p>Use one of the commands from the following list to make a reduced
version of the <code>Network_Annotations_full.tsv</code>, it should only
contain the information of the type of product, the BGC class of each
BGC, and its name. And save it in a file.</p>
<p><code>grep</code> <code>cut</code> <code>ls</code> <code>cat</code>
<code>mv</code></p>
<p><strong>Tip</strong>: The file is inside
<code>network_files/</code>.</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p>Take a look at the content of the file:</p>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="ex">$</span> head <span class="at">-n</span> 3 network_files/2022-06-10_21-27-26_auto/Network_Annotations_Full.tsv</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>BGC    Accession ID    Description    Product Prediction    BiG-SCAPE class    Organism    Taxonomy
Streptococcus_agalactiae_18RS21-AAJO01000016.1.region001    AAJO01000016.1    Streptococcus agalactiae
18RS21    arylpolyene    Others    Streptococcus agalactiae 18RS21    Bacteria,Terrabacteria
group,Firmicutes,Bacilli,Lactobacillales,Streptococcaceae,Streptococcus,Streptococcus agalactiae
Streptococcus_agalactiae_18RS21-AAJO01000043.1.region001    AAJO01000043.1    Streptococcus agalactiae
18RS21    T3PKS    PKSother    Streptococcus agalactiae 18RS21    Bacteria,Terrabacteria
group,Firmicutes,Bacilli,Lactobacillales,Streptococcaceae,Streptococcus,Streptococcus agalactiae</code></pre>
</div>
<p>This table is difficult to read because of the amount of information
it has. The first line has the names of the columns; <code>BGC</code>,
<code>Product Prediction</code> and <code>BiG-SCAPE class</code> are the
ones we are interested in. So we will extract those.</p>
<div class="codewrapper sourceCode" id="cb20">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a><span class="fu">cut</span> <span class="at">-f</span> 1,4,5 network_files/2022-06-10_21-27-26_auto/Network_Annotations_Full.tsv <span class="op">&gt;</span> type_of_BGC.tsv</span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a><span class="fu">cat</span> type_of_BGCs.tsv</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>BGC    Product Prediction    BiG-SCAPE class
Streptococcus_agalactiae_18RS21-AAJO01000016.1.region001    arylpolyene    Others
Streptococcus_agalactiae_18RS21-AAJO01000043.1.region001    T3PKS    PKSother
Streptococcus_agalactiae_515-AAJP01000027.1.region001    arylpolyene    Others
Streptococcus_agalactiae_515-AAJP01000037.1.region001    T3PKS    PKSother
.
.
.</code></pre>
</div>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="cytoscape-visualization-of-the-results">Cytoscape visualization of the results<a class="anchor" aria-label="anchor" href="#cytoscape-visualization-of-the-results"></a>
</h2>
<hr class="half-width">
<p>You can also customize and re-renderize the similarity networks of
your results with Cytoscape (<a href="https://cytoscape.org/" class="external-link uri">https://cytoscape.org/</a>). To do so, you will need some
files included in the output directory of BiG-SCAPE. Both are located in
the same folder. You can choose any folder included in the
Network_files/(date)hybrids_auto/ directory, depending on your interest.
The “Mix/” folder represents the complete network, including all the
BGCs of the analysis. There, you will need the files “mix_c0.30.network”
and “mix_clans_0.30_0.70.tsv”. When you upload the “.network” file, it
is required that you select as “source” the first column and as “target”
the second one. Then, you can upload the “.tsv” and just select the
first column as “source”. Finally, you need to click on Tools -&gt;
merge -&gt; Networks -&gt; Union to combine both GCFs and singletons.
Now you can change colors, labels, etc. according to your specific
requirements.</p>
<div class="section level3">
<h3 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h3>
<ul>
<li><p>Navarro-Muñoz, J.C., Selem-Mojica, N., Mullowney, M.W. et al. “A
computational framework to explore large-scale biosynthetic diversity”.
Nature Chemical Biology (2019).</p></li>
<li><p>Mistry, J., Chuguransky, S., Williams, L., Qureshi, M., Salazar,
G. A., Sonnhammer, E. L., … &amp; Bateman, A. (2021). Pfam: The protein
families database in 2021. Nucleic Acids Research, 49(D1),
D412-D419.</p></li>
<li><p>Kautsar, S. A., Blin, K., Shaw, S., Navarro-Muñoz, J. C.,
Terlouw, B. R., van der Hooft, J. J., … &amp; Medema, M. H. (2020).
MIBiG 2.0: a repository for biosynthetic gene clusters of known
function. Nucleic acids research, 48(D1), D454-D458.</p></li>
</ul>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<ul>
<li>BGC similarity is measured by BiG-SCAPE according to protein domain
content, adjacency and sequence identity.</li>
<li>The <code>gbks</code> of the regions identified by antiSMASH are the
input for BiG-SCAPE.</li>
<li>BiG-SCAPE delivers BGCs similarity networks with which it delimits
Gene Cluster Families and creates a phylogeny of the BGCs in each
GCF.</li>
</ul>
</div>
</div>
</div>
</div>
</section></section><section id="aio-05-bigslice-bigfam"><p>Content from <a href="05-bigslice-bigfam.html">Homologous BGC Clusterization</a></p>
<hr>
<p>Last updated on 2026-02-20 |

        <a href="https://github.com/carpentries-incubator/genome-mining/edit/main/episodes/05-bigslice-bigfam.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 50 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How can I identify Gene Cluster Families?</li>
<li>How can I predict the production of similar metabolites</li>
<li>How can I clusterize BGCs into groups that produce similar
metabolites?</li>
<li>How can I compare the metabolic capability of different bacterial
lineages?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand the purpose of using <code>BiG-SLiCE</code> and
<code>BiG-FAM</code>
</li>
<li>Learn the inputs required to run <code>BiG-SLiCE</code> and
<code>BiG-FAM</code>
</li>
<li>Run an example with our <em>Streptococcus</em> data on both
softwares</li>
<li>Understand the obtained results</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="big-slice-and-big-fam-a-set-of-tools-to-compare-the-microbial-metabolic-diversity">BiG-SLiCE and BiG-FAM a set of tools to compare the microbial
metabolic diversity<a class="anchor" aria-label="anchor" href="#big-slice-and-big-fam-a-set-of-tools-to-compare-the-microbial-metabolic-diversity"></a>
</h2>
<hr class="half-width">
<p>Around the curriculum, we have been learning about the metabolic
capability of bacteria lineages encoded in BGCs. Before we start this
lesson, let’s remember that the products of these clusters are
biomolecules that are (mostly) not essential for life. They take the
role of “tools” that offer some ecological and physiological advantage
to the bacterial lineage producing it.</p>
<p><a href="../fig/02-04-01.png"> &lt;img src=“fig/02-04-01.png”
alt=“Biosynthetic Gene Cluster encoding three different biomolecules,
which might not be essential for survival. In the figure, the first gene
in the cluster synthesizes a shield, the second gene is able to produce
a sword. All together with the third gene, the BGC provides
advantageous”tools” to the organism, represented by a warrior.” /&gt;
</a></p>
<p>The presence/absence of a set of BGCs in a genome, can be associated
to the ecological niches and (micro)environments which the lineage
faces.</p>
<p>The counterpart of this can be taken as how biosynthetically diverse
a bacterial lineage can be due to its environment.</p>
<p><a href="../fig/02-04-02.png"></a></p>
<figure><img src="../fig/02-04-02.png" alt="Three biomolecules are depicted in three different colors; blue, red and purple. These molecules are present in BGCs of diverse bacterial lineages and in turn grouped into Gene Cluster Families (GCFs). GF1 contains three domains related to the blue molecule and three from the red molecule. GF2 contains four domains of each of the molecules, blue, red and purple. GCF3 and GCF4 contain each three domains associated with the purple biomolecule but different from each other. Every GCF produces a different metabolite, here represented as weapons or tools." class="figure mx-auto d-block"></figure><p></p>
<p>To answer some of this questions, we need a way to compare the
metabolic capabilities of a set of diverse bacterial lineages. One way
to do this is to compare the state (identity) of homologous BGCs. The
homologous clusters that share a similar domain architecture and in
consequence will produce a similar metabolite, are grouped into a bigger
group that we call <strong>Gene Cluster Families (GCFs)</strong>.</p>
<p>There are some programs capable of clustering BGCs in GCFs,
<em>e.g.</em><a href="https://github.com/petercim/ClusterFinder" class="external-link">ClusterFinder</a> and
<a href="https://github.com/Merck/deepbgc" class="external-link">DeepBGC</a>.
<code>BiG-SLICE</code> is a software that has been <a href="https://academic.oup.com/gigascience/article/10/1/giaa154/6092777" class="external-link">designed</a>
to do this work in less time, even when dealing with large datasets (1.2
million BGCs). This characteristic makes it a good tool to compare the
metabolic diversity of several bacterial lineages.</p>
<div class="section level3">
<h3 id="the-input-for-big-slice">The input for BiG-SLiCE<a class="anchor" aria-label="anchor" href="#the-input-for-big-slice"></a>
</h3>
<p>Firstly, activate the conda environment, where <code>BiG-SLiCE</code>
has been installed as well as all the softwares required for its
usage:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">$</span> conda activate GenomeMining_Global</span></code></pre>
</div>
<p>You will have now a <code>(GenomeMining)</code> label at the
beginning of the prompt line.</p>
<p>If we seek for help using the <code>--help</code> flag of
<code>bigslice</code>, we can get a first glance at the input that we
need:</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="ex">$</span> bigslice <span class="at">--help</span> <span class="kw">|</span> <span class="fu">head</span> <span class="at">-n20</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>usage: bigslice [-i &lt;folder_path&gt;] [--resume] [--complete] [--threshold T]
            	[--threshold_pct &lt;N&gt;] [--query &lt;folder_path&gt;]
            	[--query_name &lt;name&gt;] [--run_id &lt;id&gt;] [--n_ranks N_RANKS]
            	[-t &lt;N&gt;] [--hmmscan_chunk_size &lt;N&gt;] [--subpfam_chunk_size &lt;N&gt;]
            	[--extraction_chunk_size EXTRACTION_CHUNK_SIZE] [--scratch]
            	[-h] [--program_db_folder PROGRAM_DB_FOLDER] [--version]
            	&lt;output_folder_path&gt;

                        	_________
 ___	_____|\____|\____|\ \____	\
|   \  |   	} 	} 	}  ___)_ _/__
| &gt;  | _--/--|/----|/----|/__/  __||  __|
|   &lt; | ||  __/ (  (| | \___/  /__/|  _|
| &gt;  || || |_ | _)  ) |_ | |\  \__ | |__
|____/|_| \___/|___/|___||_| \____||____| [ Version 1.1.0 ]

Biosynthetic Gene clusters - Super Linear Clustering Engine
(https://github.com/medema-group/bigslice)

positional arguments:</code></pre>
</div>
<p><code>BiG-SLiCE</code> is asking for a folder as the input. It is
important to highlight that the bones of this folder are the BGCs from a
group of genomes that has been obtained by <code>antiSMASH</code>. In
its <a href="https://github.com/medema-group/bigslice" class="external-link">GitHub</a> page,
we can search for an input folder <a href="https://github.com/medema-group/bigslice/tree/master/misc/input_folder_template" class="external-link">template</a>.
The folder needs three main components:</p>
<ul>
<li>
<strong>datasets.tsv:</strong> This is a file that <strong>needs to
have this exact name</strong>. This must be a tabular separated file
where the information of each dataset (BGCs from a group of genomes)
must be specified.</li>
</ul>
<p><a href="../fig/02-04-03.png">
<img src="../fig/02-04-03.png" alt="Example of tsv table composed by five columns and two rows.
                                  	The first row contains the title for each column; # Dataset name, Path to folder, Path to taxonomy, Description." class="figure"></a></p>
<ul>
<li><p><strong>dataset_n:</strong> Each dataset will be composed by the
BGCs of different genomes. We can give <code>BiG-SLiCE</code>
<strong>n</strong> groups of BGCs. The separation of the BGCs in
different datasets is due to the different characteristics that the
bacterial lineages can have. For example, if we have a set of bacterial
populations that came from maize roots and other obtained from tomato
roots, we can put them into dataset_1 and dataset_2
respectively.</p></li>
<li>
<p>**taxonomy_n.tsv:**This is also a tabular separated file that
will carry the taxonomic information of each of the genomes where the
input BGCs were found and the path to reach these BGCs inside the
input-folder:</p>
<ol style="list-style-type: decimal">
<li>Genome folder name (ends with ‘/’)</li>
<li>Kingdom / Domain name</li>
<li>Class name</li>
<li>Order name</li>
<li>Family name</li>
<li>Genus name</li>
<li>Species name</li>
<li>Organism / Strain name</li>
</ol>
</li>
</ul>
<p>Here, we have an example of the structure of the
<code>input-folder</code>:</p>
<p><a href="../fig/02-04-04.png">
<img src="../fig/02-04-04.png" alt="Example of the structure of the input-folder.
                                  	The content of the directory input-folder is displayed in a tree-like format, listing the files and directories inside it." class="figure"></a></p>
</div>
<div class="section level3">
<h3 id="creating-the-input-folder">Creating the input-folder<a class="anchor" aria-label="anchor" href="#creating-the-input-folder"></a>
</h3>
<p>We will create a <code>bigslice</code> folder inside the
<code>results</code> directory.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="ex">$</span> cd ~/gm_workshop/results/genome_mining</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="ex">$</span> mkdir bigslice</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="ex">$</span> cd bigslice</span></code></pre>
</div>
<p>Now that we know what input <code>BiG-SLiCE</code> requires, we will
do our input-folder step by step. First, let’s remember how many genomes
we have:</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="ex">$</span> ls <span class="at">-F</span> ../antismash</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>agalactiae_18RS21/  agalactiae_A909/	agalactiae_COH1/
agalactiae_515/ 	agalactiae_CJB111/  agalactiae_H36B/
athermophilus_LMD-9/  athermophilus_LMG_18311/</code></pre>
</div>
<p>We have six <em>Streptococcus</em> genomes from the original paper,
and two<br>
public genomes. We will allocate their respective BGC <code>.gbk</code>s
in two <code>dataset</code>s, one for the genomes from Tettelin <em>et
al.</em> paper, and the other for the public ones.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="ex">$</span> mkdir input-folder</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="ex">$</span> mkdir input-folder/dataset_1</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="ex">$</span> mkdir input-folder/dataset_2</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="ex">$</span> mkdir input-folder/taxonomy</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a><span class="ex">$</span> tree</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>.
└── input-folder
	├── dataset_1
	├── dataset_2
	└── taxonomy

4 directories, 0 files</code></pre>
</div>
<p>Now we will create the folders for the BGCs of the dataset_1 genomes.
We will take advantage of the strain name of our <em>Streptococcus
agalactiae</em> lineages to create their folders:</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="ex">$</span> for i in 18RS21 COH1 515 H36B A909 CJB111<span class="kw">;</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>  <span class="cf">do</span> <span class="fu">mkdir</span> input-folder/dataset_1/Streptococcus_agalactiae_<span class="va">$i</span><span class="kw">;</span></span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>  <span class="cf">done</span></span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a><span class="ex">$</span> tree <span class="at">-F</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>.
└── input-folder/
	├── dataset_1/
	│   ├── Streptococcus_agalactiae_18RS21/
	│   ├── Streptococcus_agalactiae_515/
	│   ├── Streptococcus_agalactiae_A909/
	│   ├── Streptococcus_agalactiae_CJB111/
	│   ├── Streptococcus_agalactiae_COH1/
	│   └── Streptococcus_agalactiae_H36B/
	├── dataset_2/
	└── taxonomy/

10 directories, 0 files</code></pre>
</div>
<p>We will do the same for the public genomes:</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="ex">$</span> for i in LMD-9 LMG_18311<span class="kw">;</span></span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>  <span class="cf">do</span> <span class="fu">mkdir</span> input-folder/dataset_2/Streptococcus_thermophilus_<span class="va">$i</span><span class="kw">;</span></span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a>  <span class="cf">done</span></span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="ex">$</span> tree <span class="at">-F</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="bu">.</span></span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a><span class="ex">└──</span> input-folder/</span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a>	<span class="ex">├──</span> dataset_1/</span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a>	<span class="ex">│</span>   ├── Streptococcus_agalactiae_18RS21/</span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a>	<span class="ex">│</span>   ├── Streptococcus_agalactiae_515/</span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a>	<span class="ex">│</span>   ├── Streptococcus_agalactiae_A909/</span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a>	<span class="ex">│</span>   ├── Streptococcus_agalactiae_CJB111/</span>
<span id="cb12-8"><a href="#cb12-8" tabindex="-1"></a>	<span class="ex">│</span>   ├── Streptococcus_agalactiae_COH1/</span>
<span id="cb12-9"><a href="#cb12-9" tabindex="-1"></a>	<span class="ex">│</span>   └── Streptococcus_agalactiae_H36B/</span>
<span id="cb12-10"><a href="#cb12-10" tabindex="-1"></a>	<span class="ex">├──</span> dataset_2/</span>
<span id="cb12-11"><a href="#cb12-11" tabindex="-1"></a>	<span class="ex">│</span>   ├── Streptococcus_thermophilus_LMD-9/</span>
<span id="cb12-12"><a href="#cb12-12" tabindex="-1"></a>	<span class="ex">│</span>   └── Streptococcus_thermophilus_LMG_18311/</span>
<span id="cb12-13"><a href="#cb12-13" tabindex="-1"></a>	<span class="ex">└──</span> taxonomy/</span>
<span id="cb12-14"><a href="#cb12-14" tabindex="-1"></a></span>
<span id="cb12-15"><a href="#cb12-15" tabindex="-1"></a><span class="ex">12</span> directories, 0 files</span></code></pre>
</div>
<p>Next, we will copy the BGCs <code>.gbk</code> files from the
dataset_1 genomes from each <code>AntiSMASH</code> output directory to
its respective folder .</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="ex">$</span> ls input-folder/dataset_1/ <span class="kw">|</span> <span class="cf">while</span> <span class="bu">read</span> <span class="va">line</span><span class="kw">;</span></span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>  <span class="cf">do</span> <span class="fu">cp</span> ../antismash/output/<span class="va">$line</span>/<span class="pp">*</span>region<span class="pp">*</span>.gbk input-folder/dataset_1/<span class="va">$line</span><span class="kw">;</span></span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a>  <span class="cf">done</span></span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a><span class="ex">$</span> tree <span class="at">-F</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>.
└── input-folder/
	├── dataset_1/
	│   ├── Streptococcus_agalactiae_18RS21/
	│   │   ├── AAJO01000016.1.region001.gbk
	│   │   ├── AAJO01000043.1.region001.gbk
	│   │   └── AAJO01000226.1.region001.gbk
	│   ├── Streptococcus_agalactiae_515/
	│   │   ├── AAJP01000027.1.region001.gbk
	│   │   └── AAJP01000037.1.region001.gbk
	│   ├── Streptococcus_agalactiae_A909/
	│   │   ├── CP000114.1.region001.gbk
	│   │   └── CP000114.1.region002.gbk
	│   ├── Streptococcus_agalactiae_CJB111/
	│   │   ├── AAJQ01000010.1.region001.gbk
	│   │   └── AAJQ01000025.1.region001.gbk
	│   ├── Streptococcus_agalactiae_COH1/
	│   │   ├── AAJR01000002.1.region001.gbk
	│   │   └── AAJR01000044.1.region001.gbk
	│   └── Streptococcus_agalactiae_H36B/
	│   	├── AAJS01000020.1.region001.gbk
	│   	└── AAJS01000117.1.region001.gbk
	├── dataset_2/
	│   ├── Streptococcus_thermophilus_LMD-9/
	│   └── Streptococcus_thermophilus_LMG_18311/
	└── taxonomy/

12 directories, 13 files</code></pre>
</div>
<p>Again, we will do the same for the BGCs <code>.gbk</code>s of the
dataset_2:</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="ex">$</span> ls input-folder/dataset_2/ <span class="kw">|</span> <span class="cf">while</span> <span class="bu">read</span> <span class="va">line</span><span class="kw">;</span></span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a>  <span class="cf">do</span> <span class="fu">cp</span> ../antismash/output/<span class="va">$line</span>/<span class="pp">*</span>region<span class="pp">*</span>.gbk input-folder/dataset_2/<span class="va">$line</span><span class="kw">;</span></span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a>  <span class="cf">done</span></span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a><span class="ex">$</span> tree <span class="at">-F</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="bu">.</span></span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a><span class="ex">└──</span> input-folder/</span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a>	<span class="ex">├──</span> dataset_1/</span>
<span id="cb16-4"><a href="#cb16-4" tabindex="-1"></a>	<span class="ex">│</span>   ├── Streptococcus_agalactiae_18RS21/</span>
<span id="cb16-5"><a href="#cb16-5" tabindex="-1"></a>	<span class="ex">│</span>   │   ├── AAJO01000016.1.region001.gbk</span>
<span id="cb16-6"><a href="#cb16-6" tabindex="-1"></a>	<span class="ex">│</span>   │   ├── AAJO01000043.1.region001.gbk</span>
<span id="cb16-7"><a href="#cb16-7" tabindex="-1"></a>	<span class="ex">│</span>   │   └── AAJO01000226.1.region001.gbk</span>
<span id="cb16-8"><a href="#cb16-8" tabindex="-1"></a>	<span class="ex">│</span>   ├── Streptococcus_agalactiae_515/</span>
<span id="cb16-9"><a href="#cb16-9" tabindex="-1"></a>	<span class="ex">│</span>   │   ├── AAJP01000027.1.region001.gbk</span>
<span id="cb16-10"><a href="#cb16-10" tabindex="-1"></a>	<span class="ex">│</span>   │   └── AAJP01000037.1.region001.gbk</span>
<span id="cb16-11"><a href="#cb16-11" tabindex="-1"></a>	<span class="ex">│</span>   ├── Streptococcus_agalactiae_A909/</span>
<span id="cb16-12"><a href="#cb16-12" tabindex="-1"></a>	<span class="ex">│</span>   │   ├── CP000114.1.region001.gbk</span>
<span id="cb16-13"><a href="#cb16-13" tabindex="-1"></a>	<span class="ex">│</span>   │   └── CP000114.1.region002.gbk</span>
<span id="cb16-14"><a href="#cb16-14" tabindex="-1"></a>	<span class="ex">│</span>   ├── Streptococcus_agalactiae_CJB111/</span>
<span id="cb16-15"><a href="#cb16-15" tabindex="-1"></a>	<span class="ex">│</span>   │   ├── AAJQ01000010.1.region001.gbk</span>
<span id="cb16-16"><a href="#cb16-16" tabindex="-1"></a>	<span class="ex">│</span>   │   └── AAJQ01000025.1.region001.gbk</span>
<span id="cb16-17"><a href="#cb16-17" tabindex="-1"></a>	<span class="ex">│</span>   ├── Streptococcus_agalactiae_COH1/</span>
<span id="cb16-18"><a href="#cb16-18" tabindex="-1"></a>	<span class="ex">│</span>   │   ├── AAJR01000002.1.region001.gbk</span>
<span id="cb16-19"><a href="#cb16-19" tabindex="-1"></a>	<span class="ex">│</span>   │   └── AAJR01000044.1.region001.gbk</span>
<span id="cb16-20"><a href="#cb16-20" tabindex="-1"></a>	<span class="ex">│</span>   └── Streptococcus_agalactiae_H36B/</span>
<span id="cb16-21"><a href="#cb16-21" tabindex="-1"></a>	<span class="ex">│</span>   	├── AAJS01000020.1.region001.gbk</span>
<span id="cb16-22"><a href="#cb16-22" tabindex="-1"></a>	<span class="ex">│</span>   	└── AAJS01000117.1.region001.gbk</span>
<span id="cb16-23"><a href="#cb16-23" tabindex="-1"></a>	<span class="ex">├──</span> dataset_2/</span>
<span id="cb16-24"><a href="#cb16-24" tabindex="-1"></a>	<span class="ex">│</span>   ├── Streptococcus_thermophilus_LMD-9/</span>
<span id="cb16-25"><a href="#cb16-25" tabindex="-1"></a>	<span class="ex">│</span>   │   ├── CP000419.1.region001.gbk</span>
<span id="cb16-26"><a href="#cb16-26" tabindex="-1"></a>	<span class="ex">│</span>   │   ├── CP000419.1.region002.gbk</span>
<span id="cb16-27"><a href="#cb16-27" tabindex="-1"></a>	<span class="ex">│</span>   │   ├── CP000419.1.region003.gbk</span>
<span id="cb16-28"><a href="#cb16-28" tabindex="-1"></a>	<span class="ex">│</span>   │   ├── CP000419.1.region004.gbk</span>
<span id="cb16-29"><a href="#cb16-29" tabindex="-1"></a>	<span class="ex">│</span>   │   └── CP000419.1.region005.gbk</span>
<span id="cb16-30"><a href="#cb16-30" tabindex="-1"></a>	<span class="ex">│</span>   └── Streptococcus_thermophilus_LMG_18311/</span>
<span id="cb16-31"><a href="#cb16-31" tabindex="-1"></a>	<span class="ex">│</span>   	├── NC_006448.1.region001.gbk</span>
<span id="cb16-32"><a href="#cb16-32" tabindex="-1"></a>	<span class="ex">│</span>   	├── NC_006448.1.region002.gbk</span>
<span id="cb16-33"><a href="#cb16-33" tabindex="-1"></a>	<span class="ex">│</span>   	├── NC_006448.1.region003.gbk</span>
<span id="cb16-34"><a href="#cb16-34" tabindex="-1"></a>	<span class="ex">│</span>   	├── NC_006448.1.region004.gbk</span>
<span id="cb16-35"><a href="#cb16-35" tabindex="-1"></a>	<span class="ex">│</span>   	├── NC_006448.1.region005.gbk</span>
<span id="cb16-36"><a href="#cb16-36" tabindex="-1"></a>	<span class="ex">│</span>   	└── NC_006448.1.region006.gbk</span>
<span id="cb16-37"><a href="#cb16-37" tabindex="-1"></a>	<span class="ex">└──</span> taxonomy/</span>
<span id="cb16-38"><a href="#cb16-38" tabindex="-1"></a></span>
<span id="cb16-39"><a href="#cb16-39" tabindex="-1"></a><span class="ex">12</span> directories, 24 files</span></code></pre>
</div>
<div id="exercise-1.-input-folder-structure" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="exercise-1.-input-folder-structure" class="callout-inner">
<h3 class="callout-title">Exercise 1. Input-folder structure</h3>
<div class="callout-content">
<p>As we have seen, the structure of the input folder for BiG-SLiCE is
quite difficult to get. Imagine “Sekiro” wants to copy the directory
structure to use it for future BiG-SLiCE inputs. Consider the following
directory structure:</p>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  └── input_folder/               	
   	├── datasets.tsv      	
   	├── dataset_1/
   	|   ├── genome_1A/
   	|   |   ├── genome_1A.region001.gbk
   	|   |   └── genome_1A.region002.gbk
   	|   ├── genome_1B/
   	|   |   ├── genome_1B.region001.gbk
   	|   |   └── genome_1B.region002.gbk
   	├── dataset_2/
   	|   ├── genome_2A/
   	|   |   ├── genome_2A.region001.gbk
   	|   |   └── genome_2A.region002.gbk
   	|   ├── genome_2B/
   	|   |   ├── genome_2B.region001.gbk
   	|   |   └── genome_2B.region002.gbk
   	└── taxonomy/
       	├── taxonomy_dataset_1.tsv
       	└── taxonomy_dataset_2.tsv                	 </code></pre>
</div>
<p>Which would be the commands to copy the directory structure without
the files (just the folders)? <br> a)</p>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="ex">$</span> cp <span class="at">-r</span> input_folder input_folder_copy</span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a><span class="ex">$</span> cd input_folder_copy</span>
<span id="cb18-3"><a href="#cb18-3" tabindex="-1"></a><span class="ex">$</span> rm <span class="at">-r</span></span></code></pre>
</div>
<ol start="2" style="list-style-type: lower-alpha"><li>
</li></ol>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a><span class="ex">$</span> mv <span class="at">-r</span> input_folder input_folder_copy</span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a><span class="ex">$</span> rm input_folder_copy/dataset_1/genome_1A/<span class="pp">*</span> input_folder_copy/dataset_1/genome_1B/<span class="pp">*</span></span>
<span id="cb19-3"><a href="#cb19-3" tabindex="-1"></a><span class="ex">$</span> rm input_folder_copy/dataset_2/genome_2A/<span class="pp">*</span> input_folder_copy/dataset_2/genome_2B/<span class="pp">*</span></span>
<span id="cb19-4"><a href="#cb19-4" tabindex="-1"></a><span class="ex">$</span> rm input_folder_copy/taxonomy/<span class="pp">*</span></span></code></pre>
</div>
<ol start="3" style="list-style-type: lower-alpha"><li>
</li></ol>
<div class="codewrapper sourceCode" id="cb20">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a><span class="ex">$</span> cp <span class="at">-r</span> input_folder input_folder_copy</span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a><span class="ex">$</span> rm input_folder_copy/dataset_1/genome_1A/<span class="pp">*</span> input_folder_copy/dataset_1/genome_1B/<span class="pp">*</span></span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a><span class="ex">$</span> rm input_folder_copy/dataset_2/genome_2A/<span class="pp">*</span> input_folder_copy/dataset_2/genome_2B/<span class="pp">*</span></span>
<span id="cb20-4"><a href="#cb20-4" tabindex="-1"></a><span class="ex">$</span> rm input_folder_copy/taxonomy/<span class="pp">*</span></span></code></pre>
</div>
<ol start="4" style="list-style-type: lower-alpha"><li>
</li></ol>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a><span class="ex">$</span> cp <span class="at">-r</span> input_folder input_folder_copy</span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a><span class="ex">$</span> rm input_folder_copy/dataset_1/genome_1A/<span class="pp">*</span> /genome_1B/<span class="pp">*</span></span>
<span id="cb21-3"><a href="#cb21-3" tabindex="-1"></a><span class="ex">$</span> rm input_folder_copy/dataset_2/genome_2A/<span class="pp">*</span> /genome_2B/<span class="pp">*</span></span>
<span id="cb21-4"><a href="#cb21-4" tabindex="-1"></a><span class="ex">$</span> rm input_folder_copy/taxonomy/<span class="pp">*</span></span></code></pre>
</div>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p>First we need to copy the whole directory to make an
input_folder_copy. After that we want to erase the files inside the
directories dataset_1/genome_1A, dataset_1/genome_1B,
dataset_2/genome_2A, dataset2/genome_2B and taxonomy/ <br> c)</p>
<div class="codewrapper sourceCode" id="cb22">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a><span class="ex">$</span> cp <span class="at">-r</span> input_folder input_folder_copy</span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a><span class="ex">$</span> rm input_folder_copy/dataset_1/genome_1A/<span class="pp">*</span> input_folder_copy/dataset_1/genome_1B/<span class="pp">*</span></span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a><span class="ex">$</span> rm input_folder_copy/dataset_2/genome_2A/<span class="pp">*</span> input_folder_copy/dataset_2/genome_2B/<span class="pp">*</span></span>
<span id="cb22-4"><a href="#cb22-4" tabindex="-1"></a><span class="ex">$</span> rm input_folder_copy/taxonomy/<span class="pp">*</span></span></code></pre>
</div>
</div>
</div>
</div>
</div>
<p><strong>datasets.tsv file - The <a href="https://lotr.fandom.com/wiki/Silmarils" class="external-link">silmaril</a> of
BiG-SLiCE</strong></p>
<p>The next step is to create the main file
<strong>datasets.tsv</strong>. First we put the first lines to this
file. In each of these <code>.tsv</code> files, we can put lines that
begin with <code>#</code>. This will not be read by the code, and it can
be of help to know which information is located in those files. We will
use the command <code>echo</code> with the <code>-e</code> flag that
will enable us to put tab separations between the text:</p>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a><span class="ex">$</span> cd input-folder</span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a><span class="ex">$</span> echo <span class="at">-e</span> <span class="st">"#Dataset name""\t""Path to folder""\t""Path to taxonomy""\t""Description"</span> <span class="op">&gt;</span> datasets.tsv</span>
<span id="cb23-3"><a href="#cb23-3" tabindex="-1"></a><span class="ex">$</span> more datasets.tsv</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code><span><span class="co">#Dataset name   Path to folder  Path to taxonomy      Description</span></span></code></pre>
</div>
<p>We will use <code>echo</code> <em>One more time</em> to put the
information of our two datasets. But this time, we will indicate bash
that we want to insert this new line in the existing
<code>datasets.tsv</code> with the <code>&gt;&gt;</code> option.</p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a><span class="ex">$</span> echo <span class="at">-e</span> <span class="st">"dataset_1""\t""dataset_1/""\t""taxonomy/dataset_1_taxonomy.tsv""\t""Tetteling genomes"</span> <span class="op">&gt;&gt;</span> datasets.tsv</span>
<span id="cb25-2"><a href="#cb25-2" tabindex="-1"></a><span class="ex">$</span> echo <span class="at">-e</span> <span class="st">"dataset_2""\t""dataset_2/""\t""taxonomy/dataset_2_taxonomy.tsv""\t""Public genomes"</span> <span class="op">&gt;&gt;</span> datasets.tsv</span>
<span id="cb25-3"><a href="#cb25-3" tabindex="-1"></a><span class="ex">$</span> more datasets.tsv</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>#Dataset name   Path to folder  Path to taxonomy    	Description
dataset_1   	dataset_1/  	taxonomy/dataset_1_taxonomy.tsv Tetteling genomes
dataset_2   	dataset_2/  	taxonomy/dataset_2_taxonomy.tsv Public genomes</code></pre>
</div>
<p>The main file of the input-folder is finished!</p>
<div class="section level4">
<h4 id="the-taxonomy-of-the-input-folder">The taxonomy of the input-folder<a class="anchor" aria-label="anchor" href="#the-taxonomy-of-the-input-folder"></a>
</h4>
<p>We also need to specify the taxonomic assignation of each of the
input genomes BGC’s. We will build one <code>dataset_taxonomy.tsv</code>
file for each one of our datasets, <em>i.e.</em> two in total. Firstly,
create the header of each one of these files using
<code>echo</code>:</p>
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a><span class="ex">$</span> echo <span class="at">-e</span> <span class="st">"#Genome folder""\t"</span>Kingdom<span class="st">"\t"</span>Phylum<span class="st">"\t"</span>Class<span class="st">"\t"</span>Order<span class="st">"\t"</span>Family<span class="st">"\t"</span>Genus<span class="st">"\t""Species Organism"</span> <span class="op">&gt;</span> taxonomy/dataset_1_taxonomy.tsv</span>
<span id="cb27-2"><a href="#cb27-2" tabindex="-1"></a><span class="ex">$</span> echo <span class="at">-e</span> <span class="st">"#Genome folder""\t"</span>Kingdom<span class="st">"\t"</span>Phylum<span class="st">"\t"</span>Class<span class="st">"\t"</span>Order<span class="st">"\t"</span>Family<span class="st">"\t"</span>Genus<span class="st">"\t""Species Organism"</span> <span class="op">&gt;</span> taxonomy/dataset_2_taxonomy.tsv</span>
<span id="cb27-3"><a href="#cb27-3" tabindex="-1"></a></span>
<span id="cb27-4"><a href="#cb27-4" tabindex="-1"></a><span class="ex">$</span> more taxonomy/dataset_1_taxonomy.tsv</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code><span><span class="co">#Genome folder  Kingdom Phylum  Class   Order   Family  Genus   Species Organism</span></span></code></pre>
</div>
<p>There are several ways to obtain the taxonomic data. One of them is
searching for each lineage taxonomic identification on <a href="https://www.ncbi.nlm.nih.gov/guide/taxonomy/" class="external-link">NCBI</a>. There is
also a way to get this information from our <code>.gbk</code> files:</p>
<div class="codewrapper sourceCode" id="cb29">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb29-1"><a href="#cb29-1" tabindex="-1"></a><span class="ex">$</span> head <span class="at">-n20</span> ../../annotated/Streptococcus_agalactiae_18RS21.gbk</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>LOCUS   	AAJO01000169.1      	2501 bp	DNA 	linear   UNK
DEFINITION  Streptococcus agalactiae 18RS21
ACCESSION   AAJO01000169.1
KEYWORDS	.
SOURCE  	Streptococcus agalactiae 18RS21.
  ORGANISM  Streptococcus agalactiae 18RS21
        	Bacteria; Terrabacteria group; Firmicutes; Bacilli;
        	Lactobacillales; Streptococcaceae; Streptococcus; Streptococcus
        	agalactiae.
FEATURES         	Location/Qualifiers
 	source      	1..2501
                 	/mol_type="genomic DNA"
                 	/db_xref="taxon:342613"
                 	/genome_md5=""
                 	/project="nselem35_342613"
                 	/genome_id="342613.37"
                 	/organism="Streptococcus agalactiae 18RS21"
 	CDS         	1..213
                 	/db_xref="SEED:fig|342613.37.peg.1973"
                 	/db_xref="GO:0003735"</code></pre>
</div>
<p>We will use again <code>echo</code> to write this information into
our <code>dataset_taxonomy.tsv</code>. This is going to be a little long
because we have 8 genomes and we will need to write 8 <code>echo</code>
lines, but bear with us. We are almost there!</p>
<div class="codewrapper sourceCode" id="cb31">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a><span class="ex">$</span> echo <span class="at">-e</span> <span class="st">"Streptococcus_agalactiae_18RS21/""\t"</span>Bacteria<span class="st">"\t"</span>Firmicutes<span class="st">"\t"</span>Bacilli<span class="st">"\t"</span>Lactobacillales<span class="st">"\t"</span>Streptococcaceae<span class="st">"\t"</span>Streptococcus<span class="st">"\t""Streptococcus agalactiae 18RS21"</span> <span class="op">&gt;&gt;</span> taxonomy/dataset_1_taxonomy.tsv</span>
<span id="cb31-2"><a href="#cb31-2" tabindex="-1"></a></span>
<span id="cb31-3"><a href="#cb31-3" tabindex="-1"></a><span class="ex">$</span> echo <span class="at">-e</span> <span class="st">"Streptococcus_agalactiae_515/""\t"</span>Bacteria<span class="st">"\t"</span>Firmicutes<span class="st">"\t"</span>Bacilli<span class="st">"\t"</span>Lactobacillales<span class="st">"\t"</span>Streptococcaceae<span class="st">"\t"</span>Streptococcus<span class="st">"\t""Streptococcus agalactiae 515"</span> <span class="op">&gt;&gt;</span> taxonomy/dataset_1_taxonomy.tsv</span>
<span id="cb31-4"><a href="#cb31-4" tabindex="-1"></a></span>
<span id="cb31-5"><a href="#cb31-5" tabindex="-1"></a><span class="ex">$</span> echo <span class="at">-e</span> <span class="st">"Streptococcus_agalactiae_A909/""\t"</span>Bacteria<span class="st">"\t"</span>Firmicutes<span class="st">"\t"</span>Bacilli<span class="st">"\t"</span>Lactobacillales<span class="st">"\t"</span>Streptococcaceae<span class="st">"\t"</span>Streptococcus<span class="st">"\t""Streptococcus agalactiae A909"</span> <span class="op">&gt;&gt;</span> taxonomy/dataset_1_taxonomy.tsv</span>
<span id="cb31-6"><a href="#cb31-6" tabindex="-1"></a></span>
<span id="cb31-7"><a href="#cb31-7" tabindex="-1"></a><span class="ex">$</span> echo <span class="at">-e</span> <span class="st">"Streptococcus_agalactiae_CJB111/""\t"</span>Bacteria<span class="st">"\t"</span>Firmicutes<span class="st">"\t"</span>Bacilli<span class="st">"\t"</span>Lactobacillales<span class="st">"\t"</span>Streptococcaceae<span class="st">"\t"</span>Streptococcus<span class="st">"\t""Streptococcus agalactiae CJB111"</span> <span class="op">&gt;&gt;</span> taxonomy/dataset_1_taxonomy.tsv</span>
<span id="cb31-8"><a href="#cb31-8" tabindex="-1"></a></span>
<span id="cb31-9"><a href="#cb31-9" tabindex="-1"></a><span class="ex">$</span> echo <span class="at">-e</span> <span class="st">"Streptococcus_agalactiae_COH1/""\t"</span>Bacteria<span class="st">"\t"</span>Firmicutes<span class="st">"\t"</span>Bacilli<span class="st">"\t"</span>Lactobacillales<span class="st">"\t"</span>Streptococcaceae<span class="st">"\t"</span>Streptococcus<span class="st">"\t""Streptococcus agalactiae COH1"</span> <span class="op">&gt;&gt;</span> taxonomy/dataset_1_taxonomy.tsv</span>
<span id="cb31-10"><a href="#cb31-10" tabindex="-1"></a></span>
<span id="cb31-11"><a href="#cb31-11" tabindex="-1"></a><span class="ex">$</span> echo <span class="at">-e</span> <span class="st">"Streptococcus_agalactiae_H36B/""\t"</span>Bacteria<span class="st">"\t"</span>Firmicutes<span class="st">"\t"</span>Bacilli<span class="st">"\t"</span>Lactobacillales<span class="st">"\t"</span>Streptococcaceae<span class="st">"\t"</span>Streptococcus<span class="st">"\t""Streptococcus agalactiae H36B"</span> <span class="op">&gt;&gt;</span> taxonomy/dataset_1_taxonomy.tsv</span>
<span id="cb31-12"><a href="#cb31-12" tabindex="-1"></a></span>
<span id="cb31-13"><a href="#cb31-13" tabindex="-1"></a><span class="ex">$</span> more taxonomy/dataset_1_taxonomy.tsv</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>#Genome folder   Kingdom Phylum  Class   Order   Family  Genus   Species Organism
Streptococcus_agalactiae_18RS21/    	Bacteria    	Firmicutes  	Bacilli Lactobacillales Strept
ococcaceae  	Streptococcus   Streptococcus agalactiae 18RS21
Streptococcus_agalactiae_515/   Bacteria    	Firmicutes  	Bacilli Lactobacillales Streptococcace
ae  	Streptococcus   Streptococcus agalactiae 515
Streptococcus_agalactiae_A909/  Bacteria    	Firmicutes  	Bacilli Lactobacillales Streptococcace
ae  	Streptococcus   Streptococcus agalactiae A909
Streptococcus_agalactiae_CJB111/    	Bacteria    	Firmicutes  	Bacilli Lactobacillales Strept
ococcaceae  	Streptococcus   Streptococcus agalactiae CJB111
Streptococcus_agalactiae_COH1/  Bacteria    	Firmicutes  	Bacilli Lactobacillales Streptococcace
ae  	Streptococcus   Streptococcus agalactiae COH1
Streptococcus_agalactiae_H36B/  Bacteria    	Firmicutes  	Bacilli Lactobacillales Streptococcace
ae  	Streptococcus   Streptococcus agalactiae H36B</code></pre>
</div>
<p>We will do the same for the <code>dataset_2_taxonomy.tsv</code>
file:</p>
<div class="codewrapper sourceCode" id="cb33">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a><span class="ex">$</span> echo <span class="at">-e</span> <span class="st">"Streptococcus_thermophilus_LMD-9/""\t"</span>Bacteria<span class="st">"\t"</span>Firmicutes<span class="st">"\t"</span>Bacilli<span class="st">"\t"</span>Lactobacillales<span class="st">"\t"</span>Streptococcaceae<span class="st">"\t"</span>Streptococcus<span class="st">"\t""Streptococcus thermophilus LMD-9"</span> <span class="op">&gt;&gt;</span> taxonomy/dataset_2_taxonomy.tsv</span>
<span id="cb33-2"><a href="#cb33-2" tabindex="-1"></a></span>
<span id="cb33-3"><a href="#cb33-3" tabindex="-1"></a><span class="ex">$</span> echo <span class="at">-e</span> <span class="st">"Streptococcus_thermophilus_LMG_18311/""\t"</span>Bacteria<span class="st">"\t"</span>Firmicutes<span class="st">"\t"</span>Bacilli<span class="st">"\t"</span>Lactobacillales<span class="st">"\t"</span>Streptococcaceae<span class="st">"\t"</span>Streptococcus<span class="st">"\t""Streptococcus thermophilus LMG 18311"</span> <span class="op">&gt;&gt;</span> taxonomy/dataset_2_taxonomy.tsv</span>
<span id="cb33-4"><a href="#cb33-4" tabindex="-1"></a></span>
<span id="cb33-5"><a href="#cb33-5" tabindex="-1"></a><span class="ex">$</span> more taxonomy/dataset_2_taxonomy.tsv</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>#Genome folder   Kingdom Phylum  Class   Order   Family  Genus   Species Organism
Streptococcus_thermophilus_LMD-9/   	Bacteria    	Firmicutes  	Bacilli Lactobacillales Strept
ococcaceae  	Streptococcus   Streptococcus thermophilus LMD-9
Streptococcus_thermophilus_LMG_18311/   Bacteria    	Firmicutes  	Bacilli Lactobacillales Strept
ococcaceae  	Streptococcus   Streptococcus thermophilus LMG 18311</code></pre>
</div>
<p>Now, we have the organization of our <code>input-folder</code>
complete</p>
<div class="codewrapper sourceCode" id="cb35">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb35-1"><a href="#cb35-1" tabindex="-1"></a><span class="ex">$</span> cd ..</span>
<span id="cb35-2"><a href="#cb35-2" tabindex="-1"></a><span class="ex">$</span> tree <span class="at">-F</span> input-folder/</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>input-folder/
├── dataset_1/
│   ├── Streptococcus_agalactiae_18RS21/
│   │   ├── AAJO01000016.1.region001.gbk
│   │   ├── AAJO01000043.1.region001.gbk
│   │   └── AAJO01000226.1.region001.gbk
│   ├── Streptococcus_agalactiae_515/
│   │   ├── AAJP01000027.1.region001.gbk
│   │   └── AAJP01000037.1.region001.gbk
│   ├── Streptococcus_agalactiae_A909/
│   │   ├── CP000114.1.region001.gbk
│   │   └── CP000114.1.region002.gbk
│   ├── Streptococcus_agalactiae_CJB111/
│   │   ├── AAJQ01000010.1.region001.gbk
│   │   └── AAJQ01000025.1.region001.gbk
│   ├── Streptococcus_agalactiae_COH1/
│   │   ├── AAJR01000002.1.region001.gbk
│   │   └── AAJR01000044.1.region001.gbk
│   └── Streptococcus_agalactiae_H36B/
│   	├── AAJS01000020.1.region001.gbk
│   	└── AAJS01000117.1.region001.gbk
├── dataset_2/
│   ├── Streptococcus_thermophilus_LMD-9/
│   │   ├── CP000419.1.region001.gbk
│   │   ├── CP000419.1.region002.gbk
│   │   ├── CP000419.1.region003.gbk
│   │   ├── CP000419.1.region004.gbk
│   │   └── CP000419.1.region005.gbk
│   └── Streptococcus_thermophilus_LMG_18311/
│   	├── NC_006448.1.region001.gbk
│   	├── NC_006448.1.region002.gbk
│   	├── NC_006448.1.region003.gbk
│   	├── NC_006448.1.region004.gbk
│   	├── NC_006448.1.region005.gbk
│   	└── NC_006448.1.region006.gbk
├── datasets.tsv
└── taxonomy/
	├── dataset_1_taxonomy.tsv
	└── dataset_2_taxonomy.tsv

11 directories, 27 files</code></pre>
</div>
<div id="exercise-2.-taxonomic-information" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="exercise-2.-taxonomic-information" class="callout-inner">
<h3 class="callout-title">Exercise 2. Taxonomic information</h3>
<div class="callout-content">
<p>If you use the head -n20 on one of your .gbk files, you will obtain
an output like this</p>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>LOCUS   	AAJO01000169.1      	2501 bp	DNA 	linear   UNK
DEFINITION  Streptococcus agalactiae 18RS21
ACCESSION   AAJO01000169.1
KEYWORDS	.
SOURCE  	Streptococcus agalactiae 18RS21.
 ORGANISM  Streptococcus agalactiae 18RS21
       	Bacteria; Terrabacteria group; Firmicutes; Bacilli;
       	Lactobacillales; Streptococcaceae; Streptococcus; Streptococcus
       	agalactiae.
FEATURES         	Location/Qualifiers
	source      	1..2501
                	/mol_type="genomic DNA"
                	/db_xref="taxon:342613"
                	/genome_md5=""
                	/project="nselem35_342613"
                	/genome_id="342613.37"
                	/organism="Streptococcus agalactiae 18RS21"
	CDS         	1..213
                	/db_xref="SEED:fig|342613.37.peg.1973"
                	/db_xref="GO:0003735"</code></pre>
</div>
<p>As you can see, in the gbk file we can find the taxonomic information
that we need to fill the taxonomic information of each genome. Fill in
the blank spaces to complete the command that would get the organism
information from the .gbk file.</p>
<p style="text-align: center;">
$ grep ___________ -m1 ____________ Streptococcus_agalactiae_18RS21.gbk
</p>
<p style="text-align: center;">
-B 3       -A 3       -a       ‘Streptococcus’       ‘ORGANISM’      
‘gbk’
</p>
<p>To obtain the next result:</p>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code> ORGANISM  Streptococcus agalactiae 18RS21
       	Bacteria; Terrabacteria group; Firmicutes; Bacilli;
       	Lactobacillales; Streptococcaceae; Streptococcus; Streptococcus
       	agalactiae.</code></pre>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<p>You can use the <code>grep --help</code> command to get information
about the available options for the <code>grep</code> command</p>
</div>
</div>
</div>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<p>We need to use the grep command to search for the word “ORGANISM” in
the file. Once located we just want to get the first instance of the
word that is why we use the -m1 flag. Then we want the 3 next lines of
the file after the instance, because that’s where the information is. So
the final command would be:</p>
<div class="codewrapper sourceCode" id="cb39">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb39-1"><a href="#cb39-1" tabindex="-1"></a><span class="ex">$</span> grep <span class="st">'ORGANISM'</span> <span class="at">-m1</span> <span class="at">-A3</span> Streptococcus_agalactiae_18RS21.gbk</span></code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="running-big-slice">Running BiG-SLiCE<a class="anchor" aria-label="anchor" href="#running-big-slice"></a>
</h3>
<p>We are ready to run <code>BiG-SLICE</code>. After we type the next
line of command, it will take close to 3 minutes long to end the
process:</p>
<div class="codewrapper sourceCode" id="cb40">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb40-1"><a href="#cb40-1" tabindex="-1"></a><span class="ex">$</span> bigslice <span class="at">-i</span> input-folder output-bigslice</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>...
...
BiG-SLiCE run complete!</code></pre>
</div>
<p>In order to see the results, we need to download the
<code>output-bigslice</code> to our local computer. We will use the
<code>scp</code> command to accomplish this.</p>
<p>Open another terminal and <strong>without connecting it to the remote
computer</strong> (<em>i.e.</em> in your own computer), move to a
directory where you know you can save the folder. In this example we
will copy it to the <code>Documents</code> folder of our computer.</p>
<div class="codewrapper sourceCode" id="cb42">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb42-1"><a href="#cb42-1" tabindex="-1"></a><span class="ex">$</span> cd Documents/</span>
<span id="cb42-2"><a href="#cb42-2" tabindex="-1"></a><span class="ex">$</span> scp <span class="at">-r</span> virtual-computerbetterlab@###.###.###.##/home/virtual-User/dc_workshop/results/bigslice/output-bigslice .</span>
<span id="cb42-3"><a href="#cb42-3" tabindex="-1"></a><span class="ex">$</span> ls <span class="at">-F</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>output-bigslice/</code></pre>
</div>
<p>The <code>BiG-SLiCE</code> prepared the code so that in each output
folder we will obtain the output files and some scripts that will
generate a way to visualize the data.</p>
<div class="codewrapper sourceCode" id="cb44">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb44-1"><a href="#cb44-1" tabindex="-1"></a><span class="ex">$</span> ls output-bigslice/</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>app  LICENSE.txt  requirements.txt  result  start_server.sh</code></pre>
</div>
<p>The file <code>requirements.txt</code> contains the names of two
softwares that need to be installed to use the visualization:
<code>SQLite3</code> and <code>flask</code>. We will install these with
the next command:</p>
<div class="codewrapper sourceCode" id="cb46">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb46-1"><a href="#cb46-1" tabindex="-1"></a><span class="ex">$</span> pip install <span class="at">-r</span> output-bigslice/requirements.txt</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>...
...
Successfully installed pysqlite3</code></pre>
</div>
<p>After the installation, we can use the <code>start_server.sh</code>
to generate a</p>
<div class="codewrapper sourceCode" id="cb48">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb48-1"><a href="#cb48-1" tabindex="-1"></a><span class="ex">$</span> bash output-bigslice/start_server.sh</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code> * Serving Flask app "/mnt/d/documentos/sideprojects/sc-mineria/bigSlice/output-bigslice/app/run.py"
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on http://0.0.0.0:5000/ (Press CTRL+C to quit)</code></pre>
</div>
<p>The result looks a little intimidating, but if you obtained a result
that looks like the above one do not worry. Next, we need to open an
internet browser. In a new tab type the next line.</p>
<pre class="url"><code>http://localhost:5000/</code></pre>
<p>As a result, we will obtain a web-page that looks like this:</p>
<p><a href="../fig/02-04-05.png">
<img src="../fig/02-04-05.png" alt="BiG SLiCE web page output displaying the results obtained from the example run.
                                  	A left panel presents the information generated, composed of seven tabs;
                                  	Summary, Datasets, Runs, Reports, Help, Feedback, and About.
                                  	The rest is a Summary of all datasets and runs included in the output.
                                  	Appearing firstly, the Datasets provided as input, organized as a table with five rows;
                                  	name, #genomes, total, with taxonomy, and description.
                                  	Next, the information about the Runs, also organized as a table with the following rows;
                                  	name, started, finished, status, threshold, #BGCs, and #GCFs.." class="figure"></a></p>
<p>Here we have a summary of the obtained results. In the left part, we
see a set of 7 fields. 3 of them have the information that we generated:
<code>Summary</code>, <code>Datasets</code>, and <code>Runs</code>. The
other ones are for information regarding the software, one of them is to
give feedback to the developers. You can acces to the
<code>Datasets</code> and <code>Runs</code> page also from the options
that we see in the main part of this summary page. The first part of
this summary describes the datasets that we provided as input. The
second one is the information of the process (when we ran the program)
that <code>BiG-SLiCE</code> carry out with our data.</p>
<p>If we click into the <code>Runs</code> field on the left or the
<code>run-0001</code> blue button on the <strong>Runs</strong> main
section, we will jump into the page that gives us the information
regarding the BGCs GCFs that were obtained:</p>
<p><a href="../fig/02-04-06.png">
<img src="../fig/02-04-06.png" alt="BiG SLiCE web page output displaying the information obtained from the Run-0001.
                                  	Firstly, it is shown an Overview from the Run. Indicating the Status, when was it Started and Finished,
                                  	as well as the Parameters, the HMM models and the Input.
                                  	Next, the Statistics about the run are shown as two bar-plots.
                                  	The left one plots the average distance to GCF models, whilst the right one shows the amount of BGCs in GCFs." class="figure"></a></p>
<p>In this new page we have two bar-plots that give us information
regarding if the BGCs that we sumministrated were classified as
fragmented or complete by <code>AntiSMASH</code> (according to the
<code>AntiSMASH</code> algorithm, a BGC can be classified as fragmented
if it is found at the edge of a contig), and how many families were
obtained with singletons (composed of just one BGC) and of 2-3 BGCs.</p>
<p>If we go to the bottom of this page, we will found a table of all the
GCFs with information of the number of BGCs that compose them, the
representative BGC class inside each family, and which genus has more
BGCs that belong to each family.</p>
<p>Click the <code>+</code> symbol of the GCF_7, and then on the
<code>View</code> button that is displayed. This will take us to a page
with statistics of the BGCs that are part of this family.</p>
<p>Inside this section, we can found a great amount of information
regarding the GCF_7 BGC’s. If we jump to section called
<code>Members</code>, we can click on the <code>view BGC arrowers</code>
button to get a visualization of the domains that are part of each of
the genes of these BGCs.</p>
<p><a href="../fig/02-04-07.png">
<img src="../fig/02-04-07.png" alt="BiG SLiCE web page output displaying detailed information regarding the BGCs from GCF_7.
                                  	The Arrowers show a gene arrow visualization of the domains that are part of each of the genes of the BGCs belonging to GCF_7." class="figure"></a></p>
<p>As we can see, this tool can be used to clusterize homologous BGCs
that share a structure. With this knowledge, we can try some hypothesis
or even build new ones in the light of this information.</p>
<div id="discussion" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<span class="callout-header">Discussion</span>
<div id="discussion" class="callout-inner">
<h3 class="callout-title">Discussion</h3>
<div class="callout-content">
<p>Considering the results in the runs tab, we can see that there are a
lot of Gene Cluster Families that have only one cluster associated to it
(GCF_1, GCF_2, GCF_4, GCF_5, GCF_6, GCF_8, GCF_9, GCF_13, and GCF_15).
<br> What does this tell us about the metabolites diversity of the
Streptococcus sample?</p>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<p>We can see that there are some unique gene cluster families, which
means that there are an organism that has that unique gene cluster and
therefore it has some metabolite that the others don´t and that may be
advantageous in its environment. Since there are many singleton Gene
Cluster Families, we can say that there is a big diversity from the
sample because many of the organisms in the sample have unique
“abilities” (metabolites).</p>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
<div id="discussion-1" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<span class="callout-header">Discussion</span>
<div id="discussion-1" class="callout-inner">
<h3 class="callout-title">Discussion</h3>
<div class="callout-content">
<p>Suppose we get the following results from a sample of 6
organisms.</p>
<p>|——————-+——————————————————–| | GCF | #core members |
|:—————–:+:——————————————————:| | GCF_1 | 6 |
|——————-+——————————————————–| | GCF_2 | 5 |
|——————-+——————————————————–| | GCF_3 | 5 |
|——————-+——————————————————–| | GCF_4 | 6 |
|——————-+——————————————————–|</p>
<p>What can we infer about the metabolite diversity?</p>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4"> Show me the solution </h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" data-bs-parent="#accordionSolution4" aria-labelledby="headingSolution4">
<div class="accordion-body">
<p>As most of the organisms have a gene cluster in the gene cluster
family, we can infer that most of them have the same secondary
metabolites, which means that there is little metabolite diversity
between them.</p>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="big-fam">BiG-FAM<a class="anchor" aria-label="anchor" href="#big-fam"></a>
</h2>
<hr class="half-width">
<p>To demonstrate the functionality of <code>BiG-SLICE</code>, the
authors gathered close to 1.2 million of BGCs. All this information,
they put it on a great database that they called <code>BiG-FAM</code>.
When you get to the main <a href="https://bigfam.bioinformatics.nl/home" class="external-link">page</a>, you will face a
structure that looks like the one we saw on the <code>BiG-SLICE</code>
results:</p>
<p><a href="../fig/02-04-08.png">
<img src="../fig/02-04-08.png" alt="BiG-FAM main page showing an introduction as well as a graphical representation of the database.
                                  	A left panel displays the available options; Home, GCFs, BGCs, Query, Statistics, Help, Feedback, and About. " class="figure"></a></p>
<p>One of the options on the left is <code>Query</code>. If we click
here, we will move to a new page where we can insert an
<code>antiSMASH job ID</code>. This will run a comparison of all the
BGCs found on that antiSMASH run on a single genome, against the immense
database.</p>
<p>We will use the <code>antiSMASH job ID</code> from the analysis made
on <em>Streptococcus agalactiae</em> A909. The
<code>antiSMASH job ID</code> can be found in the <code>URL</code>
direction of this page in the next format:
taxon-aaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee.</p>
<p>We will use the job ID that we obtained:</p>
<p>bacteria-f931deb5-509b-4f28-a210-212456c5139b</p>
<p>And click the <code>submit</code> option.</p>
<p><a href="../fig/02-04-09.png">
<img src="../fig/02-04-09.png" alt="BiG-FAM query page with the option for inserting an antiSMASH job and submitting it.
                                  	Below it is described how much of the job is Pending, Processing, and Processed." class="figure"></a></p>
<p>This will generate a result page that indicates with which BGCs from
the database, the BGCs from the query genome are related.</p>
<p><a href="../fig/02-04-10.png">
<img src="../fig/02-04-10.png" alt="BiG-FAM result page indicating firstly an overview of the job; the query which was submitted,
                                  	its status, as well as the time at which it was started and finished.
                                  	Next, a table indicating the BGCs from the database which are related with the query BGCs.
                                  	This is organized as a table with seven rows; query BGC, class, length (kb), completeness, best hit, distance and details" class="figure"></a></p>
<div id="keypoints2" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<ul>
<li>
<code>BiG-SLiCE</code> and <code>BiG-FAM</code> are softwares that
are useful to compare the metabolic diversity of bacterial lineages
between each other and against a big database</li>
<li>An input-folder containing the BGCs from antiSMASH and the taxonomic
information of each genome is needed to run <code>BiG-SLiCE</code>
</li>
<li>The results from the antiSMASH web-tool are needed to run
<code>BiG-FAM</code>
</li>
<li>Gene Cluster Families can help us to compare the metabolic
capabilities of a set of bacterial lineages</li>
<li>We can use <code>BiG-FAM</code> to compare a BGC against the whole
database and predict its Gene Cluster Family</li>
</ul>
</div>
</div>
</div>
</section></section><section id="aio-06-corason"><p>Content from <a href="06-corason.html">Finding Variation on Genomic Vicinities</a></p>
<hr>
<p>Last updated on 2026-02-20 |

        <a href="https://github.com/carpentries-incubator/genome-mining/edit/main/episodes/06-corason.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 35 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How can I follow variation in genomic vicinities given a reference
BGC?</li>
<li>Which gene families are the conserved part of a BGC family?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Learn to find BGC-families</li>
<li>Visualize variation in a genomic vicinity</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p><a href="../fig/corason.png">
<img src="../fig/corason.png" alt="CORASON's workflow for sorting phylogenetically BGCs. Given a query gene in a reference BGC and a genomic annotated database,
                                     CORASON firstly searches for query gene homologues, it filters out all genomic vicinities
                                     not related to the reference BGC. Then, CORASON infers a phylogenetic tree and
                                     calculates the frequency of occurrence for each gene family from the reference BGC.
                                     Using the same reference BGC, if a new query gene is selected, CORASON visualizes
                                     a new phylogeny with families containing the same molecular modifications." class="figure"></a></p>
<p>CORASON is a visual tool that identifies gene clusters that share a
common genomic core and reconstructs multi-locus phylogenies of these
gene clusters to explore their evolutionary relationships. CORASON was
developed to find and prioritize biosynthetic gene clusters (BGC), but
can be used for any kind of clusters.</p>
<p>Input: query gene, reference BGC and genome database. Output: SVG
figure with BGC in the family sorted according to the multi-locus
phylogeny of the core genes.</p>
<p>Advantages</p>
<ul>
<li>SVG graphs Scalable graphs that allow metadata easy display.</li>
<li>Interactive, CORASON is not a static database, it allows to explore
your own genomes.</li>
<li>Reproducibility, since CORASON runs on docker and conda, the
containerization allows to always perform the same analysis even if you
change your Linux/perl/blast/muscle/Gblocks/quicktree local
distributions.</li>
</ul>
<section><h2 class="section-heading" id="corason-conda">CORASON conda<a class="anchor" aria-label="anchor" href="#corason-conda"></a>
</h2>
<hr class="half-width">
<p>Here we are testing the new stand-alone <a href="https://github.com/miguel-mx/corason-conda" class="external-link">corason in a conda
environment</a> with gbk as input-files. Firstly, activate the
corason-conda environment.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">$</span> conda deactivate </span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="ex">$</span> conda activate /miniconda3/envs/corason  </span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>(corason) user@server:~$</code></pre>
</div>
<p>With the environment activated, all CORASON-dependencies are ready to
be used. The next step is to clone CORASON-software from its GitHub
repository. Firstly, place yourself at the results directory and then
clone CORASON-code.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="ex">$</span> mkdir <span class="at">-p</span> ~/pan_workshop/results/genome-mining </span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="ex">$</span> cd ~/pan_workshop/results/genome-mining</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="ex">$</span> git clone https://github.com/miguel-mx/corason-conda.git </span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="ex">$</span> ls</span></code></pre>
</div>
<p>The GitHub CORASON repository must be now in your directory.</p>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code><span><span class="va">corason</span><span class="op">-</span><span class="va">conda</span> </span></code></pre>
</div>
<p>Change directory by descending to EXAMPLE2. The file
<code>cpsg.query</code>, contains the reference protein cpsG, whose
encoding gene is part of the <a href="https://mibig.secondarymetabolites.org/repository/BGC0000744/index.html#r1c1" class="external-link">polysaccharide
BGC</a> produced by some <em>S. agalactiae</em></p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="ex">$</span> cd corason-conda/EXAMPLE2        </span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="ex">$</span> ls  </span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code><span><span class="va">cpsg.query</span>  </span></code></pre>
</div>
<p>As genomic database we will use the prokka-annotated <code>gbk</code>
files of <em>S. agalactiae</em>. This database will be stored in the
reserved directory <code>CORASON_GENOMES</code>.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="ex">$</span> mkdir  CORASON_GENOMES  </span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="ex">$</span> cp ~/pan_workshop/results/annotated/Streptococcus_agalactiae_<span class="pp">*</span>/<span class="pp">*</span>gbk CORASON_GENOMES/  </span></code></pre>
</div>
<p>CORASON was written to be used with RAST annotation as input files,
in this case we are using a genome database composed of
<code>.gbk</code> files. Thus, we need to convert gbk files into
CORASON-compatible input files.</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="ex">$</span> ../CORASON/gbkIndex.pl CORASON_GENOMES ../CORASON CORASON_GENOMES </span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Directory CORASON_GENOMES
../CORASON/gbk_to_fasta.pl CORASON_GENOMES agalactiae_18RS21_prokka.gbk 100001 ../CORASON
../CORASON/gbk_to_fasta.pl CORASON_GENOMES agalactiae_515_prokka.gbk 100002 ../CORASON
../CORASON/gbk_to_fasta.pl CORASON_GENOMES agalactiae_A909_prokka.gbk 100003 ../CORASON
../CORASON/gbk_to_fasta.pl CORASON_GENOMES agalactiae_CJB111_prokka.gbk 100004 ../CORASON
../CORASON/gbk_to_fasta.pl CORASON_GENOMES agalactiae_COH1_prokka.gbk 100005 ../CORASON
../CORASON/gbk_to_fasta.pl CORASON_GENOMES agalactiae_H36B_prokka.gbk 100006 ../CORASON
../CORASON/gbk_to_fasta.pl CORASON_GENOMES thermophilus_LMD-9_prokka.gbk 100007 ../CORASON
../CORASON/gbk_to_fasta.pl CORASON_GENOMES thermophilus_LMG_18311_prokka.gbk 100008 ../CORASON   </code></pre>
</div>
<p>Now, all converted genome files, aminoacid fasta files (.faa) and
annotation files (.txt) should be placed in the ‘GENOMES directory’ one
level up outside output.</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="ex">$</span> mv output/<span class="pp">*</span> . </span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a><span class="ex">$</span> ls</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>CORASON_GENOMES  Corason_Rast.IDs  cpsg.query  GENOMES  output </code></pre>
</div>
<p>Finally, we have the query enzyme, the IDs file and a genomic
database of <em>S. agalactiae</em> in the same directory. Run CORASON
with cpsg.query as query with 1000006 as an example of reference
BGC.</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="ex">$</span> ../CORASON/corason.pl <span class="at">-q</span> cpsg.query <span class="at">-s</span> 100006  <span class="at">-rast_ids</span> Corason_Rast.IDs</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>&gt;&gt; 100001_666.input  Cluster1237  Cluster1245  Cluster1253       cpsg.query_BGC.tre            cpsg.query_tree.svg
&gt;&gt; 100002_2002.input Cluster1238  Cluster1246  Cluster1254       cpsg.query.BLAST              Frequency
&gt;&gt; 100003_1173.input Cluster1239  Cluster1247  Cluster1255       cpsg.query.BLAST.pre          GBK
&gt;&gt; 100004_35.input   Cluster1240  Cluster1248  Cluster1256       cpsg.query.parser             Joined.svg
&gt;&gt; 100005_912.input  Cluster1241  Cluster1249  Cluster1257       cpsg.query.parser.pre         MINI
&gt;&gt; 100006_1247.input Cluster1242  Cluster1250  Concatenados.faa  cpsg.query_PrincipalHits.tre  PrincipalHits.muscle
&gt;&gt; 100007_1022.input Cluster1243  Cluster1251  CORASON           cpsg.query_Report             PrincipalHits.muscle-gb
&gt;&gt; 100008_1024.input Cluster1244  Cluster1252  cpsg.query        cpsg.query.svg                TempConcatenados.faa </code></pre>
</div>
<p>Finally, we have all the genomic vicinities sorted phylogenetically
according to the genes in the core-cluster. We can either see the output
directly on the server or download the resulting svg file to our local
computer. The format svg stands for scalable vector graphics, we are
save it as an html file in order to get the option of open it as html.
To see it in the server we must first copy the Joined.svg into
Joined.html, then, go to the EXAMPLE2 directory in the r-studio files
panel located at the bottom right. Once there clik in the Joined.html
file and chose the option View in Web Browser. The corason figure will
output in a new tab.</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="ex">$</span> cp output/cpsg.query-output/Joined.svg  Joined.html</span></code></pre>
</div>
<p><a href="../fig/ViewOutput.jpg">
<img src="../fig/ViewOutput.jpg" alt="CORASON phylogenetic svg reconstruction using cpsG as query gene and _S. agalactiae_ 1000006 as query cluster.
                                    At the bottom, it is displayed the frequency of occurrence for each gene family from the reference BGC,
                                    each with a different color." class="figure"></a></p>
<p>You can also use <code>scp</code> in a local terminal to download the
resulting svg file to your local computer.</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="ex">$</span> scp <span class="er">(</span><span class="ex">remoto</span><span class="kw">)</span><span class="ex">/corason-conda/EXAMPLE2/output/cpsg.query-output</span>  Downloads/.</span></code></pre>
</div>
<p><a href="../fig/Joined.svg">
<img src="../fig/Joined.svg" alt="CORASON phylogenetic svg reconstruction using cpsG as query gene and _S. agalactiae_ 1000006 as query cluster.
                                    At the bottom, it is displayed the frequency of occurrence for each gene family from the reference BGC,
                                    each with a different color." class="figure"></a></p>
<div id="corason-has-several-installation-and-a-conda-beta-versions" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<span class="callout-header">Callout</span>
<div id="corason-has-several-installation-and-a-conda-beta-versions" class="callout-inner">
<h3 class="callout-title">CORASON has several installation and a conda
beta versions</h3>
<div class="callout-content">
<p>CORASON has one <a href="https://github.com/nselem/corason" class="external-link">docker
installation</a> where you have to go inside the docker container, its
input are RAST files. The script <a href="https://bigscape-corason.secondarymetabolites.org/index.html" class="external-link">run_corason</a>
belongs to another installation that accepts gbk as input files. And
finally, this is a beta version of CORASON <a href="https://github.com/miguel-mx/corason-conda" class="external-link">conda
container</a>.</p>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<ul>
<li>CORASON is a command-line tool that finds BGC-families</li>
<li>Genomic vicinity variation is organized phylogenetically according
to the conserved genes in the BGC-family</li>
</ul>
</div>
</div>
</div>
</section></section><section id="aio-07-evomining"><p>Content from <a href="07-evomining.html">Evolutionary Genome Mining</a></p>
<hr>
<p>Last updated on 2026-02-20 |

        <a href="https://github.com/carpentries-incubator/genome-mining/edit/main/episodes/07-evomining.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 40 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What is Evolutionary Genome Mining?</li>
<li>Which kind of BGCs can EvoMining find?</li>
<li>What do I need in order to run an evolutionary genome mining
analysis?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand EvoMining pipeline.</li>
<li>Run an example of evolutionary analysis in cpsG gene family.</li>
<li>Explore MicroReact interactive output interface.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p><a href="../fig/evomining.gif">
<img src="../fig/evomining.gif" alt="a) EvoMining expansion-and-recruitment pipeline. A group of grey stacked cylinders representing genomes in a database (DB).
                                   	Homologues and expansions of seed enzymes, represented as an orange arrow, from the enzyme DB
                                   	are searched by blastp in the genome DB.
                                   	The outcome is integrated as the expanded enzyme families (EFs) within the genome DB.
                                   	Bidirectional best hits (BBH) of seed enzymes, red arrows, are marked as conserved metabolism.
                                   	The EFs are amplified after being compared against a DB of natural products (NP) biosynthetic enzymes,
                                   	represented by a blue cylinder, to find recruitments defined as enzymes of the family that are part of a MIBiG BGC.
                                   	b) The genome DB, represented by the gray stacked cylinders, is searched as previously described.
                                   	Additionally, antiSMASH predictions, cyan arrows, can be added by the user.
                                   	antiSMASH enzyme predictions that are at the same time marked in red are defined as transition enzymes, purple arrows.
                                   	c) EvoMining phylogenetic reconstruction and visualization. On the left side, a phylogenetic reconstruction of an EF is shown.
                                   	On the right side it is shown the EvoMining tree displaying the EvoMining predictions (green),
                                   	which are those extra copies closer to enzyme recruitments into BGC (blue) than to conserved metabolic enzymes (red).
                                   	antiSMASH predicted enzymes are represented in cyan, transition enzymes in black and
                                   	extra copies that are neither antiSMASH nor EvoMining predictions are left in grey." class="figure"></a></p>
<p>Usually, bioinformatics tools related to the prediction of Natural
Products (NP) biosynthetic genes try to find metabolic pathways of
enzymes that are known to be related with the synthesis of secondary
metabolites. However, these approaches fail for the discovery of novel
biosynthetic systems. Thus, <a href="https://www.microbiologyresearch.org/content/journal/mgen/10.1099/mgen.0.000260" class="external-link">EvoMining</a>
tries to circumvent this problem by detecting novel enzymes that may be
implicated in the synthesis of new natural products in Bacteria.</p>
<p>To know more about EvoMining you can read <a href="https://www.microbiologyresearch.org/content/journal/mgen/10.1099/mgen.0.000260" class="external-link">Selem
et al, Microbial Genomics 2019</a>.</p>
<p>EvoMining searches protein expansions that may have evolved from the
conserved metabolism into a specialized metabolism. It builds
phylogenetic trees based on all the protein copies of a certain enzyme
in a given genome database. The output tree differentiates copies that
are related with the conserved metabolism, copies that are known to be
implicated in discovered NP-producing-BGCs i.e. BGCs from <a href="https://mibig.secondarymetabolites.org/" class="external-link">MiBIG database</a> and,
optionally, protein copies that belong to BGCs predicted by <a href="https://antismash-db.secondarymetabolites.org/" class="external-link">antiSMASH</a>.
Finally, some branch in the tree will be depicted as “EvoMining hits”,
which represent enzyme expansions that are evolutionary closer to those
copies related with the secondary metabolism (MiBIG or antiSMASH BGCs)
than to those related with the conserved (primary) metabolism.</p>
<section><h2 class="section-heading" id="run-evomining-image">Run evomining image<a class="anchor" aria-label="anchor" href="#run-evomining-image"></a>
</h2>
<hr class="half-width">
<p>First, place yourself at your working directory.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">$</span> cd   ~/pan_workshop/results/genome-mining/corason-conda/EXAMPLE2  </span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="ex">$</span> ls</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>CORASON_GENOMES  Corason_Rast.IDs  cpsg.query  GENOMES  output</code></pre>
</div>
<p>The general structure of a docker container is shown in the next
bash-box. Note that it requires to specify which docker container will
run. Optionally, with <code>-v</code> flag it is possible to share a
directory with the container, with <code>-p</code> flag a port is shared
and it is also possible to specify which program will run inside the
container.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="ex">$</span> docker run <span class="at">--rm</span> <span class="at">-i</span> <span class="at">-t</span> <span class="at">-v</span> <span class="op">&lt;</span>your local directory<span class="op">&gt;</span>:<span class="op">&lt;</span>inside docker directory<span class="op">&gt;</span> -p <span class="op">&lt;</span>inside port<span class="op">&gt;</span>:80 <span class="op">&lt;</span>docker container<span class="op">&gt;</span> <span class="op">&lt;</span>program inside docker<span class="op">&gt;</span>    </span></code></pre>
</div>
<p>EvoMining is inside a docker container, so the general structure to
start your analysis will be as follows:</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="ex">$</span> docker run <span class="at">--rm</span> <span class="at">-i</span> <span class="at">-t</span> <span class="at">-v</span> <span class="va">$(</span><span class="bu">pwd</span><span class="va">)</span>:/var/www/html/EvoMining/exchange <span class="at">-p</span> 8080:80 nselem/evomining:latest /bin/bash   </span></code></pre>
</div>
<p>Let’s explain the pieces of this line.</p>
<table class="table">
<colgroup>
<col width="7%">
<col width="92%">
</colgroup>
<thead><tr class="header">
<th align="center">command</th>
<th>Explanation</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="center">docker</td>
<td>tells the system that we are running a docker command</td>
</tr>
<tr class="even">
<td align="center">run</td>
<td>the command that we are running is to run a docker container</td>
</tr>
<tr class="odd">
<td align="center">--rm</td>
<td>this container will be removed after closed</td>
</tr>
<tr class="even">
<td align="center">-i</td>
<td>this container allows user interaction</td>
</tr>
<tr class="odd">
<td align="center">-t</td>
<td>this interaction will be through a terminal</td>
</tr>
<tr class="even">
<td align="center">-v</td>
<td>a data volume (directory) will be shared between your local machine
and the container</td>
</tr>
<tr class="odd">
<td align="center">-p</td>
<td>a port will allow a web based app</td>
</tr>
</tbody>
</table>
<p>However, sometimes the port 80 is busy, in that case you can use
other ports like 8080 or 8084. If this is the case, please use the port
<code>80X</code> where X is a number between 01..30 provided by your
instructor.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="ex">$</span> docker run <span class="at">--rm</span> <span class="at">-i</span> <span class="at">-t</span> <span class="at">-v</span> <span class="va">$(</span><span class="bu">pwd</span><span class="va">)</span>:/var/www/html/EvoMining/exchange <span class="at">-p</span> 8080:80 nselem/evomining:latest /bin/bash  </span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="ex">$</span> docker run <span class="at">--rm</span> <span class="at">-i</span> <span class="at">-t</span> <span class="at">-v</span> <span class="va">$(</span><span class="bu">pwd</span><span class="va">)</span>:/var/www/html/EvoMining/exchange <span class="at">-p</span> 8084:80 nselem/evomining:latest /bin/bash  </span></code></pre>
</div>
<p>If your docker container worked, now you will see in your terminal a
new prompt. Instead of the usual dollar sign, there should be a number
<code>#</code> at the beginning of your terminal. This is because now
you are inside the docker container and you have <code>sudo</code>
permissions inside the docker.</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="co">#  </span></span></code></pre>
</div>
<p>To exit container use <code>exit</code></p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="co"># exit</span></span></code></pre>
</div>
<p>And now your prompt must be back in the dollar sign</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="ex">$</span>   </span></code></pre>
</div>
</section><section><h2 class="section-heading" id="set-evomining-genomic-database">Set EvoMining genomic database<a class="anchor" aria-label="anchor" href="#set-evomining-genomic-database"></a>
</h2>
<hr class="half-width">
<p>Start the container again with your corresponding port.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="ex">$</span> docker run <span class="at">--rm</span> <span class="at">-i</span> <span class="at">-t</span> <span class="at">-v</span> <span class="va">$(</span><span class="bu">pwd</span><span class="va">)</span>:/var/www/html/EvoMining/exchange <span class="at">-p</span> 80X:80 nselem/evomining:latest /bin/bash   </span></code></pre>
</div>
<p>Though we will NOT run the test EvoMining command, it looks as
follows:</p>
<pre class="code"><code><span><span class="co"># perl startEvoMining.pl  </span></span></code></pre>
<p>Instead of that, customize the genomic database by using the same as
CORASON.<br>
Notice that EvoMining requires RAST-like annotated genomes and for this
reason we are using the fasta files that CORASON converts from our gbk
inputs.</p>
<pre class="code"><code><span><span class="co"># perl startEvoMining.pl -g GENOMES -r  Corason_Rast.IDs</span></span></code></pre>
<p><a href="../fig/tree.png"></a></p>
<figure><img src="../../fig/tree.png" alt="EvoMining phylogenetic reconstruction providing evolutionary insights into the metabolic origin
and the fate of members of diverse EF from the Streptococcus example.
Seed enzymes are labeled in orange. The most conserved copies or central metabolism copies are marked in red.
Enzyme copies recruited into specialized metabolism, contained in MIBiG, are labeled in blue.
Enzyme copies that are closer to blue enzyme recruitments than to red conserved enzymes are labeled in green
and represent EvoMining Hits. Extra copies with an unknown metabolic fate are shown in gray." class="figure mx-auto d-block"></figure><p></p>
<p>Finally, remember that <code>X</code> means your user-number and open
your browser at the address:
<code>http://132.248.196.38:80X/EvoMining/html/index.html</code>. Once
there, just click the start button and enjoy! (click on the submit
buttons!)</p>
<p>When you finish using this container, please exit it.</p>
<pre class="code"><code><span><span class="co">#exit</span></span></code></pre>
</section><section><h2 class="section-heading" id="visualize-your-results-in-microreact">Visualize your results in MicroReact<a class="anchor" aria-label="anchor" href="#visualize-your-results-in-microreact"></a>
</h2>
<hr class="half-width">
<p>Firstly, you have to run all the pipeline in the website: <a href="http://" class="external-link uri">http://</a><yourip>/EvoMining/html/index.html, and then all
the output files will be generated. You can use the EvoMining basic
interface or take your results into MicroReact.</yourip></p>
<p>EvoMining outputs are stored in the directory
<code>&lt;conserved-db&gt;_&lt;natural-db&gt;_&lt;genomes-db&gt;</code></p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="ex">$</span> ls</span></code></pre>
</div>
<p>To explore EvoMining outputs, you need to upload 1.nwk and 1.csv
files to <a href="https://microreact.org/" class="external-link">microReact</a>. There are
many methods to download files from the server to your local
computer.</p>
<p>If you are using JupyterHub you explore the file folders and select
the files and then press the download button.</p>
<p>You can use the export button in the file panel of R studio. To
download the files, first in the files panel open in the directory
<code>~/pan_workshop/results/genome-mining/corason-conda/EXAMPLE2/ALL_curado.fasta_MiBIG_DB.faa_GENOMES/blast/seqf/tree</code>,<br><a href="../../fig/ExportFiles1.jpg">
<img src="../../fig/ExportFiles1.jpg" alt="Select the path to download file" class="figure"></a></p>
<p>Then, select files 1.nwk and 1.csv in that directory, click
<code>more</code> in the engine icon, and select the <code>export</code>
option in the menu. The files will be downloaded to your local computer,
and now you will be able to upload them to MicroReact.</p>
<p><a href="../fig/ExportFiles2.png">
<img src="../fig/ExportFiles2.png" alt="Select the path to download file" class="figure"></a></p>
<p>Alternatively, If your prefer to use the terminal to download files
the <code>scp</code> protocol can download the files into your local
machine.</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="fu">scp</span> betterlab@132.248.196.38:~/pan_workshop/results/genome-mining/corason-conda/EXAMPLE2/ALL_curado.fasta_MiBIG_DB.faa_GENOMES/blast/seqf/tree/1.nwk ~/Downloads/.</span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a><span class="fu">scp</span> betterlab@132.248.196.38:~/pan_workshop/results/genome-mining/corason-conda/EXAMPLE2/ALL_curado.fasta_MiBIG_DB.faa_GENOMES/blast/seqf/tree/1.csv ~/Downloads  </span></code></pre>
</div>
<p>Here you can find the <a href="https://microreact.org/project/e8b7wWZkovtavPFFpBXRPp-evomining-streptococcus-example" class="external-link">MicroReact
visualization</a> of this EvoMining run.</p>
<p><a href="../fig/EvoMiningMicroReact.png">
<img src="../fig/EvoMiningMicroReact.png" alt="MicroReact visualization of the EvoMining run Streptococcus example.
                                             	At the left a bar-chart with the EF in the X axis and the number of entries in the Y axis.
                                             	At the right, the EvoMining phylogenetic tree using the same color code as the chart.
                                             	Right of the tree the legend indicating the colors by metabolism; central metabolism enzymes in red,
                                             	expansion enzymes in gray, recruited enzymes contained in MIBiG in blue,
                                             	secondary metabolism enzymes (EvoMining hits) are marked in green,
                                             	and seed enzymes are colored in orange. Below appears the metadata from the run,
                                             	organized in a five row table including Id, metabolism, genome, function and copies." class="figure"></a></p>
</section><section><h2 class="section-heading" id="other-resources">Other resources<a class="anchor" aria-label="anchor" href="#other-resources"></a>
</h2>
<hr class="half-width">
<p>To run EvoMining with a larger conserved-metabolite DB you can use <a href="https://zenodo.org/record/1219709#.YqEsFqjMLrc" class="external-link">EvoMining
Zenodo</a> data.</p>
<p>To explore more EvoMining options, please explore <a href="https://github.com/nselem/evomining/wiki" class="external-link">EvoMining wiki</a>.</p>
</section><section><h2 class="section-heading" id="set-the-conserved-enzymes-database">Set the conserved-enzymes database<a class="anchor" aria-label="anchor" href="#set-the-conserved-enzymes-database"></a>
</h2>
<hr class="half-width">
<p>When using EvoMining, oftenly you will desire to construct your own
conserved enzymes database. To know more about how to configure a
database, consult the EvoMining wiki in the <a href="https://github.com/nselem/evomining/wiki/Databases-Conformation" class="external-link">EvoMining
databases</a> part. Natural products database could also be replaced for
another set of genes that are “true positives”, for example a set of
regulatory genes.</p>
<p>As an example, transform the file <code>cpsg.query</code> into the
format of this database. This file contains the aminoacid sequence of
the <em>cpsG</em> gene. Firstly, copy this file into what will become
the conserved-enzymes database.</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="ex">$</span> cp cpsg.query cpsg_cdb</span></code></pre>
</div>
<p>Now, it requires some editing. Open nano editor and change the first
line <code>&gt;cpsg</code> to
<code>&gt;SYSTEM1|1|phosphomannomutase|Saga</code>. EvoMining
conserved-database needs a four-field format pipe-separated that
contains; the name of the metabolic system to which the enzyme belongs
(SYSTEM1), a consecutive number of the enzyme (1 in this case), the
function of the enzyme, and finally, an abbreviation of the organism
Saga, (<em>S. Agalactiae</em>).<br>
The reason behind this is that this was the way we needed EvoMining for
its first use and we have not changed the headers since.</p>
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="ex">$</span> nano cpsg_cdb</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>&gt;SYSTEM1|1|phosphomannomutase|Saga
MIFVTVGTHEQQFNRLIKEVDRLKGTGAIDQEVFIQTGYSDFEPQNCQWSKFLSYDDMNSYMKEAEIVITHGGPATFMSVISLGKLPVVVPRRKQFGEHINDHQIQFLKKIAHLYPLAWIED</code></pre>
</div>
<p>Run your EvoMining docker</p>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="ex">$</span> docker run <span class="at">--rm</span> <span class="at">-i</span> <span class="at">-t</span> <span class="at">-v</span> <span class="va">$(</span><span class="bu">pwd</span><span class="va">)</span>:/var/www/html/EvoMining/exchange <span class="at">-p</span> 80X:80 nselem/evomining:latest /bin/bash   </span></code></pre>
</div>
<p>and inside this new container:</p>
<pre class="code"><code><span><span class="co"># perl startEvoMining.pl -g GENOMES -r  Corason_Rast.IDs -c cpsg_cdb</span></span></code></pre>
<p>Use the website again and think about the results.</p>
<div id="exercise-1.-set-evomining-parameters" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="exercise-1.-set-evomining-parameters" class="callout-inner">
<h3 class="callout-title">Exercise 1. Set EvoMining parameters</h3>
<div class="callout-content">
<p>Complete the blanks in the following EvoMining run:
<code>actinoSMASH</code> A file with the ids of antiSMASH recognized
genes. <code>Actinos</code> a directory with RAST-like fasta and
annotation files. <code>Histidine-db</code> A fasta file with some
proteins in the histidine pathway.<br><code>Actinos.ids</code> tabular files with the RAST ids and the name of
the organisms.</p>
<pre class="code"><code><span> <span class="co"># perl starEvoMining.pl -g ____ -c _____ -r _____ -a ___________  </span></span></code></pre>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a><span class="co"># perl starEvoMining.pl -g Actinos -c Histidine-db -r Actinos.ids -a actinoSMASH  </span></span></code></pre>
</div>
<p>Actinos is the genomic database, Histidine-db is the
conserved-enzymes database, Actinos.ids is the file that relates Rast
ids with the organisms names, and actinoSMASH contains the genes
identified by antiSMASH.</p>
</div>
</div>
</div>
</div>
<div id="discussion-1.-retro-evomining-in-enzyme-database" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<span class="callout-header">Discussion</span>
<div id="discussion-1.-retro-evomining-in-enzyme-database" class="callout-inner">
<h3 class="callout-title">Discussion 1. Retro EvoMining in enzyme
database</h3>
<div class="callout-content">
<p>What do you learn from running in a conserved-enzymes database the
gene <em>cpsG</em> that is part of a specialized BGC?</p>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p><em>cpsG</em> does not have extra copies in Streptococcus agalactiae,
so there are no expansions that may be functional divergent.
<em>cpsG</em> single copies in the genomes look red-colored in EvoMining
output, as if they belong to the conserved-metabolism. However, this is
not the case, the color is because there is only one copy and it is
merged into MIBiG true-positives because it was originally a gene in the
specialized metabolism. So it is important to know the seed enzymes.</p>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code><span><span class="va">cpsg_cdb_MiBIG_DB.faa_GENOMES</span></span></code></pre>
</div>
<p><a href="https://arts.ziemertlab.com/" class="external-link">ARTS</a> is another
evolutionary genome mining software with its corresponding database <a href="https://arts-db.ziemertlab.com/" class="external-link">ARTS-db</a> .</p>
<div id="evolutionary-genome-mining" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<span class="callout-header">Callout</span>
<div id="evolutionary-genome-mining" class="callout-inner">
<h3 class="callout-title">Evolutionary Genome Mining</h3>
<div class="callout-content">
<p>To learn more about Genome Mining you can read these references:</p>
<ul>
<li>The confluence of big data and evolutionary genome mining for the
discovery of natural products. <a href="https://doi.org/10.1039/D1NP00013F" class="external-link">Chevrette et al,
2021</a>`</li>
<li>Evolutionary Genome Mining for the Discovery and Engineering of
Natural Product Biosynthesis. <a href="https://doi.org/10.1007/978-1-0716-2273-5_8" class="external-link">Chevrette et al,
2022</a>
</li>
</ul>
</div>
</div>
</div>
<blockquote>
<h2 id="more-about-docker">More about docker</h2>
<p>To see the running container use <code>ps</code></p>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a><span class="ex">$</span> docker ps</span></code></pre>
</div>
</blockquote>
<blockquote>
<p>If there are containers in use you will see a list of all of
them.</p>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>2f879ba6e337   nselem/evomining:latest   "/bin/bash"   11 hours ago   Up 11 hours   0.0.0.0:8014-&gt;80/tcp, :::8014-&gt;80/tcp   relaxed_dirac</code></pre>
</div>
</blockquote>
<div id="callout2" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<span class="callout-header">Callout</span>
<div class="callout-inner">
<div class="callout-content">
<p>To stop the running container use <code>docker stop</code> and to
remove them use <code>docker rm</code></p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a><span class="ex">$</span> docker stop relaxed_dirac  </span>
<span id="cb25-2"><a href="#cb25-2" tabindex="-1"></a><span class="ex">$</span> docker remove relaxed_dirac  </span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>2f879ba6e337</code></pre>
</div>
<p>can be downloaded by specifying our queries with
<code>--format genbank</code>.</p>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<ul>
<li>EvoMining is a command-line tool that performs evolutionary genome
mining over gene families</li>
<li>EvoMining hits can belong to new BGC</li>
<li>MicroReact is an interactive genomic visualizer compatible with
EvoMining output</li>
</ul>
</div>
</div>
</div>
</section></section><section id="aio-08-GatorGC"><p>Content from <a href="08-GatorGC.html">GATOR-GC: Genomic Assessment Tool for Orthologous Regions and Gene Clusters</a></p>
<hr>
<p>Last updated on 2026-02-20 |

        <a href="https://github.com/carpentries-incubator/genome-mining/edit/main/episodes/08-GatorGC.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 60 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What is GATOR-GC, and how does it differ from other BGC exploration
tools?</li>
<li>How does GATOR-GC establish BGC boundaries using evolutionary
principles?</li>
<li>What types of biosynthetic diversity can GATOR-GC identify?</li>
<li>What do I need to perform a targeted exploration using
GATOR-GC?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand the GATOR-GC algorithm and its evolutionary approach to
BGC boundary definition.</li>
<li>Learn how to conduct a targeted search for key enzymes within BGCs
and genomic islands using GATOR-GC.</li>
<li>Execute an example analysis using customizable parameters for
precision in identifying GATOR windows.</li>
<li>Visualize and interpret the output of GATOR-GC, including gene
cluster diagrams and genomic neighborhoods.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p><a href="../fig/gator_gc_logo.jpg">
<img src="../fig/gator_gc_logo.jpg" alt="GATOR methods" style="width: 60%; height: auto;" class="figure"></a></p>
<p>GATOR-GC is a user-friendly algorithm designed for targeted
exploration of BGC and genomic islands diversity. It focuses on key
biosynthetic enzymes and offers flexibility in defining the taxonomic
scope of the analysis. Unlike methods relying on arbitrary cutoffs,
GATOR-GC establishes BGC boundaries based on evolutionary principles and
implements an enzyme-aware scoring system for assessing BGC-BGC
distances, moving beyond a binary presence-absence framework. This
approach enhances the tool’s capability to identify and prioritize
novelty, effectively mapping biosynthetic diversity into distinct
groups.</p>
<p>To know more about GATOR GC you can go to the <a href="https://github.com/chevrettelab/gator-gc/" class="external-link">Github
repository</a>.</p>
<section><h2 class="section-heading" id="most-relevant-features">Most relevant features<a class="anchor" aria-label="anchor" href="#most-relevant-features"></a>
</h2>
<hr class="half-width">
<ul>
<li>
<strong>Targeted Search</strong>: Conduct targeted searches for
essential key enzymes and optional tailor enzymes within Biosynthetic
Gene Clusters (BGCs) and genomic islands, streamlining the discovery
process.</li>
<li>
<strong>Modular Domain Screening</strong>: Automatically screen
user-provided protein files to identify critical modular domains, such
as Non-Ribosomal Peptide Synthetases (NRPSs) and Polyketide Synthases
(PKSs), using state-of-the-art HMM profiles from antiSMASH for
unparalleled precision.</li>
<li>
<strong>Customizable Parameters</strong>: Customize search
parameters to include required and optional proteins, ensure complete
assembly with all necessary proteins, and define specific distances
between required proteins. This customization enhances the specificity
of GATOR window identification, offering tailored analysis to meet
research needs.</li>
<li>
<strong>Dereplication for Gator Windows</strong>: Remove identical
Gator Windows to avoid redundant calculations in subsequent steps, while
keeping track of which windows are duplicates</li>
<li>
<strong>GATOR Focal Scores</strong>: Employ a novel enzyme-aware
scoring system to accurately compare GATOR windows against targeted
focal windows. This approach ensures precise evaluation of genomic
contexts and enzyme functionalities.</li>
<li>
<strong>GATOR Conservation</strong>: Generate dynamic gene cluster
diagrams that visually differentiate between required and optional
proteins using color coding. Transparency levels indicate the gene’s
presence within GATOR windows, providing a clear visual representation
of gene conservation</li>
<li>
<strong>GATOR Neighborhoods</strong>: Visualize each GATOR window’s
genomic neighborhood with organized tracks based on GATOR focal scores.
Homology between genes is intuitively illustrated with gray bars,
facilitating easy understanding of genetic relationships and
conservation</li>
</ul>
<p><a href="../fig/gator_gc_alogorithm.png">
<img src="../fig/gator_gc_alogorithm.png" alt="GATOR methods" class="figure"></a></p>
</section><section><h2 class="section-heading" id="gator-gc">GATOR GC<a class="anchor" aria-label="anchor" href="#gator-gc"></a>
</h2>
<hr class="half-width">
<p>Before starting, make sure to activate the <code>gator-gc</code>
environment.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">$</span> conda activate /miniconda3/envs/gator-gc/</span></code></pre>
</div>
<p>The first part of running GatorGC has to do with creating a database.
To facilitate the process first we need to create a directory where we
can find all Gator-GC results:</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="ex">$</span> cd   ~/pan_workshop/results/</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="ex">$</span> mkdir gator_gc_res/</span></code></pre>
</div>
<p>After creating this, we will move to this folder</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="ex">$</span> cd   ~/pan_workshop/results/gator_gc_res</span></code></pre>
</div>
<p>To create the database, we need all the annotated genomes that we
have previously processed. For the database, we only require the genbank
files, which end with gbk. We can use the <code>cp</code> command to
copy all the genbank files of our annotated genomes to a folder in this
directory. First, we will create the directory where we will copy the
files.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="ex">$</span> mkdir   gbks/</span></code></pre>
</div>
<div id="exercise-1.-copy-all-the-gbk-files-to-the-new-directory" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="exercise-1.-copy-all-the-gbk-files-to-the-new-directory" class="callout-inner">
<h3 class="callout-title">Exercise 1. Copy all the gbk files to the new
directory</h3>
<div class="callout-content">
<p>We know that all the genbank files are in the following path
<code>~/pan_workshop/results/annotated/</code> and inside the annotated
folder, there is a folder for each genome and each folder contains the
output files from prokka. You can use
<code>tree ~/pan_workshop/results/annotated/</code>to confirm this.
Using <code>cp</code>, try to complete the gaps in the following command
to copy all the genbank files from the previously processed genomes to
the new directory you created <code>gbks</code>. Remember you can use
<code>*</code> to match any symbol(s).</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="ex">$</span> cp  ~/pan_workshop/results/annotated/_________prokka/_________  gbks/. </span></code></pre>
</div>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">cp</span> ~/pan_workshop/results/annotated/<span class="pp">*</span>prokka/<span class="pp">*</span>gbk gbks/.</span></code></pre>
</div>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="set-gator-gc-database">Set GATOR-GC database<a class="anchor" aria-label="anchor" href="#set-gator-gc-database"></a>
</h2>
<hr class="half-width">
<p>Now that we have the genbank files in a folder we can create the
necessary database for gator-gc. The command to create the database is
<code>pre-gator-gc</code>. You can use the <code>-h</code> flag to
explore the options of this command.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="ex">$</span> pre-gator-gc <span class="at">-h</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>usage: pre-gator-gc [-h] -g  [...] [-e] [-t] -o  [-v]


     -\ ---\--\ -------\ ----\ ---\--\ ---\ --\ ----\ ----\--------\ /---
   /--/ ---/--/ -------/ ----/ ---/--/ ---/ --/ ----/ ----/--------/ \----\
  ________   _____  __________________  __________         _________________
 /  _____/  /  _  \ \__    ___/_____  \ \______   \       /  _____/\_   ___ \
/   \  ___ /  /_\  \  |    |   /   |   \ |       _/ _____    \  ___/    \  \/
\    \_\  \    |    \ |    |  /    |    \|    |   \/_____/    \_\  \     \_____
 \______  /____|__  / |____|  \_______  /|____|_  /       \______  /\_______  /
        \/        \/                  \/        \/               \/         \/
    -----\--\ -------\ ----\ ---\--\ ---\ --\ ------\ ----\---------\ /----
    \----/--/ -------/ ----/ ---/--/ ---/ --/ ------/ ----/---------/ \----/

GATOR-GC: Genomic Assessment Tool for Orthologous Regions and Gene Clusters
Developer: José D. D. Cediel-Becerra
Afiliation: Microbiology &amp; Cell Science Deparment, University of Florida
Please contact José at jcedielbecerra@ufl.edu if you have any issues
Version: v0.9.0

optional arguments:
  -h, --help  show this help message and exit
  -v          Enable verbose output (default: False).

Input Options:
  -g  [ ...]  Directories containing GenBank files (*.gbff/*.gbk/*.gb). You can specify multiple directories separated by spaces. Directories can be specified with or without wildcards.

HMMER Options:
  -e          E-value threshold for HMMER hmmsearch (default: 1e-4).
  -t          Number of CPU threads to use for hmmsearch (default: all available threads).

Output Options:
  -o          Directory where the gator databases (protein,  DIAMOND, and modular domtblout databases) will be saved.</code></pre>
</div>
<p>As we can see in the help output, we have to feed the command with -g
which is the path to the folder containing the genbank files and -o
which is the directory where we want gator to save the database to be
processed (output folder). To create the database we will use the
following line of code:</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="ex">$</span> pre-gator-gc <span class="at">-g</span> gbks/ <span class="at">-o</span> gator_databases <span class="at">-v</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>

     -\ ---\--\ -------\ ----\ ---\--\ ---\ --\ ----\ ----\--------\ /---
   /--/ ---/--/ -------/ ----/ ---/--/ ---/ --/ ----/ ----/--------/ \----\
  ________   _____  __________________  __________         _________________
 /  _____/  /  _  \ \__    ___/_____  \ \______   \       /  _____/\_   ___ \
/   \  ___ /  /_\  \  |    |   /   |   \ |       _/ _____    \  ___/    \  \/
\    \_\  \    |    \ |    |  /    |    \|    |   \/_____/    \_\  \     \_____
 \______  /____|__  / |____|  \_______  /|____|_  /       \______  /\_______  /
        \/        \/                  \/        \/               \/         \/
    -----\--\ -------\ ----\ ---\--\ ---\ --\ ------\ ----\---------\ /----
    \----/--/ -------/ ----/ ---/--/ ---/ --/ ------/ ----/---------/ \----/

GATOR-GC: Genomic Assessment Tool for Orthologous Regions and Gene Clusters
Developer: José D. D. Cediel-Becerra
Afiliation: Microbiology &amp; Cell Science Deparment, University of Florida
Please contact José at jcedielbecerra@ufl.edu if you have any issues
Version: v0.9.0
[1] - 2024-08-20 14:06:32,797 - INFO - The gator_databases directory was created successfully.
[2] - 2024-08-20 14:06:32,797 - INFO - Total genome files found: 9
[3] - 2024-08-20 14:06:34,223 - INFO - Successfully created the gator protein database to gator_databases/gator_databases.faa
[4] - 2024-08-20 14:06:34,295 - INFO - Successfully created the gator DIAMOND database to gator_databases/gator_databases.dmnd
[5] - 2024-08-20 14:06:34,946 - INFO - Successfully created the gator domtblout database to gator_databases/gator_databases.domtblout
[6] - 2024-08-20 14:06:34,946 - INFO - Execution time: 2.15 seconds</code></pre>
</div>
<p><strong>QUESTION</strong>: You already read the help page of
<code>pre-gator-gc</code>. Why we added the flag <code>-v</code>?</p>
</section><section><h2 class="section-heading" id="gator-gc-1">GATOR-GC<a class="anchor" aria-label="anchor" href="#gator-gc-1"></a>
</h2>
<hr class="half-width">
<p>Now that we have the database ready, let’s review what we need to run
<code>gator-gc</code>. To find out, we can type the command with the
<code>-h</code> help flag.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="ex">$</span>  gator-gc <span class="at">-h</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>usage: gator-gc [-h] -rq  [-op] -g  [...] -d  [-t] [-qc] [-idt] [-e] [-rd] [-we] -o  [-nc] [-nn] [-v]

     -\ ---\--\ -------\ ----\ ---\--\ ---\ --\ ----\ ----\--------\ /---
   /--/ ---/--/ -------/ ----/ ---/--/ ---/ --/ ----/ ----/--------/ \----\
  ________   _____  __________________  __________         _________________
 /  _____/  /  _  \ \__    ___/_____  \ \______   \       /  _____/\_   ___ \
/   \  ___ /  /_\  \  |    |   /   |   \ |       _/ _____    \  ___/    \  \/
\    \_\  \    |    \ |    |  /    |    \|    |   \/_____/    \_\  \     \_____
 \______  /____|__  / |____|  \_______  /|____|_  /       \______  /\_______  /
        \/        \/                  \/        \/               \/         \/
    -----\--\ -------\ ----\ ---\--\ ---\ --\ ------\ ----\---------\ /----
    \----/--/ -------/ ----/ ---/--/ ---/ --/ ------/ ----/---------/ \----/

GATOR-GC: Genomic Assessment Tool for Orthologous Regions and Gene Clusters
Developer: José D. D. Cediel-Becerra
Afiliation: Microbiology &amp; Cell Science Deparment, University of Florida
Please contact José at jcedielbecerra@ufl.edu if you have any issues
Version:v0.9.0

optional arguments:
  -h, --help  show this help message and exit
  -v          Enable verbose output. (Default: False)

Input Options:
  -rq         Path to the query protein FASTA file containing required proteins.
  -op         Path to the query protein FASTA file containing optional proteins.
  -g  [ ...]  Directory containing the Genbank files (*.gbff/*.gbk/*.gb).You can specify multiple directories separated by spaces. Directories can be specified with or without wildcards.
  -d          Directory containing the PRE-GATOR-GC databases (.dmnd and .domtblout files). (Required)

Diamond Options:
  -t          Number of CPUs to use for diamond search and hmmsearch. (Default: all available CPUs)
  -qc         Minimum percent query cover for diamond search. (Default: 70)
  -idt        Minimum percent identity for diamond search. (Default: 35)

HMMER Options:
  -e          E-value threshold for hmmsearch. (Default: 1e-4)

GATOR-GC Options:
  -rd         Maximum distance in kilobases between required genes to define a gator window. (Default: 86 kb)
  -we         Extension in kilobases from the start and end positions of the gator windows. (Default: 10 kb)

Output Options:
  -o          Directory to save GATOR-GC results.
  -nc         Disable creation of GATOR conservation figures.
  -nn         Disable creation of GATOR neighborhoods figures.</code></pre>
</div>
<p>From this output, we can see the Gator-GC options. The minimum
requirements to run gator-gc are:</p>
<ul>
<li>
<code>-rq</code> Path to the query protein FASTA file containing
required proteins.</li>
<li>
<code>-g</code> Directory containing the Genbank files.</li>
<li>
<code>-d</code> Directory containing the PRE-GATOR-GC
databases.</li>
<li>
<code>-o</code> Output directory</li>
</ul>
<p>To compare the results obtained by the different tools we have used,
we will use the reference protein cpsG, whose encoding gene is part of
the polysaccharide BGC produced by some <em>S. agalactiae</em>. The
aminoacid fasta is already in the server in
<code>~/pan_workshop/results/genome-mining/corason-conda/EXAMPLE2/cpsg.query</code>.
So the command to run gator-gc is the following:</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="ex">$</span>  gator-gc <span class="at">-rq</span> ~/pan_workshop/results/genome-mining/corason-conda/EXAMPLE2/cpsg.query <span class="at">-g</span> gbks/ <span class="at">-d</span> gator_databases/ <span class="at">-o</span> cpsg_gator</span></code></pre>
</div>
<p>Once Gator-GC finishes, we can observe the structure of the files
created by the program using <code>tree</code>.</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="ex">$</span>  tree cpsg_gator/</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>cpsg_gator/
├── all_merged_queries_38nh2fip.faa
├── all_merged_queries_unvwb3pq.domtbl
├── allvall_proteins_nuxdlir7.faa
├── concatenated_scores
│   ├── clustermap_gfs.svg
│   └── concatenated_gfs.csv
├── deduplicate_allvall_proteins_e3svhd1_.faa
├── deduplicate_dmnd_db__lno1x3_.dmnd
├── deduplicate_dmnd_out_3ve4lo6v.dmnd
├── deduplication_data
│   ├── deduplication_gfs.csv
│   └── unq_comp.tsv
├── dmnd_db_r0qpquhc.dmnd
├── dmnd_out_aun2m_k5.txt
├── gator_conservation_plots
│   ├── window_1--Streptococcus_agalactiae_2603V_prokka.svg
│   ├── window_2--agalactiae_515_prokka.svg
│   ├── window_4--Streptococcus_agalactiae_COH1_prokka.svg
│   ├── window_5--Streptococcus_agalactiae_CJB111_prokka.svg
│   ├── window_8--Streptococcus_agalactiae_NEM316_prokka.svg
│   └── window_9--Streptococcus_agalactiae_18RS21_prokka.svg
├── gator_neighborhoods_plots
│   ├── window_1--Streptococcus_agalactiae_2603V_prokka_neighborhoods.svg
│   ├── window_2--agalactiae_515_prokka_neighborhoods.svg
│   ├── window_4--Streptococcus_agalactiae_COH1_prokka_neighborhoods.svg
│   ├── window_5--Streptococcus_agalactiae_CJB111_prokka_neighborhoods.svg
│   ├── window_8--Streptococcus_agalactiae_NEM316_prokka_neighborhoods.svg
│   └── window_9--Streptococcus_agalactiae_18RS21_prokka_neighborhoods.svg
├── gator_scores
│   ├── window_1--Streptococcus_agalactiae_2603V_prokka_GFS.csv
│   ├── window_2--agalactiae_515_prokka_GFS.csv
│   ├── window_4--Streptococcus_agalactiae_COH1_prokka_GFS.csv
│   ├── window_5--Streptococcus_agalactiae_CJB111_prokka_GFS.csv
│   ├── window_8--Streptococcus_agalactiae_NEM316_prokka_GFS.csv
│   └── window_9--Streptococcus_agalactiae_18RS21_prokka_GFS.csv
├── non_modular_queries_82tvei7d.faa
├── presence_absence
│   ├── window_1--Streptococcus_agalactiae_2603V_prokka.csv
│   ├── window_2--agalactiae_515_prokka.csv
│   ├── window_4--Streptococcus_agalactiae_COH1_prokka.csv
│   ├── window_5--Streptococcus_agalactiae_CJB111_prokka.csv
│   ├── window_8--Streptococcus_agalactiae_NEM316_prokka.csv
│   └── window_9--Streptococcus_agalactiae_18RS21_prokka.csv
└── windows_genbanks
    ├── gator_hits.tsv
    ├── window_1--Streptococcus_agalactiae_2603V_prokka.gbk
    ├── window_2--agalactiae_515_prokka.gbk
    ├── window_3--Streptococcus_agalactiae_A909_prokka.gbk
    ├── window_4--Streptococcus_agalactiae_COH1_prokka.gbk
    ├── window_5--Streptococcus_agalactiae_CJB111_prokka.gbk
    ├── window_6--Streptococcus_agalactiae_H36B_prokka.gbk
    ├── window_7--Streptococcus_agalactiae_515_prokka.gbk
    ├── window_8--Streptococcus_agalactiae_NEM316_prokka.gbk
    └── window_9--Streptococcus_agalactiae_18RS21_prokka.gbk

7 directories, 47 files</code></pre>
</div>
<p>The output folder <code>cpsg_gator/</code> for gator-gc includes the
following subfolders:</p>
<ul>
<li>
<strong>windows_genbanks</strong>: This folder contains GenBank
files, each generated for a specific window.</li>
<li>
<strong>presence_absence</strong>: Within this folder, you can find
gene-level presence-absence tables for each gator window in CSV format.
The tables are named based on the window number and the corresponding
GenBank filename.</li>
<li>
<strong>gator_scores</strong>: This folder houses tables with a
structure similar to the presence-absence tables. However, instead of
binary numbers, normal distributions were applied to each protein. The
highest distribution value is set to 1, and these values were multiplied
by the presence-absence values.</li>
<li>
<strong>concatenated_scores</strong>: This folder contains a CSV
file, which consolidates all the gator focal scores. Additionally, a
“clustermap_GFSs” file displays the distribution of gator windows based
on the gator focal scores using a heatmap and a dendrogram.</li>
</ul>
<p>Here is the resulting heatmap and dendrogram. You can view the images
in the file explorer on the left side of the server screen. Remember to
go to the specific folder to open the file.</p>
<p><a href="../fig/heatmap.svg">
<img src="../fig/heatmap.svg" alt="Heatmap resulting from cpsg gator-gc analysis" class="figure"></a></p>
<ul>
<li>
<strong>gator_conservation_plots</strong>: This folder includes
high-quality vectorized figures, each corresponding to a gator focal
window. These figures display the gene organization and cluster size.
The figure corresponding to the first window is shown below:</li>
</ul>
<p><a href="../fig/coservation_window_1.svg%20%20%20%20">
<img src="../fig/coservation_window_1.svg" alt="Conservation plot for the first window" class="figure"></a></p>
<p>In these figures required genes are represented in purple, optional
genes in orange, and genes not present in the protein queries files in
green. The transparency of each gene’s color is determined by the
conservation of homologous genes found in the gator windows.</p>
<ul>
<li>
<strong>gator_neighborhoods_plots</strong>: Here you can find
high-quality vectorized figures showcasing genome neighborhoods. Each
file corresponds to a gator focal window (top track), with the remaining
gator windows sorted based on the gator focal scores. This arrangement
positions the second top track as the most similar gator window, and the
last track at the bottom represents the most dissimilar gator window.
The genomic organization of the gator windows is flipped based on the
first required gene in the gator focal window, ensuring that the genomic
organization aligns with that of the gator focal window. Homology rails
are displayed, depicting the diamond protein alignment position hits in
the genes. The first one shoul look like this:</li>
</ul>
<p><a href="../fig/window_1_neighborhood.svg">
<img src="../fig/window_1_neighborhood.svg" alt="Fisrt window_1_neighborhood plot for the first window" class="figure"></a></p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<ul>
<li>GATOR-GC is an innovative tool that uses an enzyme-aware scoring
system and evolutionary principles to explore BGC diversity.</li>
<li>Unlike traditional methods, GATOR-GC offers flexibility in defining
the taxonomic scope and prioritizes the identification of novel
biosynthetic pathways.</li>
<li>GATOR-GC can be customized to search for essential and optional
enzymes, making it a powerful tool for targeted exploration.</li>
<li>Dynamic gene cluster diagrams and GATOR neighborhood visualizations
provide clear insights into gene conservation and genomic
relationships.</li>
</ul>
</div>
</div>
</div>
</section></section><section id="aio-09-metabolomics"><p>Content from <a href="09-metabolomics.html">Metabolomics workshop</a></p>
<hr>
<p>Last updated on 2026-02-20 |

        <a href="https://github.com/carpentries-incubator/genome-mining/edit/main/episodes/09-metabolomics.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> NA minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How can I evaluate the similarity between MS spectra?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand how GNPS molecular networks work.</li>
<li>Use raw mzML metabolomic data for annotation.</li>
<li>Visualize and explore molecular networks</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr class="half-width">
<p>In the previous sections, we explored the biosynthetic potential of
various strains by analyzing the BGCs within their genomes. We utilized
BiG-SCAPE to create BGC networks, which compare all the BGCs detected by
antiSMASH to determine their relatedness. Similarly, using metabolomics
we can investigate the metabolites produced by selected strains in the
laboratory. While crude extracts can be made to analyze the metabolites
of specific strains, it is also possible to study the metabolites
present in complex samples, such as soil, marine environments, host
organisms, or stool samples. There are two approaches to annotating
metabolomics data. The first comprises using the information obtained by
LC-MS, comparing the exact mass and UV spectra of each detected
metabolite through natural product databases. The introduction of the
global natural products social molecular networking <a href="https://gnps.ucsd.edu/ProteoSAFe/static/gnps-splash.jsp" class="external-link">GNPS</a>
platform for molecular networking (M. Wang et al., 2016), significantly
influenced dereplication techniques. Molecular networking groups
metabolites into molecular families (MFs), thereby improving the
annotation process of unknown metabolites.</p>
<p><a href="../fig/Molecular-network.png">
<img src="../fig/Molecular-network.png" alt="GNPS output can be directly visualized in the GNPS webpage, or using other visualization tools such as [Cytoscape](https://cytoscape.org/) " width="900" class="figure"></a></p>
</section><section><h2 class="section-heading" id="creating-a-gnps-account">Creating a GNPS account<a class="anchor" aria-label="anchor" href="#creating-a-gnps-account"></a>
</h2>
<hr class="half-width">
<p>Before starting this tutorial, it will be useful to have a GNPS
account. For that go to the <a href="https://gnps.ucsd.edu/ProteoSAFe/static/gnps-splash.jsp" class="external-link">GNPS</a>
webpage and select Create a new account.</p>
<p><a href="../fig/create_GNPSaccount.jpg">
<img src="../fig/create_GNPSaccount.jpg" alt="Create an account in GNPS" width="800" class="figure"></a></p>
<p>Then fill in the following information:</p>
<ul>
<li>Username</li>
<li>Name</li>
<li>Organization</li>
<li>Email</li>
<li>Password</li>
</ul>
<p>Wait for a confirmation email, and you now have a GNPS account.</p>
</section><section><h2 class="section-heading" id="download-mzmine-v3-9">Download MZMine v3.9<a class="anchor" aria-label="anchor" href="#download-mzmine-v3-9"></a>
</h2>
<hr class="half-width">
<ol style="list-style-type: decimal">
<li><p>First, go to the MZMine 3.9 release <a href="https://github.com/mzmine/mzmine/releases/tag/v3.9.0" class="external-link">MZMine
3.9</a></p></li>
<li><p>Select the installable file depending on your computer</p></li>
</ol>
<p><a href="../fig/Download_MZMine.jpg">
<img src="../fig/Download_MZMine.jpg" alt="MZmine 3, an MS data analysis platform " width="500" class="figure"></a></p>
<ol start="3" style="list-style-type: decimal">
<li>Double-click on the file, and install the software</li>
</ol></section><section><h2 class="section-heading" id="download-cytoscape">Download Cytoscape<a class="anchor" aria-label="anchor" href="#download-cytoscape"></a>
</h2>
<hr class="half-width">
<p>Cytoscape is a software frequently used to visualize networks, such
as BGC networks or molecular networks.</p>
<ol style="list-style-type: decimal">
<li><p>Go to <a href="https://cytoscape.org/download.html" class="external-link">Cytoscape
webpage</a></p></li>
<li><p>Click on download Cytoscape for your operating system</p></li>
<li><p>Install Cytoscape in your computer.</p></li>
</ol></section><section><h2 class="section-heading" id="download-the-metabolomics-dataset">Download the metabolomics dataset<a class="anchor" aria-label="anchor" href="#download-the-metabolomics-dataset"></a>
</h2>
<hr class="half-width">
<p>First, we need to go to Zenodo and download all the mzML raw data
collected from the described strains. <a href="https://zenodo.org/api/records/13352458/files-archive" class="external-link uri">https://zenodo.org/api/records/13352458/files-archive</a></p>
<p>After downloading the compressed file, we need to decompress it and
store the files in a folder on our computer.</p>
</section><section><h2 class="section-heading" id="import-the-dataset-and-use-the-batch-file-to-analyze-your-data">Import the dataset, and use the batch file to analyze your data<a class="anchor" aria-label="anchor" href="#import-the-dataset-and-use-the-batch-file-to-analyze-your-data"></a>
</h2>
<hr class="half-width">
<p>This data was collected from crude extracts from two marine
<em>Streptomyces</em>: <em>Streptomyces</em> sp. H-KF8, and
<em>Streptomyces</em> sp. Vc74B-19. Two media were used, ISP2 and ISP2
prepared with artificial seawater (ASW), to evaluate the effect of
replicating the natural environment from which these strains were
isolated.</p>
<p><a href="../fig/Datasets.png">
<img src="../fig/Datasets.png" alt="Data collection from *Streptomyces* sp. H-KF8, and *Streptomyces* sp. Vc74B-19." width="800" class="figure"></a></p>
<p>We downloaded 18 LC-MS/MS-derived files in mzML format. This data was
collected by Dr. Mauricio Caraballo-Rodriguez in the Dorrestein Lab, at
the University of California San Diego. There are files from each
strain, in ISP2 and ISP2-ASW, besides the crude extracts from the
culture media. The data is in triplicates.</p>
<p>Besides mzML files, there is a file metadata_table.tsv, that contains
all the relevant information from this dataset. Includes the names of
the samples, relevant data collection, and taxonomic information.</p>
<p>In addition, there is information relevant to the analysis, such as
the names of the strains, the media used for culturing, and the
antimicrobial activity. All this information is included in the format
ATTRIBUTE_*</p>
<p><a href="../fig/Metadata.jpg">
<img src="../fig/Metadata.jpg" alt="Data collection from *Streptomyces* sp. H-KF8, and *Streptomyces* sp. Vc74B-19." width="800" class="figure"></a></p>
<p>At last, there is a file named MZMine_FBMN_batch.xml that collects
all the information necessary for the analysis using MZMine</p>
</section><section><h2 class="section-heading" id="analysis-using-mzmine">Analysis using MZMine<a class="anchor" aria-label="anchor" href="#analysis-using-mzmine"></a>
</h2>
<hr class="half-width">
<p>Load batch file</p>
<p>Open MZMine3, click on “Open”, and then in “Batch Mode”</p>
<p><a href="../fig/batch_01.jpg">
<img src="../fig/batch_01.jpg" alt="Load batch file" width="400" class="figure"></a></p>
<p>Here you should select load, and search for your downloaded files on
your computer. Then select the MZMine_FBMN_batch.xml file In
confirmation, you should select Replace the batch steps.</p>
<p><a href="../fig/batch_02.jpg">
<img src="../fig/batch_02.jpg" alt="Load batch file" width="600" class="figure"></a></p>
<p>Then double-click on import MS data</p>
<p><a href="../fig/batch_03.jpg">
<img src="../fig/batch_03.jpg" alt="Load batch file" width="800" class="figure"></a></p>
<p>Select from your computer the 18 mzML files from this dataset</p>
<p><a href="../fig/batch_04.jpg">
<img src="../fig/batch_04.jpg" alt="Load batch file" width="800" class="figure"></a></p>
<p>After this, every file should be included in the batch-processing
mode. Select OK afterward so the files begin to process in the
meantime.</p>
<p><a href="../fig/batch_05.jpg">
<img src="../fig/batch_05.jpg" alt="Load batch file" width="800" class="figure"></a></p>
<p>Briefly, MZMine is now detecting all the masses present in your
samples, grouping them, and then aligning them, so you can know in which
sample each detected spectrum is present</p>
<p>You should note, that the parameters used are specific for this
dataset, you might need to change some of the values when analyzing your
own samples.</p>
<p>For more information: Nothias, LF., Petras, D., Schmid, R. et
al. Feature-based molecular networking in the GNPS analysis environment.
Nat Methods 17, 905–908 (2020). <a href="https://doi.org/10.1038/s41592-020-0933-6" class="external-link uri">https://doi.org/10.1038/s41592-020-0933-6</a></p>
</section><section><h2 class="section-heading" id="explore-the-structure-of-our-data">Explore the structure of our data<a class="anchor" aria-label="anchor" href="#explore-the-structure-of-our-data"></a>
</h2>
<hr class="half-width">
<p>In the “MS data files” from MZMine you can observe all the 18
LC-MS/MS files in mzML format that we loaded</p>
<p><a href="../fig/MZMine_datafiles.jpg">
<img src="../fig/MZMine_datafiles.jpg" alt="Datasets" width="400" class="figure"></a></p>
<p>Let’s inspect what the two of these files look like. We could select
one file from <em>Streptomyces</em> sp. H-KF8 in ISP2, and one in
ISP2-ASW We can select both files, then right-click and select “Show
chromatograms”</p>
<p><a href="../fig/chromatogram_01.jpg">
<img src="../fig/chromatogram_01.jpg" alt="Chromatogram" width="400" class="figure"></a></p>
<p>Here we can select the mass range that we want to observe. Since we
want to see all the spectra detected, click “Auto range”. It will
automatically will select masses ranging from 100 <em>m/z</em> to almost
3,749 <em>m/z</em> Click “OK” then</p>
<p><a href="../fig/chromatogram_02.jpg">
<img src="../fig/chromatogram_02.jpg" alt="Chromatogram" width="500" class="figure"></a></p>
<p>The software will display the Total Ion Chromatogram (TIC) from both
samples. In this case, strain H-KF8 is displayed in pink when cultured
in ISP2-ASW, and in black when cultured in ISP2</p>
<p><a href="../fig/chromatogram_03.jpg">
<img src="../fig/chromatogram_03.jpg" alt="Chromatogram" width="800" class="figure"></a></p>
<p>We can select a section of the chromatogram to inspect the
differences of the metabolomic profiles of these samples</p>
<p><a href="../fig/chromatogram_04.jpg">
<img src="../fig/chromatogram_04.jpg" alt="Chromatogram" width="800" class="figure"></a></p>
<p>We can observe that several spectra are produced exclusively by
strain H-KF8 in ISP2-ASW</p>
<p><a href="../fig/chromatogram_05.jpg">
<img src="../fig/chromatogram_05.jpg" alt="Chromatogram" width="800" class="figure"></a></p>
</section><section><h2 class="section-heading" id="analyze-the-final-output-from-the-analysis">Analyze the final output from the analysis<a class="anchor" aria-label="anchor" href="#analyze-the-final-output-from-the-analysis"></a>
</h2>
<hr class="half-width">
<p>After processing all files. We should look at the feature lists tab
from MZMine. There we can observe that we have a file called “Aligned
feature list 13C gaps”. Double-click on that</p>
<p><a href="../fig/feat_list_01.jpg">
<img src="../fig/feat_list_01.jpg" alt="Feature list" width="400" class="figure"></a></p>
<p>Here we can observe the feature list, where each row is one detected
MS spectra with its <em>m/z</em> and retention time (RT). Each column is
one of the 18 samples. If MS spectra are detected in a sample, then the
height of the peak is displayed in the table</p>
<p><a href="../fig/feat_list_02.jpg">
<img src="../fig/feat_list_02.jpg" alt="Feature List" width="800" class="figure"></a></p>
</section><section><h2 class="section-heading" id="remove-media-blanks-ms-spectra">Remove media blanks MS spectra<a class="anchor" aria-label="anchor" href="#remove-media-blanks-ms-spectra"></a>
</h2>
<hr class="half-width">
<p>Now we want to remove all the MS spectra that are part of the culture
media and not produced by our strains.</p>
<p>For that, we need to go to “Feature List Methods”, then click on
“Feature List Filtering”, and then on “Feature List Blank
Subtraction”</p>
<p><a href="../fig/blank_01.jpg">
<img src="../fig/blank_01.jpg" alt="Blank substraction" width="600" class="figure"></a></p>
<p>In the “Blank/Control raw data files” section, we need to select
“Specific raw data files”, and then select</p>
<p><a href="../fig/blank_02.jpg">
<img src="../fig/blank_02.jpg" alt="Blank substraction" width="600" class="figure"></a></p>
<p>Here we need to select all the samples that belong to the crude
extracts from the Media. There are 6 in total. Press OK afterward</p>
<p><a href="../fig/blank_03.jpg">
<img src="../fig/blank_03.jpg" alt="Blank substraction" width="600" class="figure"></a></p>
<p>Now we have two Feature lists</p>
<ol style="list-style-type: decimal">
<li>Aligned feature list 13C gaps. That is the original feature list
including media MS spectra</li>
<li>Aligned feature list 13C gaps subtracted. Feature list with media
blanks removed</li>
</ol>
<p><a href="../fig/blank_04.jpg">
<img src="../fig/blank_04.jpg" alt="Blank substraction" width="400" class="figure"></a></p>
</section><section><h2 class="section-heading" id="export-feature-lists-in-gnps-format">Export Feature lists in GNPS format<a class="anchor" aria-label="anchor" href="#export-feature-lists-in-gnps-format"></a>
</h2>
<hr class="half-width">
<p>We are going to export both Feature lists.</p>
<p>First, select “Aligned feature list 13C gaps”. Then go to “Feature
List Methods”, “Export Feature List”, and select Molecular “networking
files”</p>
<p><a href="../fig/export_01.jpg">
<img src="../fig/export_01.jpg" alt="Export files" width="600" class="figure"></a></p>
<p>Then click “Select”, and in “File name” write the name that you want
your files to be named. In this case, I selected
“GM_workshop_Featurelist_complete”. So I know that this file is from the
Latin American genome mining workshop and that the feature list includes
the media blank MS spectra. then press “save”</p>
<p><a href="../fig/export_02.jpg">
<img src="../fig/export_02.jpg" alt="Export files" width="600" class="figure"></a></p>
<p>Make sure that in Filter rows you select “MS2 or ION IDENTITY”, so
only MS spectra with MS2 are selected.</p>
<p><a href="../fig/export_03.jpg">
<img src="../fig/export_03.jpg" alt="Export files" width="600" class="figure"></a></p>
<p>Then, in your selected folder, you should have two files</p>
<ol style="list-style-type: decimal">
<li>GM_workshop_Featurelist_complete_quant.csv</li>
</ol>
<p>This file is a table that includes all the feature lists in your
samples. Again, each row is an MS spectrum, and each column is each of
the 18 samples.</p>
<ol start="2" style="list-style-type: decimal">
<li>GM_workshop_Featurelist_complete_quant.mgf</li>
</ol>
<p>This file contains the information on each spectrum. Contains the
parent mass in <em>m/z</em>, and the <em>m/z</em> values of each
fragment from that spectra, with the peak intensity of each
spectrum.</p>
<p><a href="../fig/export_04.jpg">
<img src="../fig/export_04.jpg" alt="Export files" width="600" class="figure"></a></p>
<p>Now we need to repeat the export step but with the media blanks
removed. This time the files will be named
“GM_workshop_Featurelist_filtered” so we can know that there is no MS
spectra that are originally from the culture media.</p>
<p>After this, we have 4 files. And we are done with the processing
steps in MZMine 3</p>
</section><section><h2 class="section-heading" id="create-a-molecular-network">Create a molecular network<a class="anchor" aria-label="anchor" href="#create-a-molecular-network"></a>
</h2>
<hr class="half-width">
<p>Go into <a href="https://gnps.ucsd.edu/ProteoSAFe/static/gnps-splash.jsp" class="external-link">GNPS</a>
webpage</p>
<p>and login using your username and password</p>
<p>Then we should go to “Advanced Analysis Tools”, and Select “Analyze”
in the Feature Networking</p>
<p><a href="../fig/FBMN_01.jpg">
<img src="../fig/FBMN_01.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>Then we should write a title for our network. We could use something
like “GM_workshop_FBMN_filtered”, because we are going to use the
feature list with the media blanks removed.</p>
<p>Following that, in “File Selection”, click on “Select Input File”</p>
<p><a href="../fig/FBMN_02.jpg">
<img src="../fig/FBMN_02.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>We now need to upload our feature lists and our metadata table. Click
on “Upload files”</p>
<p><a href="../fig/FBMN_03.jpg">
<img src="../fig/FBMN_03.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>Here we can create a folder in our GNPS. I created a new folder
called “LATAM_GM_workshop”. Click on that folder. Then in the “File Drag
and Drop”, drag and drop the following files:</p>
<ol style="list-style-type: decimal">
<li>GM_workshop_Featurelist_filtered.mgf</li>
<li>GM_workshop_Featurelist_filtered_quant.csv</li>
<li>metadata_table.tsv</li>
</ol>
<p><a href="../fig/FBMN_04.jpg">
<img src="../fig/FBMN_04.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>Return to the “Select Input Files” after uploading your files. You
should be able to see the three uploaded files in your directory</p>
<p><a href="../fig/FBMN_05.jpg">
<img src="../fig/FBMN_05.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>We are going to select the “GM_workshop_Featurelist_filtered.mgf”
file as “MS2 file in MGF format”</p>
<p><a href="../fig/FBMN_06.jpg">
<img src="../fig/FBMN_06.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>then we are going to select
“GM_workshop_Featurelist_filtered_quant.csv” as the “Feature
Quantification Table”</p>
<p><a href="../fig/FBMN_07.jpg">
<img src="../fig/FBMN_07.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>Finally, we are going to select the “metadata_table.tsv” as our
“Sample Metadata Table”</p>
<p><a href="../fig/FBMN_08.jpg">
<img src="../fig/FBMN_08.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>In the “Selected Files” section, we should be able to see the three
files in their corresponding sections. Click on finish selection after
checking everything is ok.</p>
<p><a href="../fig/FBMN_09.jpg">
<img src="../fig/FBMN_09.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>After selecting the files, we need to adjust the parameters for our
network Since our data was collected using a high-resolution LC-MS/MS,
we could adjust the</p>
<ul>
<li>“Precursor Ion Mass Tolerance” to 0.02 Da This value affects the
clustering of nearly identical MS/MS spectra through MS-Cluster.</li>
<li>“Fragment Ion Mass Tolerance” to 0.02 Da Sets the allowable
deviation in m/z values for fragment ions when clustering MS/MS
spectra</li>
</ul>
<p><a href="../fig/FBMN_10.jpg">
<img src="../fig/FBMN_10.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>Then we need to select our thresholds to create the molecular
network</p>
<ul>
<li>“min pair cos” to 0.7 The minimum cosine score required between a
pair of consensus MS/MS spectra for an edge to be created in the
molecular network.</li>
<li>“Minimum matched fragment ions” to 6 Specify the minimum number of
common fragment ions that two consensus MS/MS spectra must share to be
connected by an edge in the molecular network.</li>
</ul>
<p><a href="../fig/FBMN_11.jpg">
<img src="../fig/FBMN_11.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>After this, write your email, so you know when your network is
finished. Then click “Submit”</p>
<p><a href="../fig/FBMN_12.jpg">
<img src="../fig/FBMN_12.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>For more information about the rest of the parameters used for
molecular networking:</p>
<ul>
<li><a href="https://ccms-ucsd.github.io/GNPSDocumentation/networking/" class="external-link">visit
the GNPS documentation</a></li>
<li>Also visit Aron, A.T., Gentry, E.C., McPhail, K.L. et al. <a href="https://doi.org/10.1038/s41596-020-0317-5" class="external-link">Reproducible molecular
networking of untargeted mass spectrometry data using GNPS.</a> Nat
Protoc 15, 1954–1991 (2020).</li>
</ul></section><section><h2 class="section-heading" id="statistics-analysis-using-fbmn-stats-guide-web-server">Statistics analysis using FBMN STATS guide web server<a class="anchor" aria-label="anchor" href="#statistics-analysis-using-fbmn-stats-guide-web-server"></a>
</h2>
<hr class="half-width">
<p>Now we want to observe how different are the metabolic profiles of
our samples. For that, we are going to calculate a Principal Coordinate
Analysis (PCoA)</p>
<p>Go into <a href="https://fbmn-statsguide.gnps2.org/" class="external-link">FBMN STATS
guide</a></p>
<p>Select “Data Preparation”</p>
<p><a href="../fig/FBMN-STATS_01.jpg">
<img src="../fig/FBMN-STATS_01.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>In file origin select “Quantification table and metadata files”.</p>
<ul>
<li>In the Quantification Table section, select the MZMine output file
“GM_workshop_Featurelist_complete_quant.csv”. This table contains the
information of the two strains and the media. We want to observe how
different our samples’ metabolomic profiles are, in comparison with the
culture media.</li>
<li>in the Meta Data Table section, include our metadata file
“metadata_table.tsv”.</li>
</ul>
<p><a href="../fig/FBMN-STATS_02.jpg">
<img src="../fig/FBMN-STATS_02.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>After loading your files, click “Submit Data for Statistics!”.</p>
<p><a href="../fig/FBMN-STATS_03.jpg">
<img src="../fig/FBMN-STATS_03.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>Check that your data have been properly submitted by checking that
you have the “Data preparation was successful!”.</p>
<p><a href="../fig/FBMN-STATS_04.jpg">
<img src="../fig/FBMN-STATS_04.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>Now we should go to the “PERMANOVA &amp; PCoA” section.</p>
<p><a href="../fig/FBMN-STATS_05.jpg">
<img src="../fig/FBMN-STATS_05.jpg" alt="FBMN" width="300" class="figure"></a></p>
<p>Select Principal Coordinate Analysis.</p>
<p><a href="../fig/FBMN-STATS_06.jpg">
<img src="../fig/FBMN-STATS_06.jpg" alt="FBMN" width="600" class="figure"></a></p>
<p>Now we could change how we want to color our samples. Select
“attribute for multivariate analysis” ATTRIBUTE_media. This will color
our samples according to the media (ISP2 or ISP2-ASW)</p>
<p><a href="../fig/FBMN-STATS_07.jpg">
<img src="../fig/FBMN-STATS_07.jpg" alt="FBMN" width="600" class="figure"></a></p>
<p>We can observe that there is a difference between the samples
prepared with ISP2 and ISP2-ASW</p>
<p><a href="../fig/FBMN-STATS_08.jpg">
<img src="../fig/FBMN-STATS_08.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>Now if we color our samples by strain, selecting ATTRIBUTE_strain in
“attribute for multivariate analysis”</p>
<p>We can observe that <em>Streptomyces</em> sp. Vc74B-19 metabolic
profile is quite different than the ones from <em>Streptomyces</em> sp.
H-KF8 and the culture media.</p>
<p>Also, it is possible to observe that <em>Streptomyces</em> sp. H-KF8
metabolomic profile differs more from the media when cultured in
ISP2-ASW</p>
<p><a href="../fig/FBMN-STATS_09.jpg">
<img src="../fig/FBMN-STATS_09.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>There are several statistics that you can do using FBMN STATS guide,
you can check the preprint <a href="https://doi.org/10.26434/CHEMRXIV-2023-WWBT0" class="external-link">here</a>.</p>
</section><section><h2 class="section-heading" id="visualize-the-network-using-cytoscape">Visualize the network using Cytoscape<a class="anchor" aria-label="anchor" href="#visualize-the-network-using-cytoscape"></a>
</h2>
<hr class="half-width">
<p>We need to check if our GNPS network is done processing. You might
have received an email, otherwise, go to your GNPS account and click on
“Jobs”</p>
<p><a href="../fig/cytoscape_01.jpg">
<img src="../fig/cytoscape_01.jpg" alt="FBMN" width="600" class="figure"></a></p>
<p>If your network is Done, then click on it. Afterward, click on
“Direct Cytoscape Preview/Download”</p>
<p><a href="../fig/cytoscape_02.jpg">
<img src="../fig/cytoscape_02.jpg" alt="FBMN" width="600" class="figure"></a></p>
<p>If your network is not Done, then we could use a previously computed
network, that holds the same data that we used in this workshop</p>
<p><a href="https://cytoscape.gnps2.org/process?task=568cc93ed2e84f20b2b94bd186b48ee8#%7B%7D" class="external-link">Cytoscape</a></p>
<p>Click on “Download Cytoscape File”, and save it in your computer</p>
<p><a href="../fig/cytoscape_03.jpg">
<img src="../fig/cytoscape_03.jpg" alt="FBMN" width="600" class="figure"></a></p>
<p>Open the network on Cytoscape. We want to change the style of the
network and color the nodes according to the strains. Click on
“Style”</p>
<p><a href="../fig/cytoscape_04.jpg">
<img src="../fig/cytoscape_04.jpg" alt="FBMN" width="600" class="figure"></a></p>
<p>Select the pie chart icon on Image/Chart. This will help us figure
out the abundance of each MS spectra, depending on the metadata
parameter that we choose</p>
<p><a href="../fig/cytoscape_05.jpg">
<img src="../fig/cytoscape_05.jpg" alt="FBMN" width="600" class="figure"></a></p>
<p>In this case, we will select the two strains, and also the media
Although we removed all the MS spectra present in the media, there might
still be some nodes with MS spectra detected in some of the media
samples.</p>
<p><a href="../fig/cytoscape_06.jpg">
<img src="../fig/cytoscape_06.jpg" alt="FBMN" width="600" class="figure"></a></p>
<p>If we click “Apply”, we can observe the nodes present in each strain,
and the media. In this case, the light blue means the MS spectra
detected in <em>Streptomyces</em> sp. Vc74B-19</p>
<p>In this Molecular Family, we can observe some MS spectra with
annotations. Some are similar to Urdamycinone B, Dehydroxyaquayamycin,
and other Angucycline-related compounds.</p>
<p><a href="../fig/cytoscape_07.jpg">
<img src="../fig/cytoscape_07.jpg" alt="FBMN" width="800" class="figure"></a></p>
<p>We can color now the nodes depending on the media where they are
detected. Select the pie chart again in “Image/Chart”, and now select
ISP2 and ISP2-ASW. This will color ISP2 in red, and ISP2-ASW in light
blue</p>
<p><a href="../fig/cytoscape_08.jpg">
<img src="../fig/cytoscape_08.jpg" alt="FBMN" width="600" class="figure"></a></p>
<p>We can observe that some nodes are detected mostly in ISP2. However,
several angucycline-related compounds are detected almost exclusively in
ISP2-ASW</p>
<p><a href="../fig/cytoscape_09.jpg">
<img src="../fig/cytoscape_09.jpg" alt="FBMN" width="800" class="figure"></a></p>
</section><section><h2 class="section-heading" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<hr class="half-width">
<ul>
<li>Wang, M., Carver, J., Phelan, V. et al. Sharing and community
curation of mass spectrometry data with Global Natural Products Social
Molecular Networking. Nat Biotechnol 34, 828–837 (2016). <a href="https://doi.org/10.1038/nbt.3597" class="external-link uri">https://doi.org/10.1038/nbt.3597</a>.</li>
<li>Schmid, R., Heuckeroth, S., Korf, A. et al. Integrative analysis of
multimodal mass spectrometry data in MZmine 3. Nat Biotechnol 41,
447–449 (2023). <a href="https://doi.org/10.1038/s41587-023-01690-2" class="external-link uri">https://doi.org/10.1038/s41587-023-01690-2</a>.</li>
<li>Nothias, LF., Petras, D., Schmid, R. et al. Feature-based molecular
networking in the GNPS analysis environment. Nat Methods 17, 905–908
(2020). <a href="https://doi.org/10.1038/s41592-020-0933-6" class="external-link uri">https://doi.org/10.1038/s41592-020-0933-6</a>
</li>
<li>Aron, A.T., Gentry, E.C., McPhail, K.L. et al. Reproducible
molecular networking of untargeted mass spectrometry data using GNPS.
Nat Protoc 15, 1954–1991 (2020). <a href="https://doi.org/10.1038/s41596-020-0317-5" class="external-link uri">https://doi.org/10.1038/s41596-020-0317-5</a>
</li>
<li>Shah, A. K. P., Walter, A., Ottosson, F., Russo, F., Navarro-Díaz,
M., Boldt, J., Kalinski, J.-C., Kontou, E. E., Elofson, J., Polyzois,
A., González-Marín, C., Farrell, S., Aggerbeck, M. R., Pruksatrakul, T.,
Chan, N., Wang, Y., Pöchhacker, M., Brungs, C., Cámara, B., … Petras, D.
(2023). The Hitchhiker’s Guide to Statistical Analysis of Feature-based
Molecular Networks from Non-Targeted Metabolomics Data. ChemRxiv, 1–83.
<a href="https://doi.org/10.26434/CHEMRXIV-2023-WWBT0" class="external-link uri">https://doi.org/10.26434/CHEMRXIV-2023-WWBT0</a>
</li>
</ul>
<p>In the future it will include:</p>
<ul>
<li>
<a href="https://www.tfleao.com/general-8" class="external-link uri">https://www.tfleao.com/general-8</a>,</li>
<li><a href="https://pairedomicsdata.bioinformatics.nl/" class="external-link">Paired
omics</a></li>
<li>In the future it will include: SIRIUS: Kai Dührkop, Markus
Fleischauer, Marcus Ludwig, Alexander A. Aksenov, Alexey V. Melnik,
Marvin Meusel, Pieter C. Dorrestein, Juho Rousu, and Sebastian Böcker,
SIRIUS 4: Turning tandem mass spectra into metabolite structure
information. Nature Methods 16, 299–302, 2019.</li>
</ul>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<ul>
<li>Data is generated using Liquid Chromatography coupled to a tandem
mass spectrometer (LC-MS/MS or MS2).</li>
<li>Dereplication is the process of identifying previously known
compounds.</li>
<li>Molecular networking is a computational method that organizes MS2
data based on spectral similarity, allowing us to infer relationships
between chemical structures</li>
<li>Feature-Based Molecular Networking (FBMN) enhances classical
molecular networking by integrating relative quantitative data, enabling
more robust metabolomics statistical analysis.</li>
</ul>
</div>
</div>
</div>
</section></section><section id="aio-10-OtherResources"><p>Content from <a href="10-OtherResources.html">Other Resources</a></p>
<hr>
<p>Last updated on 2026-02-20 |

        <a href="https://github.com/carpentries-incubator/genome-mining/edit/main/episodes/10-OtherResources.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 20 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What else can I do?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>First learning objective. (FIXME)</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="other-resources">Other resources<a class="anchor" aria-label="anchor" href="#other-resources"></a>
</h2>
<hr class="half-width">
<p>We have seen how to conduct a genome mining project with some
particular characteristics. However, each research project may need some
deviations to these workflows. You may need to make use of RNA-Seq data
to search just for some specific domains instead of complete BGCs or to
combine metabolomics or proteomics with genomic data. Here we will
provide a list of resources that are also useful for genome mining
projects.</p>
<div class="section level3">
<h3 id="assemblers">Assemblers<a class="anchor" aria-label="anchor" href="#assemblers"></a>
</h3>
<p>First, since BGCs usually encompass genes with high levels of repeat
regions (mainly NRPSs and PKSs), genome assemblers are not always
capable of reconstructing BGCs and sometimes divide them into two
contigs. To overcome this difficulty, you can use biosyntheticSPADes to
assemble your reads into complete BGCs. This algorithm is implemented
within the <a href="https://github.com/ablab/spades" class="external-link">SPAdes</a>
tool.</p>
</div>
<div class="section level3">
<h3 id="other-genome-mining-programs">Other genome mining programs<a class="anchor" aria-label="anchor" href="#other-genome-mining-programs"></a>
</h3>
<p>Depending on your interests, you can use some alternative program to
search for BGCs in your genomes. The <a href="https://www.secondarymetabolites.org/" class="external-link">Secondary Metabolite
Bioinformatics Portal</a> lists and introduces many diverse
bioinformatic tools that pursue this goal. For example, <a href="https://bitbucket.org/tilmweber/clusean/src/master/" class="external-link">CLUSEAN</a>
and <a href="https://npdomainseeker.sdsc.edu/" class="external-link">NaPDoS</a> allow to
detect PKS and NRPS domains, while <a href="https://bagel4.molgenrug.nl/" class="external-link">BAGEL</a>, <a href="https://www.ripp.rodeo/" class="external-link">RODEO</a> and <a href="https://www.nii.ac.in/~priyesh/lantipepDB/new_predictions/index.php" class="external-link">RiPPMiner</a>
are dedicated to find ribosomally synthesized and post-translationally
modified peptides (RIPPs). Then, implementing a further step, <a href="https://arts.ziemertlab.com" class="external-link">ARTS</a> prioritizes previously
detected BGCs (with antiSMASH) and identifies drug targets from these
data.</p>
<p>There are also some tools and workflows that are dedicated to finding
new biosynthetic systems (new types of BGCs or biosynthetic genes).
Among them, as it was explained in previous lessons, <a href="https://github.com/nselem/evomining" class="external-link">EvoMining</a> arose as a
promising tool to detect biosynthetic enzymes that may have evolved from
developing core functions (‘central’ or ‘primary’ metabolism) to carry
on specialized functions (secondary metabolism). Likewise, <a href="https://github.com/petercim/ClusterFinder" class="external-link">ClusterFinder</a>
enables the prediction of uncharacterized BGCs in genomes through
different algorithms.</p>
</div>
<div class="section level3">
<h3 id="tools-for-metabolomics">Tools for metabolomics<a class="anchor" aria-label="anchor" href="#tools-for-metabolomics"></a>
</h3>
<p>You can also try to link diverse -omics data with genome predictions.
<a href="https://sema-trap.ziemertlab.com/" class="external-link">SeMa-Trap</a> allows the use
RNA-Seq data to find co-expression patterns between certain genes and
BGCs. Similarly, if you need to combine metabolomic/proteomic data with
putative BGCs-products, you can use <a href="https://pep2path.sourceforge.net" class="external-link">Pep2Path</a>, <a href="https://github.com/mohimanilab/MetaMiner" class="external-link">MetaMiner</a> or <a href="https://gnps.ucsd.edu/ProteoSAFe/static/gnps-splash.jsp?redirect=auth" class="external-link">GNPS</a>,
amongst others.</p>
</div>
<div class="section level3">
<h3 id="phylogeny-based-tools">Phylogeny based tools<a class="anchor" aria-label="anchor" href="#phylogeny-based-tools"></a>
</h3>
<p>Finally, to prioritize genomes or species, <a href="https://automlst.ziemertlab.com/" class="external-link">autoMLST</a> allows you not only
to find phylogenetically closed strains/species from your input genome
but also to explore the biosynthetic potential of those relatives.</p>
</div>
<div class="section level3">
<h3 id="index-of-novelty">Index of novelty<a class="anchor" aria-label="anchor" href="#index-of-novelty"></a>
</h3>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9973853/" class="external-link">Novelty
Index</a></p>
</div>
<div class="section level3">
<h3 id="visualize-annotated-genomes">Visualize annotated genomes<a class="anchor" aria-label="anchor" href="#visualize-annotated-genomes"></a>
</h3>
<p><a href="https://github.com/paulstothard/cgview" class="external-link">CGView</a></p>
</div>
<div class="section level3">
<h3 id="specialized-data-bases">Specialized data bases<a class="anchor" aria-label="anchor" href="#specialized-data-bases"></a>
</h3>
<ul>
<li>BGCs in human Microbiome <a href="https://ccb-web.cs.uni-saarland.de/abc_humi/" class="external-link">Humi</a>
</li>
</ul>
</div>
<div class="section level3">
<h3 id="massive-studies">MAssive studies<a class="anchor" aria-label="anchor" href="#massive-studies"></a>
</h3>
<ul>
<li><a href="https://pubmed.ncbi.nlm.nih.gov/33010170/" class="external-link">Big FAM</a></li>
<li><a href="https://www.nature.com/articles/s41587-020-0718-6" class="external-link">Earth
Microbiome Project</a></li>
<li><a href="https://www.nature.com/articles/s41467-023-43000-z" class="external-link">Soil
Microbiomes</a></li>
</ul>
</div>
</section><section><h2 class="section-heading" id="carpentries-philosophy">Carpentries Philosophy<a class="anchor" aria-label="anchor" href="#carpentries-philosophy"></a>
</h2>
<hr class="half-width">
<p>A good lesson should be as complete and transparent as easy to teach
by any instructor. Carpentries lessons are developed for the community;
now, you are part of us. This lesson is being developed, and we are sure
that you can collaborate and help us improve it.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<ul>
<li>First key point. Brief Answer to questions. (FIXME)</li>
</ul>
</div>
</div>
</div>
</section></section>
</div>
    </main>
</div>
<!-- END  : inst/pkgdown/templates/content-extra.html -->

      </div>
<!--/div.row-->
      		<footer class="row footer mx-md-3"><hr>
<div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://github.com/carpentries-incubator/genome-mining/edit/main/README.md" class="external-link">Edit on GitHub</a>

	
        | <a href="https://github.com/carpentries-incubator/genome-mining/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/carpentries-incubator/genome-mining/" class="external-link">Source</a></p>
        <p>
        <a href="https://github.com/carpentries-incubator/genome-mining/blob/main/" class="external-link">Cite</a>
        | <a href="mailto:axel.ramos3737@gmail.com">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
		</div>
		<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.18.5" class="external-link">sandpaper (0.18.5)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.9" class="external-link">pegboard (0.7.9)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.9" class="external-link">varnish (1.0.9)</a></p>
		</div>
		</footer>
</div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "LearningResource",
  "@id": "https://carpentries-incubator.github.io/genome-mining/instructor/aio.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/LearningResource/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "software, data, lesson, The Carpentries",
  "name": "All in One View",
  "creativeWorkStatus": "active",
  "url": "https://carpentries-incubator.github.io/genome-mining/instructor/aio.html",
  "identifier": "https://carpentries-incubator.github.io/genome-mining/instructor/aio.html",
  "dateCreated": "2021-06-20",
  "dateModified": "2026-02-20",
  "datePublished": "2026-02-20"
}

  </script><script>
		feather.replace();
	</script><!-- Matomo --><script>
          var _paq = window._paq = window._paq || [];
          /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
          _paq.push(["setDocumentTitle", document.domain + "/" + document.title]);
          _paq.push(["setDomains", ["*.lessons.carpentries.org","*.datacarpentry.github.io","*.datacarpentry.org","*.librarycarpentry.github.io","*.librarycarpentry.org","*.swcarpentry.github.io", "*.carpentries.github.io"]]);
          _paq.push(["setDoNotTrack", true]);
          _paq.push(["disableCookies"]);
          _paq.push(["trackPageView"]);
          _paq.push(["enableLinkTracking"]);
          (function() {
              var u="https://matomo.carpentries.org/";
              _paq.push(["setTrackerUrl", u+"matomo.php"]);
              _paq.push(["setSiteId", "1"]);
              var d=document, g=d.createElement("script"), s=d.getElementsByTagName("script")[0];
              g.async=true; g.src="https://matomo.carpentries.org/matomo.js"; s.parentNode.insertBefore(g,s);
          })();
        </script><!-- End Matomo Code -->
</body>
</html><!-- END:   inst/pkgdown/templates/layout.html-->

